--- 
title: "ANALYSIS OF SINGLE CELL RNA-SEQ DATA"
author: ["Orr Ashenberg", "Dana Silverbush", "Kirk Gosik"]
date: "03/23/2020 - 03/27/2020"
site: bookdown::bookdown_site
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: true
description: "This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook."
url: 'https://broadinstitute.github.io/2020_scWorkshop/'
github-repo: "broadinstitute/2019_scWorkship"
cover-image: "images/cover.png"
---

# Introduction

## COURSE OVERVIEW

In recent years single cell RNA-seq (scRNA-seq) has become widely used for transcriptome analysis in many areas of biology. In contrast to bulk RNA-seq, scRNA-seq provides quantitative measurements of the expression of every gene in a single cell. However, to analyze scRNA-seq data, novel methods are required and some of the underlying assumptions for the methods developed for bulk RNA-seq experiments are no longer valid. In this course we will cover all steps of the scRNA-seq processing, starting from the raw reads coming off the sequencer. The course includes common analysis strategies, using state-of-the-art methods and we also discuss the central biological questions that can be addressed using scRNA-seq.


## TARGETED AUDIENCE & ASSUMED BACKGROUND

This course is aimed at researchers and technical workers who are or will be analyzing scRNA-seq data. The material is suitable both for experimentalists who want to learn more about data-analysis as well as computational biologists who want to learn about scRNASeq methods. Examples demonstrated in this course can be applied to any experimental protocol or biological system.

The requirements for this course are:
  1.  Working knowledge of unix (managing files, running programs)
  2. Programming experience in R (writing a function, basic I/O operations, variable types, using packages). Bioconductor  experience is a plus. 
  3. Familiarity with next-generation sequencing data and its analyses (using alignment and quantification tools for bulk sequencing data)


## COURSE FORMAT

The course will be delivered over the course of five days. Each day will include a lecture and laboratory component. The lecture will introduce the topics of discussion and the laboratory sessions will be focused on practical hands-on analysis of scRNA-seq data. These sessions will involve a combination of both mirroring exercises with the instructor to demonstrate a skill as well as applying these skills on your own to complete individual exercises. After and during each exercise, interpretation of results will be discussed as a group. Computing will be done using a combination of tools installed on the attendees laptop computer and web resources accessed via web browser.



## Getting Started


## SESSION CONTENT

### Monday
  - Classes from 08:00 to 16:00 (lunch break-1 hr, 40 min of total coffee breaks)
  - Shared Google doc - course notes, ideas/questions/challenges/interesting topics you would like to explore.

#### Lecture 1 – scRNA-Seq experimental design (Orr)

  - Overview of course
  - General introduction: cell atlas overviews
  - Comparison of Bulk and single cell RNA-Seq
  - Overview of available scRNA-seq technologies (10x) and experimental protocols
  - scRNA-Seq experimental design and analysis workflow?

#### Lab 1 – Understanding sequencing raw data, downloading Docker if not done already (Kirk)

Lab based around data wrangling from public data repositories: get data from 10x website, single cell portal, from GEO (fastqs, counts)
  - Shell and Unix commands to navigate directories, create folders, open files
  - Raw file formats 

#### Lecture 2-3 - Intro to Data processing: from bcl file to bam file
**Transcriptome quantification: from bam file to counts (Dana)**

  - scRNA-Seq processing workflow starting with choice of sequencer (NextSeq, HiSeq, MiSeq) / barcode swapping and bcl files
  - Overview of Popular tools and algorithms
  - Common single-cell analyses and interpretation
  - Sequencing data: alignment and quality control
  - Looking at cool things in alignment like where reads are, mutations, splicing
  - Read & UMI counting (Kallisto alignment-free pseudocounts as well), how RSEM works (length dependence, sequencing depth, multimapping reads), CellRanger (dropest), bustools
  - 10x barcode structure and links to Perturb-seq
  - Gene length & coverage
  - Gene expression units (count data Smart-seq2 counts or 10x UMIs vs expression data)

#### Lab 2 – Processing raw scRNA-Seq data (Dana), Docker setup (Kirk)
  - Data outputs from different scRNAseq technologies (10x, Smart-seq2) - process both?
  - Demultiplexing sequencing data
  - Read Quality Control (CellRanger, dropEst, fastqc)
  - Run bowtie2 on 2 wells to demonstrate alignment
  - Read alignment and visualization (kallisto, RSEM, Igviewer)
  - Demultiplexing
  - FastQC
  - Align (STAR/TOPHAT/Kallisto)
  - IGViewer - what do we want here? I use it for mutation detections, copying sequences, searching for alternative splicing.

#### Flash talks (1.5 hr, break into 2 groups of 13)

1 slide advertising or summarizing the poster. So you can introduce yourselves and we can get to know each other. No questions, 2 minutes. Two sessions, 15 people each.

### Tuesday – Classes from 08:00 to 16:00

#### Lab 3 - Introduction to R (Kirk)

  - Some R overview slides, https://r4ds.had.co.nz/
  - Installing packages
  - Data-types
  - Data manipulation, slicing
  - Strings manipulations
  - Introducing object oriented programming / S4 objects
  - Visualization tools
  - Bonus create FeaturePlot from Seurat in base ggplot
  - Bonus: run RSEM on Dana’s bam files if you are bored

#### Lecture 2-3 - Expression QC, normalisation and gene-level batch correction (Orr)
  - What CellRanger does for quality filtering
  - PBMC data
  - Normalisation methods https://www.nature.com/articles/nmeth.4292
  - Doublets, empty droplets, CellBender
  - Barcode swapping
  - Regression with technical covariates
  - What about imputation?

#### Lab 4 – Data wrangling for scRNAseq data (Dana)
  - Data structures and file formats for single-cell data
  - Quality control of cells and genes (doublets, ambient, empty drops)
  - Data exploration: violin plots…
  - Introducing Seurat object
    - Genes
    - House keeping genes
    - Mitochondrial genes (never used these ones)
    - Filter - Do we remove both cells and genes here?
  - Normalize (introduce more options, other than log transform?)
  - Find variable genes (Is it a first reduction? Why the binning?)
  - Scaling
  - Regression
  - Heatmap of desired genes?
  - Signatures?
  - Bonus - imputation (magic? One of the two Gocken recommended?)


#### Flash talks (1.5 hr, break into 2 groups of 13)

1 slide advertising or summarizing the poster. So you can introduce yourselves and we can get to know each other. No questions, 2 minutes. Two sessions, 15 people each.


### Wednesday – Classes from 08:00 to 16:00

#### Lecture 4 (may start late Tuesday) - Identifying cell populations (Kirk)

  - Feature selection
  - Dimensionality reduction
  - Clustering and assigning identity (Louvain, NMF, topic models, variational autoencoder)
  - Differential expression tests

#### Lab 5 – Feature selection & Clustering analysis (Kirk)
  - Parameters and clustering
  - Comparison of feature selection methods

#### Lecture 5 - Batch effects correction (Orr)
  - Batch correction methods (regress out batch, scaling within batch, Seurat v3, MNN, Liger, Harmony, scvi, scgen)
  - Evaluation methods for batch correction (ARI, average silhouette width, kBET…)

#### Lab 6 - Correcting batch effects (Orr)
  - Comparison of batch correction methods, Seurat pancreas
  - Use Seurat Wrappers?

### Thursday – Classes from 08:00 to 16:00
**Deciding on discussion topics for next day based on shared google doc.**

#### Lecture 6 - Advanced topics (Kirk)
  - Waddington Landscape
  - Pseudotime inference
  - Differential expression through pseudotime
  - Deep learning or spatial data depending on questionnaire (20ish min, autoencoder as nonlinear dimension reduction, scvi, what questions to ask to evaluate whether a more advanced model helps, how to decide it's safe to use a method, tradeoffs between method complexity and interpretability)


#### Lab 7 - Functional and Pseudotime analysis (Orr)
  - Popular tools and packages for functional analysis (https://github.com/dynverse/dynmethods#list-of-included-methods)
  - Review concepts from papers
  - Comparison of pseudotime methods

#### Lecture 7 - Single-cell multi-omic technologies (Dana)
  - Introduction to other omic data types
  - Integrating scRNA-seq with other single-cell modalities (CITE, Perturb, ATAC, methylation…)

#### Lab 8 - Analysis of CITE-seq, scATAC-seq (Orr)
  - https://github.com/Hoohm/CITE-seq-Count
  - https://cite-seq.com/eccite-seq/
  - https://support.10xgenomics.com/single-cell-vdj/index/doc/technical-note-assay-scheme-and-configuration-of-chromium-single-cell-vdj-libraries
  - https://satijalab.org/seurat/multimodal_vignette.html
  - https://www.bioconductor.org/packages/devel/bioc/vignettes/cicero/inst/doc/website.html


### Friday – Classes from 08:00 to 16:00
Small group discussion on selected topics through hangouts.

#### Lab 10 - small dataset for analysis and office hours focused on select topics (Dana)
For project on last day (plan for whole day), Dana will prepare datasets for 3 or more ish mut glioma tumors that they will download beforehand. The datasets may need to be subsampled to save time. Can do pseudotime, can run scvi, nmf. Groups of 3 students. 


#### Group presentations
Review, Questions and Answers





```{r include=FALSE}
# automatically create a bib database for R packages
knitr::write_bib(c(
  .packages(), 'bookdown', 'knitr', 'rmarkdown'
), 'packages.bib')
```

<!--chapter:end:index.Rmd-->

---
title: "01-Introduction-lecture"
author: "Orr Ashenberg"
date: "3/23/2020"
output: html_document
---

# Lecture Template

To get the link to paste into the iframe open up slides in Google Slides and go to,
  - File -> Publish to the Web
  - Select the Embed tab section
  - Hit Publish
  - Copy and Paste the link in between the quotations in the <iframe src=""  part


<iframe src="https://docs.google.com/presentation/d/e/2PACX-1vQFLazNqxsHpi8PCLHu-7ZmSGzn4Q4HpOoSJGGEcwzL7hcuDJGJ-kVJ1Qtf5gNiGSTF8zv7K3f-i-xo/embed?start=false&loop=false&delayms=3000" frameborder="0" width="1280" height="749" allowfullscreen="true" mozallowfullscreen="true" webkitallowfullscreen="true"></iframe>

<!--chapter:end:01-Introduction-lecture.Rmd-->

---
title: "01-template-lab"
author: "Kirk Gosik"
date: "2/25/2020"
output: html_document
---

# Lab Template

```{r}
knitr::opts_chunk$set(echo = TRUE)
library(Seurat)
```

## Template Example

```{r read_data}
counts <- Read10X(data.dir = "data/filtered_gene_bc_matrices/GRCh38")
```

## Make Object

```{r pressure, echo=FALSE}
obj <- CreateSeuratObject(counts)
```

## Process Object

<!--chapter:end:01-template-lab.Rmd-->

---
title: "01-template-lecture"
author: "Kirk Gosik"
date: "2/25/2020"
output: html_document
---

# Lecture Template

To get the link to paste into the iframe open up slides in Google Slides and go to,
  - File -> Publish to the Web
  - Select the Embed tab section
  - Hit Publish
  - Copy and Paste the link in between the quotations in the <iframe src=""  part


<iframe src="https://docs.google.com/a/broadinstitute.org/presentation/......" frameborder="0" width="760" height="569" allowfullscreen="true" mozallowfullscreen="true" webkitallowfullscreen="true"></iframe>

<!--chapter:end:01-template-lecture.Rmd-->

---
title: "03-Understanding-Sequencing-Raw-Data"
output: html_document
---

# Understanding Sequencing Raw Data

## Class Environment

### Getting into AWS Instance

There is a nice breakdown from another Physalia course on instructions for different operating systems and accessing AWS.  It is called [Connection to the Amazon EC2 service](https://gitlab.com/bfosso/metabarcoding_physalia/-/blob/master/unix_short_tutorial/how_to_connect.md).  This will help with connecting to the AWS instance to run docker.  

```{bash ssh_aws, eval = FALSE}
## Example
ssh -i berlin.pem ubuntu@<PUBLIC IP ADDRESS> (e.g.34.219.254.245)

## Actual Command Example
ssh -i berlin.pem ubuntu@34.213.180.241
```


## Shell and Unix commands

<iframe src="https://docs.google.com/presentation/d/e/2PACX-1vSavPAPY3WXe2EiqaRXJmAajFTeGcRbD1qJq4Sp4YlcF1_m7I1X89D1uC-8jDDGinqFLNO2oJNALbNx/embed?start=false&loop=false&delayms=3000" frameborder="0" width="760" height="569" allowfullscreen="true" mozallowfullscreen="true" webkitallowfullscreen="true"></iframe>


### Common Linux Commands

#### Lab 1a

  - check the your present directory
```{bash lab1a_pwd, eval = FALSE}
pwd
```

  - check history
```{bash lab1a_hist, eval = FALSE}
history
```

  - pipe history to grep to search for the cd command
```{bash, eval = FALSE}
history | grep cd
```

  - put history into a history.txt file
```{bash lab1a_hist_txt, eval = FALSE}
history > history.txt
```

  - make a directory called data
```{bash lab1a_mkdir, eval = FALSE}
mkdir data
```

  - change into data directory
```{bash lab1a_cd, eval = FALSE}
cd data
```

  - move history.txt file into data directory
```{bash lab1a_mv, eval = FALSE}
mv ../history.txt ./
```

  - check manual page of wget command
```{bash lab1a_man, eval = FALSE}
man wget
```

  - redirect wget maunual page output into a file called wget.txt
```{bash lab1a_man_wget, eval = FALSE}
man wget > wget.txt
```

  - return the lines that contain output in the wget.txt file
```{bash lab1a_cat, eval = FALSE}
cat wget.txt | grep output
```

```{bash lab1a_grep, eval = FALSE}
grep -i output wget.txt
```

  - Compress wget.txt file
```{bash lab1a_gzip, eval = FALSE}
gzip wget.txt
```

  - View Compressed file
```{bash lab1a_catgz, eval = FALSE}
cat wget.txt.qz
```
  

```{bash lab1a_zcat, eval = FALSE}
zcat wget.txt.qz
```


```{bash lab1a_less, eval = FALSE}
zcat wget.txt.qz | less
```


#### Git Commands 

Git is a distributed version-control system for tracking changes in source code during software development. It is designed for coordinating work among programmers, but it can be used to track changes in any set of files. Its goals include speed, data integrity, and support for distributed, non-linear workflows.

Go to your user directory and run the following command from git.  This will create a directory of all the course material inside your user directory.  After it is done cloning change directory into the 2020_scWorkshop directory where the course material is.  The commands are below.  

```{bash clone_repo, eval = FALSE}
## clone repository
git clone https://github.com/broadinstitute/2020_scWorkshop.git
cd 2020_scWorkshop
```


## File formats

  - bcl
  - fastq
  - bam
  - mtx, tsv
  - hdf5 (.h5, .h5ad)
  
### View FASTQ Files

#### Viewing entire file
```{bash cat_fastq, eval = FALSE}
cat data/example_fastq.fastq
```

#### Viewing first 10 lines

```{bash head_fastq, eval = FALSE}
head data/example_fastq.fastq
```

#### Stream Viewing with less command

```{bash less_fastq, eval = FALSE}
less data/example_fastq.fastq
```


### View BAM Files

#### Viewing first 10 lines

```{bash sam_view_less, eval = FALSE}
samtools view data/ex1.bam | head
```

#### Stream Viewing with less command

```{bash sam_view_head, eval = FALSE}
samtools view data/ex1.bam | less
```


## Public data repositories

### Cellranger/10x

#### Lab 1b

10x PBMC data are hosted in https://s3-us-west-2.amazonaws.com/10x.files/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz

  - get 10x PBMC data
  - unzip data
  - explore directory
  - explore files


```{bash wget_pbmc, eval = FALSE}

wget https://s3-us-west-2.amazonaws.com/10x.files/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz -O data/pbmc3k_filtered_gene_bc_matrices.tar.gz
cd data; tar -xzf pbmc3k_filtered_gene_bc_matrices.tar.gz
cd ..
```


### GEO

 - https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE81905


#### Lab 1c

**Get GEO Data**
  - ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE81nnn/GSE81905/matrix/GSE81905-GPL19057_series_matrix.txt.gz
  - ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE81nnn/GSE81905/matrix/GSE81905-GPL17021_series_matrix.txt.gz

 - go into that directory
 - get files and place them in the directory
 - View files (try keeping in compressed format and view that way)
 
```{bash wget_geo, eval = FALSE}
wget ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE81nnn/GSE81905/matrix/GSE81905-GPL19057_series_matrix.txt.gz
cd data; tar -xzf GSE81905-GPL19057_series_matrix.txt.gz
cd ..
```

### Single Cell Portal

  - https://portals.broadinstitute.org/single_cell
  - Study: Salk Institute - Single-cell Methylome Sequencing Identifies Distinct Neuronal Populations in Mouse Frontal Cortex

#### Lab 1d

  - Get R2 fastq file from the Salk Institute study 
  - Look at files
  
  
#### Lab 1e

  - Get Docker on your local computer for you to have 
  - Explore Single Cell Portal
  - Explore GEO
  
  
  
## Docker Commands

Docker provides a consistent compute enviornment to ensure all software that you need is on the machine and able to be used. It will give you the version you need and help reduce software conflicts that may arise.

  - make sure you are in the directory from the cloned repository directory
  - run following command to start docker script
  
```{bash run_script_bash, eval = FALSE}
# chmod 600 ./docker/run_docker_bash.sh
./docker/run_docker_bash.sh
```

The full command inside the script is below.  There is also an explaination of each part for your reference.  

```{bash full_bash_cmd, eval = FALSE}
## if you are the super user on your computer
docker run --rm -it -v $PWD:/home/rstudio kdgosik/2020scworkshop bash

## if you need to access permission to run the command
sudo docker run --rm -it -v $PWD:/home/rstudio kdgosik/2020scworkshop bash

```


**Explaination of commands**
```{bash explain_cmd, eval = FALSE}
  - docker: command to run docker
  - run: asking docker to run a container
  - --rm: flag to remove the container when you exit from it
      - nothing will be saved from your session to access again later
      - this flag can be removed to keep container
  - -it: flag to run the container interactively
    - this will keep all session output displaying on the terminal
    - to stop container go to terminal and press Crtl+c
    -v $PWD:/home/rstudio: map your home directory to a directory inside docker container called home
  - kdgosik/2020scworkshop: the image to run.  It will be the image into a container if not already built on your computer
    - [image link](https://hub.docker.com/r/kdgosik/2020scworkshop)
  - bash: The entry point into the container.  Start on the bash command line
```


<!--chapter:end:03-Understanding-Sequencing-Raw-Data.Rmd-->

---
title: "02-Transcriptome-quantification"
output: html_document
---

# Transcriptome Quantification

## Google Slides

<iframe src="https://docs.google.com/presentation/d/e/2PACX-1vSswp7LJIN-9W-tz4eh5WPRKCGLBecRVwl5vtXvIDG3ywSR-_8S4sYZgOCnu5AD_A/embed?start=false&loop=false&delayms=3000" frameborder="0" width="760" height="560" allowfullscreen="true" mozallowfullscreen="true" webkitallowfullscreen="true"></iframe>

<!--chapter:end:04-Transcriptome-Quantification.Rmd-->

---
title: "04-Processing-scRNAseq-Data"
output: html_document
---

# Processing scRNAseq Data

## Goal

- To give you experience with examining and aligning fastq files

## Further reading

This lab is based on a lab given in:
http://hemberg-lab.github.io/scRNA.seq.course/processing-raw-scrna-seq-data.html
For more exercises and ideas please visit their web-site!

## Download data

Please downlaod the 6 files from the dropbox folder: https://www.dropbox.com/sh/98573jes82w0fi7/AAB7Yhwe05MCZTnTZmppxZuta?dl=0 into the data folder of your copy of github:
```{r copy files, eval = FALSE}
cd 2020_scWorkshop/data
mkdir lab2data
```
and copy the files into it
## FastQC

Once you’ve obtained your single-cell RNA-seq data, the first thing you need to do with it is check the quality of the reads you have sequenced. For this task, today we will be using a tool called FastQC. FastQC is a quality control tool for sequencing data, which can be used for both bulk and single-cell RNA-seq data. FastQC takes sequencing data as input and returns a report on read quality. Copy and paste this link into your browser to visit the FastQC website:

https://www.bioinformatics.babraham.ac.uk/projects/fastqc/

This website contains links to download and install FastQC and documentation on the reports produced. Fortunately we have already installed FastQC for you today, so instead we will take a look at the documentation. Scroll down the webpage to ‘Example Reports’ and click ‘Good Illumina Data’. This gives an example of what an ideal report should look like for high quality Illumina reads data.

Now let’s make a FastQC report ourselves.

Today we will be performing our analysis using a single cell from an mESC dataset produced by (Kolodziejczyk et al. 2015), which you downloaded froom the github link. The cells were sequenced using the SMART-seq2 library preparation protocol and the reads are paired end.

First, let's open the docker in a bash mode. open a terminal, cd to the docer folder (the folder you downloaded from github) and run this command:
```{r run docker, eval = FALSE}
docker run --rm -ti -v $PWD:/home/rstudio -e DISABLE_AUTH=true kdgosik/2020scworkshop bash
```

navigate to your data folder:
```{r navigate to files, eval = FALSE}
cd home/rstudio/lab2data
ls
```

You should see the files that you downloaded from the dropbox link.

Now let’s look at the files:
```{r examine, eval = FALSE}
less Teichmann_2i_2_2_1.fastq
less Teichmann_2i_2_2_2.fastq
```

We run fastqc from /usr/local/src/FastQC. You may need to give yourself permissions to run the file (hint: chmod)

Task 1: run fastqc to view the quality of the reads
```{r explore fastqc, eval = FALSE }
chmod 755 /usr/local/src/FastQC/fastqc
/usr/local/src/FastQC/fastqc -h
```

This command will tell you what options are available to pass to FastQC. Let us direct our output to our personal directories (under the folder results). Feel free to ask for help if you get stuck! If you are successful, you should generate a .zip and a .html file for both the forwards and the reverse reads files. Once you have been successful, feel free to have a go at the next section.

```{r run fastqc - this is the answer, eval = FALSE }
/usr/local/src/FastQC/fastqc -o <output_folder> Teichmann_2i_2_2_1.fastq Teichmann_2i_2_2_2.fastq
```

Once the command has finished executing, you should have a total of four files - one zip file for each of the paired end reads, and one html file for each of the paired end reads. The report is in the html file. 

for those working in AWS, if you want to view the file you will need to download it to your computer. The scp command is:
```{r scp, eval = FALSE }
scp -r -i <your pem file> <username>@ec2-34-213-180-241.us-west-2.compute.amazonaws.com:~/<file to copy> <destination in your computer>
```

Once the file is on you computer, click on it. Your FastQC report should open. Have a look through the file. Remember to look at both the forwards and the reverse end read reports! How good quality are the reads? Is there anything we should be concerned about? 


### Fastq file format

FastQ is the most raw form of scRNASeq data you will encounter. All scRNASeq protocols are sequenced with paired-end sequencing. Barcode sequences may occur in one or both reads depending on the protocol employed. However, protocols using unique molecular identifiers (UMIs) will generally contain one read with the cell and UMI barcodes plus adapters but without any transcript sequence. Thus reads will be mapped as if they are single-end sequenced despite actually being paired end.

FastQ files have the format:
```{r file format, eval = FALSE}
>ReadID
READ SEQUENCE
+
SEQUENCING QUALITY SCORES
```

## Align the reads

### STAR align

Now we have established that our reads are of good quality, we would like to map them to a reference genome. This process is known as alignment. Some form of alignment is generally required if we want to quantify gene expression or find genes which are differentially expressed between samples.

Many tools have been developed for read alignment, but today we will focus on STAR. For each read in our reads data, STAR tries to find the longest possible sequence which matches one or more sequences in the reference genome. Because STAR is able to recognize splicing events in this way, it is described as a ‘splice aware’ aligner.

Usually STAR aligns reads to a reference genome, potentially allowing it to detect novel splicing events or chromosomal rearrangements. However, one issue with STAR is that it needs a lot of RAM, especially if your reference genome is large (eg. mouse and human). To speed up our analysis today, we will use STAR to align reads from to a reference transcriptome of 2000 transcripts. Note that this is NOT normal or recommended practice, we only do it here for reasons of time. We recommend that normally you should align to a reference genome.

Two steps are required to perform STAR alignment. In the first step, the user provides STAR with reference genome sequences (FASTA) and annotations (GTF), which STAR uses to create a genome index. In the second step, STAR maps the user’s reads data to the genome index.

Let’s create the index now. Remember, for reasons of time we are aligning to a transcriptome rather than a genome today, meaning we only need to provide STAR with the sequences of the transcripts we will be aligning reads to. You can obtain transcriptomes for many model organisms from Ensembl (https://www.ensembl.org/info/data/ftp/index.html).


Task 2: Create a genome index

First create the output folder for the index in your personal folder under results (recommended /home/rstudio/lab2data/STAR/indices).

We run STAR from:
```{r STAR location, eval = FALSE}
/usr/local/src/STAR/bin/Linux_x86_64
```

using the command:
```{r create STAR index, eval = FALSE}
/usr/local/src/STAR/bin/Linux_x86_64/STAR --runThreadN 4 --runMode genomeGenerate --genomeDir <output STAR indices folder> --genomeFastaFiles /home/rstudio/lab2data/2000_reference.transcripts.fa
```

Now that we have created the index, we can perform the mapping step.

Task 4: Try to work out what command you should use to map our fastq files to the index you created. Use the STAR manual to help you. Once you think you know the answer use ./STAR command to align the fastq files to a BAM file.
You can either create a SAM file and convert it to BAM using samtools, or use STAR to directly output a BAM file (--outSAMtype BAM Unsorted)

```{r align read using STAR, eval = FALSE}
/usr/local/src/STAR/bin/Linux_x86_64/STAR --runThreadN 4 --genomeDir <genome_reference> --readFilesIn /home/rstudio/lab2data/Teichmann_2i_2_2_1.fastq /home/rstudio/lab2data/Teichmann_2i_2_2_2.fastq --outFileNamePrefix <output_folder> --outSAMtype BAM Unsorted
```

The alignment may take awhile, if you wish to you can complete tasks 7-10 in the meanwhile.


### Bam file format

BAM file format stores mapped reads in a standard and efficient manner. The human-readable version is called a SAM file, while the BAM file is the highly compressed version. BAM/SAM files contain a header which typically includes
information on the sample preparation, sequencing and mapping; and a tab-separated row for each individual alignment of each read.

Alignment rows employ a standard format with the following columns:

QNAME : read name (generally will include UMI barcode if applicable)

FLAG : number tag indicating the “type” of alignment, link to explanation of all possible “types”

RNAME : reference sequence name (i.e. chromosome read is mapped to).

POS : leftmost mapping position

MAPQ : Mapping quality

CIGAR : string indicating the matching/mismatching parts of the read (may include soft-clipping).

RNEXT : reference name of the mate/next read

PNEXT : POS for mate/next read

TLEN : Template length (length of reference region the read is mapped to)

SEQ : read sequence

QUAL : read quality


BAM/SAM files can be converted to the other format using ‘samtools’:
```{r samtools bam/sam, eval = FALSE}
samtools view -S -b file.sam > file.bam
samtools view -h file.bam > file.sam
```

Some sequencing facilities will automatically map your reads to the a standard genome and deliver either BAM or CRAM formatted files. Generally they will not have included ERCC sequences in the genome thus no ERCC reads will be mapped in the BAM/CRAM file. To quantify ERCCs (or any other genetic alterations) or if you just want to use a different alignment algorithm than whatever is in the generic pipeline (often outdated), then you will need to convert the BAM/CRAM files back to FastQs:

BAM files can be converted to FastQ using bedtools. To ensure a single copy for multi-mapping reads first sort by read name and remove secondary alignments using samtools. Picard also contains a method for converting BAM to FastQ files.

Bonus: 

To make our aligned BAM file easy to navigate (needed for IGViewer) we will sort and index it using samtools.
Sam tools can be run from everywhere (no need to go to a special directory!) using the command:
```{r samtools command, eval = FALSE}
samtools
```

Let us start by sorting the BAM file:
```{r sort, eval = FALSE}
samtools sort Aligned.out.bam -o Aligned.out.sorted.bam
```

Task 6: can you index the file? hint: try looking at samtools -h

```{r index, eval = FALSE}
samtools index Aligned.out.sorted.bam
```

Once you sorted and indexed the files you should have a BAM and a bai files. The BAM file is the aligned reads, and the bai is an index file. To view them in IGViewer (IGV) first copy them into your computer. Go ahead and copy the fa file as well, we will need a reference genome file. 

## Visualization

To view the file we will use the IGV you installed on your personal computer. 
Open IGV: the default genomes are human HG19 and HG38. Through the class we will be using the PBMC dataset. You have the BAM file in your data folder. Go ahead and transfer it to your computer and upload it to IGV with hg38 as reference genome.

```{r IGV viewing pbmc bam, eval = FALSE}

Task 7: Browse to MS4A1, this is a blood cell marker. Can you see the exons and the introns? Where are most of the aligned reads?

Task 8: Search in IGV or online - can you present splice junctions? (right click -> “Show splice junction track”)

Task 9: Try further tasks that interest you in IGV. For example, can you detect reads that are within one exon and reads that start in one exon and continue in the next? Can you copy the sequence of exon2 in MS4A1?

Task 10: What would have happened if you chose the wrong reference genome, such as hg19?
```


Bonus 2 (IGV sometimes has difficulties loading small fa files. So if this becomes difficult - don't worry! It's not your alignment):

The default genomes are human HG19 and HG38. However you can also upload your reference genome of choice. As we created our own fasta file we can now upload it as a reference genome. 

```{r IGV viewing your newlt created bam file, eval = FALSE}
Task 11: Load new genome: go to “Genomes”->”Load from file” and load the file 2000_reference.transcripts.fa

Task 12: Now load your reads: go to “File”->”Load from file” and load your BAM file. Notice that IGV needs a BAM and a bai saved in the same location. IGV uses the bai to navigate through the BAM file. 

Task 13: Some of the reads have a nucleotide substitution in position 993 - what is the reference nucleotide? What is the substitution?
```






<!--chapter:end:05-Processing-scRNA-Data.Rmd-->

---
title: "06-Introduction-R"
output: html_document
---

```{r settings, echo=FALSE}
library(knitr)
opts_chunk$set(out.width='90%', fig.align = 'center')

library(Seurat)
library(Matrix)
library(ggplot2)
```


# Introduction R/Bioconductor

```{bash, eval = FALSE, echo = FALSE}
# mkdir data
wget https://s3-us-west-2.amazonaws.com/10x.files/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz -O data/pbmc3k_filtered_gene_bc_matrices.tar.gz
cd data; tar -xzf pbmc3k_filtered_gene_bc_matrices.tar.gz
cd ..
```

## Start Environment

### Local Command

```{bash docker_script, eval = FALSE}
# chmod 600 ./docker/run_docker.sh
./docker/run_docker.sh
```


**full command**
```{bash start_env_local, eval = FALSE}
## maybe take away the --rm so they can save the container for later
## run from your home directory
cd 2020scworkshop

## example for user17
docker run --rm -it 
  -e DISABLE_AUTH=true \
  -v $PWD:/home/rstudio \
  -p 9017:8787 kdgosik/2020scworkshop
```


### AWS Command

```{bash start_env_aws, eval = FALSE}
## maybe take away the --rm so they can save the container for later
## run from your home directory
cd 2020scworkshop

## example for user17
sudo docker run --rm -it 
  -e PASSWORD=train \
  -v $PWD:/home/rstudio \
  -p 9017:8787 kdgosik/2020scworkshop
```

**Explaination of commands**
```{bash docker_cmds, eval = FALSE}
  - docker: command to run docker
  - run: asking docker to run a container
  - --rm: flag to remove the container when you exit from it
      - nothing will be saved from your session to access again later
      - this flag can be removed to keep container
  - -it: flag to run the container interactively
  - - this will keep all session output displaying on the terminal
  - - to stop container go to terminal and press Crtl+c
  - -v $PWD:/home/rstudio: map your home directory to a directory inside docker container called home/rstudio
  - -p 9017:8787: map docker container port of 8787(rstudio port default) to your computer port 9017
  - kdgosik/2020scworkshop: the image to run.  It will be the image into a container if not already built on your computer
    - [image link](https://hub.docker.com/r/kdgosik/2020scworkshop)
```

  - [localhost:9099](https://localhost:9017)  or on AWS
  - [<AWS PUBLIC IP ADDRESS>:9017](https://:9017)

  - ec2-<AWS PUBLIC IP ADDRESS>.us-west-2.compute.amazonaws.com:$PORT_NUMBER
  - ec2-54-202-32-102.us-west-2.compute.amazonaws.com:9017

  - R/Rstudio parts
  - Data Types and classes
  - Packages and where to get them
  - S3 vs S4
  - Visualizations and ggplot
  
  
  - Installing packages
  - Data-types
  - Data manipulation, slicing
  - Strings manipulations
  - Introducing object oriented programming / S4 objects
  - Visualization tools
  - Bonus create FeaturePlot from Seurat in base ggplot
  - Bonus: run RSEM on Dana’s bam files if you are bored 
  

## Installing packages

### CRAN

The Comprehensive R Archive Network [CRAN](https://cran.r-project.org/) is the biggest archive of R packages. There are few requirements for uploading packages besides building and installing succesfully, hence documentation and support is often minimal and figuring how to use these packages can be a challenge it itself. CRAN is the default repository R will search to find packages to install:

```{r cran_inst, eval=FALSE}
install.packages("devtools")

# or multiple packages

install.packages(c("ggplot2", "stringr"))
```

### Github

[Github](https://github.com/) isn't specific to R, any code of any type in any state can be uploaded. There is no guarantee a package uploaded to github will even install, nevermind do what it claims to do. R packages can be downloaded and installed directly from github using the "devtools" package installed above.

```{r devtools_inst, eval=FALSE}
## username/repository
devtools::install_github("satijalab/seurat") # latest stable version of Seurat package
```

Github is also a version control system which stores multiple versions of any package. By default the most recent "master" version of the package is installed. If you want an older version or the development branch this can be specified using the "ref" parameter:

```{r devtools2_inst, eval=FALSE}
# different branch
devtools::install_github("satijalab/seurat", ref="release3.0")

# previous commit
## Merge branch 'develop' into feat/MultiModal
##  - Shiwei Zheng committed on Jul 2, 2018
devtools::install_github("tallulandrews/M3Drop", ref="551014f488770627ab154a62e59d49df5df98a3f")
```
Note: make sure you re-install the M3Drop master branch for later in the course.

### Bioconductor

Bioconductor is a repository of R-packages specifically for biological analyses. It has the strictest requirements for submission, including installation on every platform and full documentation with a tutorial (called a vignette) explaining how the package should be used. Bioconductor also encourages utilization of standard data structures/classes and coding style/naming conventions, so that, in theory, packages and analyses can be combined into large pipelines or workflows. 


Bioconductor also requires creators to support their packages and has a regular 6-month release schedule. Make sure you are using the most recent release of bioconductor before trying to install packages for the course.

```{r bioc_inst, eval = FALSE}
## >= R 3.5.0
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("Rsamtools", ask = FALSE)
```

### Source

The final way to install packages is directly from source. In this case you have to download a fully built source code file, usually packagename.tar.gz, or clone the github repository and rebuild the package yourself. Generally this will only be done if you want to edit a package yourself, or if for some reason the former methods have failed.  You can also get previous packages that aren't supported any more on the [CRAN package archive](https://cran.r-project.org/)

```{r src_isnt, eval=FALSE}
## Get an old package and install from source
install.packages("GenABEL_1.8-0.tar.gz", type="source")
```

## Installation instructions:

All the packages necessary for this course are available [here](https://github.com/broadinstitute/2020_scWorkshop). A list of the packages will be on the README.md for the repository.  A script is also available inside the docker/install.R file.  



### Classes/Types


R is a high level language so the underlying data-type is generally not important. The exception if you are accessing R data directly using another language such as C, but that is beyond the scope of this course. Instead we will consider the basic data classes: numeric, integer, logical, and character, and the higher level data class called "factor". You can check what class your data is using the "class()" function.

#### Integer

```{r r_ints}
x <- 4 ## assign value of 4 to x

class(x) ## check class of x

is.integer(x) ## check if x is an integer

is.numeric(x) ## check if x is numeric

x <- as.numeric(x) ## assign y to be an numeric

is.numeric(x) ## check if the assignment worked

class(x) ## check if the assignment worked

x ## check value of x
```


#### Numeric

```{r num_types}
## assign value of 1.414 to y
y <- 1.414
## check class of y
class(y)
## check if y is numeric
is.numeric(y)
## check if y is an integer
is.integer(y)
## assign y to be an integer
y <- as.integer(y)
## check if the assignment worked
is.integer(y)
## check value of y
y
```


#### Logical/ Boolean

The `logical` class stores boolean truth values, i.e. TRUE and FALSE. It is used for storing the results of logical operations and conditional statements will be coerced to this class. Most other data-types can be coerced to boolean without triggering (or "throwing") error messages, which may cause unexpected behaviour.

```{r logi_types}
z <- TRUE ## assign value of TRUE to z
class(z) ## check class of z
is.logical(z) ## check if z is of logical type
```


### Data structures
  - Homogeneous
    - 1D: atomic vector
    - 2D: matrix
    - nD: array
  - Heterogeneous
    - 1D: list
    - 2D: data.frame


#### Character Vectors

```{r char_types}
## assign a character vector with c() operator
character_vector <- c("A", "C", "T", "G", "C", "T", "G", "C", "G", "A", "T", "G", "A", "C", "G", "A", "C")

## check class
class(character_vector)


## access the 3rd element with [] operator
## *note*: R is index starts at 1 (other programming languages start at 0)
character_vector[3]


## access 3rd through 6th elemenet
character_vector[3:6]

## access the elements 1,4,7,10 with c()
character_vector[c(1, 4, 7, 10)]

## access all the A's
character_vector[grep("A", character_vector)]
```


#### Numeric Vector

The "numeric" class is the default class for storing any numeric data - integers, decimal numbers, numbers in scientific notation, etc...

```{r num_vecs}
## assign a character vector with c() operator
numeric_vector <- c(1, 5, 21, 17, 98, 35, 11, 13)

## check class
class(numeric_vector)

## access the 5th element with [] operator
numeric_vector[5]

## access 2nd through 4th elemenet
numeric_vector[2:4]
```

```{r num_vec2}
## backticks ` ` allow you to give names with non-typical characters
`numeric?_vecotr` <- c("A", 1, 5, 21, 17, 98, 35, 11, 13)

## check vector
`numeric?_vecotr`

## check class (Notice the quotation marks on the numbers!)
class(`numeric?_vecotr`)
```

#### Factor Vector

String/Character data is very memory inefficient to store, each letter generally requires the same amount of memory as any integer. Thus when storing a vector of strings with repeated elements it is more efficient assign each element to an integer and store the vector as integers and an additional string-to-integer association table. Thus, by default R will read in text columns of a data table as factors. 

```{r factor_type}
factor_vector <- factor(numeric_vector)
factor_vector
```


#### Named Vector

```{r name_vec}
names(numeric_vector) <- paste0("Patient", 1 : length(numeric_vector))
numeric_vector
```



#### List

```{r list_type}
## change the c() operator to list() operator
new_list <- list("A", 1, 5, 21, 17, 98, 35, 11, 13)
new_list
```

```{r list_elem}
## get 2nd element of list
new_list[[2]]
```


```{r name_list}
names(new_list) <- paste0("Patient", 1 : length(new_list))
new_list
```

```{r name_2list}
## get 2nd element of list
new_list[[2]]
```

  - 2D

#### matrix

**Create Matrix**
```{r mat_type}
## create numeric matrix
numeric_matrix <- matrix(sample(1:10, 100, replace = TRUE), nrow = 10, ncol = 10)
class(numeric_matrix) ## check class
```


**Check Structure**
```{r strct}
str(numeric_matrix)
```

**Get 3rd Row**
```{r num_mat}
## get 3rd row
numeric_matrix[3, ]
```


**Get 4th Column**
```{r num_mat_col}
## get 4th colum
numeric_matrix[, 4]
```



####  data.frame

**Get data.frame**
```{r df_type}
## built in R data.frame iris
head(iris)
```

**Check Class**
```{r df_class}
class(iris)
```

**Check Structure**
```{r df_struct}
str(iris)
```

**Get 3rd Row**
```{r df_col_sub}
## get 3rd row
iris[3,]
```


**Get 4th Column**
```{r df_row_sub}
## get 4th colum
iris[,4]
```

**Get 3rd Row**
```{r num_mat_row}
## get 3rd row
numeric_matrix[3, ]
```


**Get Species Variable**
```{r df_var}
## get variable
iris$Species
```



### Detour to S3/S4

  - S3 most of R uses
  - Bioconductor requires R packages to be written as S4 objects
  - OO field [guide](http://adv-r.had.co.nz/OO-essentials.html)
  - Closer to a typical programming language
    - Classes/Methods and Generics
    - Lots of Generics implemented for Bioinformatics!
  
Different way to access values.  Need to use the @ symbol instead of $

  - (@ is equivalent to $, and slot() to [[.)

```{r obj_type, eval= FALSE}
## example
object@
```


#### Sparse Matrix

Triplet format for storing a matrix
row, column, value
i, p, x


Different from base R.  Uses the S4 methods that Bioconductor uses. 
```{r sparse_mat, eval= FALSE}
sparse_matrix <- pbmc_small@data[1:10, ]
class(sparse_matrix)
```


**ith row - 1**
```{r sparse_mat_idx, eval= FALSE}
sparse_matrix@i
```

**pth column - 1**
```{r sp_mat_col, eval= FALSE}
sparse_matrix@p
```


**value**
```{r sp_x, eval= FALSE}
sparse_matrix@x
```

**Get First Value**
```{r sp_mat_sub, eval= FALSE}
sparse_matrix[2,1]
```
**dense matrix**
```{r dense_mat, eval= FALSE}
dense_matrix <- as.matrix(sparse_matrix)
class(dense_matrix)
```

```{r struct_mat, eval= FALSE}
str(dense_matrix)
```

**Get First Value**
```{r dense_mat_sub, eval= FALSE}
dense_matrix[2,1]
```



#### Functions

```{r funcs, eval = FALSE}

create_function <- function(x, y) {
  
}
```



#### Reading Files

```{r read_file, eval = FALSE}

## read csv files
read.csv("PATH/TO/FILENAME.csv")

## read tsv files
read.delim("PATH/TO/FILENAME.tsv", sep = '\t')

```


## More information

You can get more information about any R commands relevant to these datatypes using by typing `?function` in an interactive session.

### Checking for help for any function!

  - start with a ? (this indicates you need the help menu)
  - then the function name to get help on

```{r ggplot_help}
library(ggplot2)

?ggplot  ## ggplot is a function, how do we use it?
```



    
## Grammer of Graphics (ggplot2)


### What is ggplot2?

ggplot2 is an R package designed by Hadley Wickham which facilitates data plotting. In this lab, we will touch briefly on some of the features of the package. If you would like to learn more about how to use ggplot2, we would recommend reading "ggplot2 Elegant graphics for data analysis", by Hadley Wickham or checking out his original paper on the [package](http://vita.had.co.nz/papers/layered-grammar.pdf)


  - Data: Always start with the data, identify the dimensions you want to visualize.
  - Aesthetics: Confirm the axes based on the data dimensions, positions of various data points in the plot. Also check if any form of encoding is needed including size, shape, color and so on which are useful for plotting multiple data dimensions.
  - Scale: Do we need to scale the potential values, use a specific scale to represent multiple values or a range?
  - Geometric objects: These are popularly known as ‘geoms’. This would cover the way we would depict the data points on the visualization. Should it be points, bars, lines and so on?
  - Statistics: Do we need to show some statistical measures in the visualization like measures of central tendency, spread, confidence intervals?
  - Facets: Do we need to create subplots based on specific data dimensions?
  - Coordinate system: What kind of a coordinate system should the visualization be based on — should it be cartesian or polar?

```{r img_ggplot, echo = FALSE}
knitr::include_graphics("images/components_grammar_graphics.png")
```


```{r ggplot_intro, eval = FALSE, echo = FALSE}
# If you don't have Tmisc installed, first install devtools, then install
# from github: install.packages('devtools')
# devtools::install_github('stephenturner/Tmisc')
library(Tmisc)
library(ggplot2)
data(quartet)

p <- ggplot(quartet, aes(x, y)) + geom_point()
p <- p + geom_smooth(method = lm, se = FALSE)
p <- p + facet_wrap(~set)
p
```



### Principles of ggplot2

* Your data must be a dataframe if you want to plot it using ggplot2. 
* Use the `aes` mapping function to specify how variables in the dataframe map to features on your plot
* Use geoms to specify how your data should be represented on your graph eg. as a scatterplot, a barplot, a boxplot etc.


  - Data: Always start with the data, identify the dimensions you want to visualize.
```{r ggplot_seurat}
library(Seurat)
library(ggplot2)

gbm <- pbmc_small@assays$RNA@data

gbm <- as.data.frame(as.matrix(t(gbm)))

new_plot <- ggplot(gbm)
```

  - Aesthetics: Confirm the axes based on the data dimensions, positions of various data points in the plot. Also check if any form of encoding is needed including size, shape, color and so on which are useful for plotting multiple data dimensions.

**1D Plots**
```{r 1d_plot}
new_plot_1dx <- ggplot(gbm, aes(x = MS4A1))
new_plot_1dx
```

  - Scale: Do we need to scale the potential values, use a specific scale to represent multiple values or a range?
  
  - Geometric objects: These are popularly known as ‘geoms’. This would cover the way we would depict the data points on the visualization. Should it be points, bars, lines and so on?
  
```{r 1dx_plot}
## ggplot(gbm, aes(x = MS4A1)) + geom_histogram()
## or
## new_plot_1dx <- new_plot_1dx + geom_histogram()  ## reassign
new_plot_1dx + geom_histogram()
```

#### Lab A

Use different geom_ to make a different plots
  - try _bar()
  - try _density()
  
```{r dense_plot}
new_plot_1dx + geom_density()
```


  - Statistics: Do we need to show some statistical measures in the visualization like measures of central tendency, spread, confidence intervals?
  
  
```{r hist_plot}
ggplot(gbm, aes(x = MS4A1)) + geom_histogram() + stat_bin(bins = 10)
```


**2D Plots**
 
```{r 2d_plot}
new_plot_2d <- ggplot(gbm, aes(x = MS4A1, y = CD79B))
```
  
```{r scatter_plot}
## scatter plot
new_plot_2d + geom_point()
```


#### Lab B

Use different geom_ to make a different plots
  - try _bar_abline()
  - try _bin2d()
  
```{r 2d_exer}
new_plot_2d
```

  - Adding Statisitics in 2D plots
    - Regression line (lm - linear model using OLS regression)
    
```{r 2d_exer_2}
ggplot(gbm, aes(MS4A1, CD79B)) + geom_point() + stat_smooth(method = "lm")
```

  - Adding Text Labels

```{r add_text}
## notice plus `+` at the end of each line, adding a new layer!
ggplot(gbm, aes(MS4A1, CD79B)) +            ## Data layer
  geom_point() +                            ## Geometry layer
  stat_smooth(method = "lm") +              ## 
  geom_text(aes(label = rownames(gbm)))
```


#### Lab C

Play arund with ggplot2.  See what geoms to add and layers to include.

```{r labc}
ggplot(gbm, aes(x = MS4A1, y = CD79B))
```



## Reference

  - [R for Data Science](https://r4ds.had.co.nz/)
  - [Advanced R](http://adv-r.had.co.nz/)
  - [Bioconductor Workflows](http://bioconductor.org/packages/release/BiocViews.html#___Workflow)
  - [Bioconductor Presentation](http://www.bioconductor.org/help/course-materials/2010/AdvancedR/S4InBioconductor.pdf)
  - [Original ggplot2 paper](http://vita.had.co.nz/papers/layered-grammar.pdf)
  - [ggplot2 reference](https://ggplot2.tidyverse.org/reference/index.html)
  - [ggplot2 cheatsheet](https://github.com/rstudio/cheatsheets/blob/master/data-visualization-2.1.pdf)
  - [blog post](https://towardsdatascience.com/a-comprehensive-guide-to-the-grammar-of-graphics-for-effective-visualization-of-multi-dimensional-1f92b4ed4149)
  - [Hemberg Lab](https://hemberg-lab.github.io/scRNA.seq.course/introduction-to-rbioconductor.html#an-introduction-to-ggplot2)
  

<!--chapter:end:06-Introduction-R-Bioconductor.Rmd-->

---
title: "09-Identifying-Cell-Populations"
output: html_document
---

# Identifying Cell Populations

## Google Slides

**NEEDS UPDATED**

<iframe src="https://docs.google.com/presentation/d/e/2PACX-1vTvbj3ozORDSetpF5Ke36FxTHh910f8yWcVgJ37ELhitlPHDup75aV94J5w0Orfq4BKvrxiCiD46AG_/embed?start=false&loop=false&delayms=3000" frameborder="0" width="760" height="560" allowfullscreen="true" mozallowfullscreen="true" webkitallowfullscreen="true"></iframe>


<!--chapter:end:09-Identifying-Cell-Populations.Rmd-->

---
title: "10-Cluster-Analysis"
output: html_document
---

# Feature Selection and Cluster Analysis

**NEEDS UPDATED!!**

## Abstract

Many methods have been used to determine differential gene expression from single-cell RNA (scRNA)-seq data. We evaluated 36 approaches using experimental and synthetic data and found considerable differences in the number and characteristics of the genes that are called differentially expressed. Prefiltering of lowly expressed genes has important effects, particularly for some of the methods developed for bulk RNA-seq data analysis. However, we found that bulk RNA-seq analysis methods do not generally perform worse than those developed specifically for scRNA-seq. We also present conquer, a repository of consistently processed, analysis-ready public scRNA-seq data sets that is aimed at simplifying method evaluation and reanalysis of published results. Each data set provides abundance estimates for both genes and transcripts, as well as quality control and exploratory analysis reports. [@soneson2018bias]



Cells are the basic building blocks of organisms and each cell is unique. Single-cell RNA sequencing has emerged as an indispensable tool to dissect the cellular heterogeneity and decompose tissues into cell types and/or cell states, which offers enormous potential for de novo discovery. Single-cell transcriptomic atlases provide unprecedented resolution to reveal complex cellular events and deepen our understanding of biological systems. In this review, we summarize and compare single-cell RNA sequencing technologies, that were developed since 2009, to facilitate a well-informed choice of method. The applications of these methods in different biological contexts are also discussed. We anticipate an ever-increasing role of single-cell RNA sequencing in biology with further improvement in providing spatial information and coupling to other cellular modalities. In the future, such biological findings will greatly benefit medical research. [@hedlund2018single]



## Seurat Tutorial Redo

For this tutorial, we will be analyzing the a dataset of Peripheral Blood Mononuclear Cells (PBMC) freely available from 10X Genomics. There are 2,700 single cells that were sequenced on the Illumina NextSeq 500. The raw data can be found [here](https://s3-us-west-2.amazonaws.com/10x.files/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz).


OR MAYBE USE THE vdj data?

```{bash get_data, eval = FALSE}
wget https://s3-us-west-2.amazonaws.com/10x.files/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz -O data/pbmc3k_filtered_gene_bc_matrices.tar.gz
cd data; tar -xzf pbmc3k_filtered_gene_bc_matrices.tar.gz
cd ..
```

**Task: Check the dirname to directory where you saved your data**
```{r read_data_clust, eval = FALSE}
knitr::opts_knit$set(progress=FALSE, verbose=FALSE)

library(Seurat)
library(dplyr)
library(ggplot2)
library(CountClust)

dirname <- "data/"
## dirname <- "/Users/kgosik/Documents/data/cellranger/"
counts_matrix_filename = paste0(dirname, "filtered_gene_bc_matrices/hg19/")
counts <- Read10X(data.dir = counts_matrix_filename)  # Seurat function to read in 10x count data

```

```{r create_seurat_clust, eval = FALSE} 
### seurat<-CreateSeuratObject(raw.data = counts, ? = 3, ? = 350, project = "10X_NSCLC")
seurat <- CreateSeuratObject(counts = counts, min.cells = 3, min.features = 350, project = "10X_PBMC")
```



### Preprocessing Steps

This was all covered in the last Lab!

```{r pt_mito, eval = FALSE}
# The number of genes and UMIs (nFeature_RNA nCount_RNA) are automatically calculated
# for every object by Seurat.  For non-UMI data, nCount_RNA represents the sum of
# the non-normalized values within a cell We calculate the percentage of
# mitochondrial genes here and store it in percent.mito using AddMetaData.
# We use object@raw.data since this represents non-transformed and
# non-log-normalized counts The % of UMI mapping to MT-genes is a common
# scRNA-seq QC metric.
seurat[["percent.mt"]] <- PercentageFeatureSet(object = seurat, pattern = "^MT-")
##VlnPlot(object = seurat, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
```


```{r scatter_plot_v3, eval = FALSE}
# FeatureScatter is typically used to visualize gene-gene relationships, but can
# be used for anything calculated by the object, i.e. columns in
# object@meta.data, PC scores etc.  Since there is a rare subset of cells
# with an outlier level of high mitochondrial percentage and also low UMI
# content, we filter these as well
par(mfrow = c(1, 2))
FeatureScatter(object = seurat, feature1 = "nCount_RNA", feature2 = "percent.mt")
FeatureScatter(object = seurat, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
```


```{r subset_data, eval = FALSE}
# We filter out cells that have unique gene counts over 2,500 or less than
# 200 Note that low.thresholds and high.thresholds are used to define a
# 'gate'.  -Inf and Inf should be used if you don't want a lower or upper
# threshold.

# seurat <- SubsetData(object = seurat, 
#                     subset.names = c("nFeature_RNA", "percent.mt"), 
#                     low.thresholds = c(200, -Inf), 
#                     high.thresholds = c(2500, 0.1))


seurat <- subset(seurat, nFeature_RNA < 2500)
seurat <- subset(seurat, nFeature_RNA > 200)
seuart <- subset(seurat, percent.mt < 10)
```




```{r norm_data, eval = FALSE}
seurat <- NormalizeData(object = seurat, normalization.method = "LogNormalize", scale.factor = 10000)
```


```{r cell_cycle_genes_pbmc, eval = FALSE}
# Read in a list of cell cycle markers, from Tirosh et al, 2015.
# We can segregate this list into markers of G2/M phase and markers of S phase.
s.genes <- Seurat::cc.genes$s.genes
s.genes <- s.genes[s.genes %in% rownames(seurat)] # genes in dataset
g2m.genes <- Seurat::cc.genes$g2m.genes
g2m.genes <- g2m.genes[g2m.genes %in% rownames(seurat)] # genes in dataset

seurat <- CellCycleScoring(object = seurat, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
```

```{r var_genes_clust, eval = FALSE}
seurat <- FindVariableFeatures(object = seurat, 
                               mean.function = ExpMean, 
                               dispersion.function = LogVMR)
```



### Start of Identifying Cell Types

#### Scaling

This part is where you mean center the data, substract the mean.  You also divide by the standard deviation to make everything to a 'standard normal', where the mean is zero and the standard deviation is 1.

```{r regress, eval = FALSE}
seurat <- ScaleData(object = seurat, vars.to.regress = c("batch", "percent.mt"))
```

**Task: Try Regressing Other Variables**
```{r regrress_other, eval = FALSE}
## randomly making a batch id data.frame
batch_ids <- data.frame(barcode = rownames(seurat@meta.data), 
                        batch_id = sample(0:2, NROW(seurat@meta.data), replace = TRUE),
                        stringsAsFactors = FALSE)


row.names(batch_ids) <- row.names(seurat@meta.data)
seurat <- AddMetaData(object = seurat, metadata = batch_ids, col.name = NULL)
seurat <- ScaleData(object = seurat, vars.to.regress = 'batch_id')
```


#### Perform linear dimensional reduction (PCA)

This will run pca on the 

```{r run_pca, eval = FALSE}
seurat <- RunPCA(object = seurat, 
               # features = seurat@assays$RNA@var.features, 
               ndims.print = 1:5, 
               nfeatures.print = 5)
```


#### Visualizing PCA in Different Ways

```{r plot_pca, eval = FALSE}
DimPlot(seurat, reduction = "pca")
```

#### Perform linear dimensional reduction (ICA)

**Task:** Try running Independent Component Analysis.  If you need help with the inputs try using the ?RunICA menu.  


```{r run_ica, eval = FALSE}
seurat <- RunICA()
```


#### Visualizing ICA in Different Ways

```{r plot_ica, eval = FALSE}
DimPlot()
```



```{r project_pca, eval = FALSE}
# ProjectDim scores each gene in the dataset (including genes not included
# in the PCA) based on their correlation with the calculated components.
# Though we don't use this further here, it can be used to identify markers
# that are strongly correlated with cellular heterogeneity, but may not have
# passed through variable gene selection.  The results of the projected PCA
# can be explored by setting use.full=T in the functions above
seurat <- ProjectDim(object = seurat, reduction = "pca")
```

#### Genes by PCs

```{r pca_heatmap, eval = FALSE}
DimHeatmap(object = seurat, 
          dims = 1:6, 
          cells = 50, 
          reduction = "pca",
          balanced = TRUE)
```


Check other PCs to plot

**Task: Check other PCs**
```{r, eval = FALSE}
DimHeatmap()
```


```{r jack_straw, eval = FALSE}
seurat <- JackStraw(object = seurat, reduction = "pca")
seurat <- ScoreJackStraw(seurat, dims = 1:20)
```



```{r jack_straw_plot, eval = FALSE}
JackStrawPlot(seurat, dims = 1:15)
```


```{r pcsiggenes, eval = FALSE, echo = FALSE}
PCASigGenes(object = seurat, pcs.use = 1:2, pval.cut = 0.1)
```




```{r pca_elbow, eval = FALSE}
ElbowPlot(object = seurat, ndims = 10, reduction = "pca")
```



```{r find_clusters, eval = FALSE}
# save.SNN = T saves the SNN so that the clustering algorithm can be rerun
# using the same graph but with a different resolution value (see docs for
# full details)
set.seed(2020)
seurat <- FindNeighbors(object = seurat, dims = 1:10)
seurat <- FindClusters(object = seurat, 
                       reduction = "pca", 
                       dims = 1:10, 
                       resolution = 0.5,
                       random.seed = 2020)
```


### Run non-linear dimensional reduction (UMAP/tSNE)

Seurat offers several non-linear dimensional reduction techniques, such as tSNE and UMAP, to visualize and explore these datasets. The goal of these algorithms is to learn the underlying manifold of the data in order to place similar cells together in low-dimensional space. Cells within the graph-based clusters determined above should co-localize on these dimension reduction plots. As input to the UMAP and tSNE, we suggest using the same PCs as input to the clustering analysis.

```{r run_tsne, eval = FALSE}
set.seed(2020)
seurat <- RunTSNE(seurat, reduction.use = "pca", dims.use = 1:10, perplexity=10)

# note that you can set do.label=T to help label individual clusters
DimPlot(object = seurat, reduction = "tsne")
```


**Task: Try using UMAP for the non-linear dimension reduction technique (hint: `?RunUMAP`)**
```{r run_umap, eval = FALSE}
set.seed(2020)
seurat <- RunUMAP()

# note that you can set label=TRUE to help label individual clusters
DimPlot()
```


#### Finding differentially expressed features (cluster biomarkers)

Seurat can help you find markers that define clusters via differential expression. By default, it identifes positive and negative markers of a single cluster (specified in ident.1), compared to all other cells. FindAllMarkers automates this process for all clusters, but you can also test groups of clusters vs. each other, or against all cells.

The min.pct argument requires a feature to be detected at a minimum percentage in either of the two groups of cells, and the thresh.test argument requires a feature to be differentially expressed (on average) by some amount between the two groups. You can set both of these to 0, but with a dramatic increase in time - since this will test a large number of features that are unlikely to be highly discriminatory. As another option to speed up these computations, max.cells.per.ident can be set. This will downsample each identity class to have no more cells than whatever this is set to. While there is generally going to be a loss in power, the speed increases can be significiant and the most highly differentially expressed features will likely still rise to the top.

```{r find_cluster1, eval = FALSE}
# find all markers of cluster 1 using default parameters
cluster1.markers <- FindMarkers(object = seurat, ident.1 = 1, min.pct = 0.25)

head(cluster1.markers)
```


**Task: Try tuning different parameters. How does that affect results?**

```{r find_cluster5, eval = FALSE}
# find all markers distinguishing cluster 5 from clusters 0 and 1
cluster5.markers <- FindMarkers(object = seurat, 
                                ident.1 = 5, ident.2 = c(0, 1), 
                                min.pct = ?? 
                                test.use = ??, 
                                only.pos = ??)

head(cluster5.markers)
```


```{r cluster3_markers, eval = FALSE}
cluster3.markers <- FindMarkers(object = seurat, 
                                ident.1 = 3, 
                                thresh.use = 0.25, 
                                test.use = "roc", 
                                only.pos = TRUE)

head(cluster3.markers)
```


```{r plot_cluster3_markers, eval = FALSE}
VlnPlot(object = seurat, features = c("MS4A1", "CD79A"))
```


```{r plot_by_umi, eval = FALSE}
# you can plot raw UMI counts as well
VlnPlot(object = seurat, 
        features = c("NKG7", "PF4"), 
        log = TRUE)
```


```{r find_markers, eval = FALSE}
# find markers for every cluster compared to all remaining cells, report
# only the positive ones
pbmc.markers <- FindAllMarkers(object = seurat, only.pos = TRUE, min.pct = 0.25, thresh.use = 0.25)
pbmc.markers %>% group_by(cluster) %>% top_n(2, avg_logFC)
```


```{r gene_tsne, eval = FALSE}
FeaturePlot(object = seurat, 
            features = c("MS4A1", "GNLY", "CD3E", "CD14", "FCER1A", "FCGR3A", "LYZ", "PPBP", "CD8A"),
            cols = c("grey", "blue"), 
            reduction = "tsne")
```



```{r do_heatmap, eval = FALSE}
top10 <- pbmc.markers %>% group_by(cluster) %>% top_n(10, avg_logFC)
# setting slim.col.label to TRUE will print just the cluster IDS instead of
# every cell name
DoHeatmap(object = seurat, features = top10$gene, label = TRUE)
```

```{r new_names, eval = FALSE}
new.cluster.ids <- c("Memory CD4 T", "Naive CD4 T", "CD14+ Mono", "B", "CD8 T", "FCGR3A+ Mono", "NK", "DC", "Mk")
names(x = new.cluster.ids) <- levels(x = seurat)
seurat <- RenameIdents(object = seurat, new.cluster.ids)

DimPlot(object = seurat, reduction = 'tsne', label = TRUE, pt.size = 0.5) + NoLegend()
```




#### Further subdivisions within cell types

If you perturb some of our parameter choices above (for example, setting resolution=0.8 or changing the number of PCs), you might see the CD4 T cells subdivide into two groups. You can explore this subdivision to find markers separating the two T cell subsets. However, before reclustering (which will overwrite object@ident), we can stash our renamed identities to be easily recovered later.

```{r tune_params, eval = FALSE}
# First lets stash our identities for later
seurat[["ClusterNames_0.6"]] <- Idents(object = seurat)
# Note that if you set save.snn=T above, you don't need to recalculate the
# SNN, and can simply put: pbmc <- FindClusters(pbmc,resolution = 0.8)
seurat <- FindClusters(object = seurat, 
                       reduction = "pca", 
                       dims = 1:10, 
                       resolution = 0.8)
```


```{r compare_params, eval = FALSE}
set.seed(2020)
## Warning in BuildSNN(object = object, genes.use = genes.use, reduction.type
## = reduction.type, : Build parameters exactly match those of already
## computed and stored SNN. To force recalculation, set force.recalc to TRUE.
# Demonstration of how to plot two tSNE plots side by side, and how to color
# points based on different criteria
plot1 <- DimPlot(object = seurat, 
                 reduction= "tsne",
                 label = TRUE) + NoLegend()

plot2 <- DimPlot(object = seurat, 
                  reduction = "tsne", 
                  group.by = "ClusterNames_0.6",
                  label = TRUE) + NoLegend()

CombinePlots(list(plot1, plot2))
```



```{r tcell_markers, eval = FALSE}
# Find discriminating markers
tcell.markers <- FindMarkers(object = seurat, ident.1 = 0, ident.2 = 1)

# Most of the markers tend to be expressed in C1 (i.e. S100A4). However, we
# can see that CCR7 is upregulated in C0, strongly indicating that we can
# differentiate memory from naive CD4 cells.  cols demarcates the color
# palette from low to high expression
FeaturePlot(object = seurat, features = c("S100A4", "CCR7"), cols = c("green", "blue"))
```


## Feature Selection

### Differential Expression Analysis

#### Differential Expression Tests

  One of the most commonly performed tasks for RNA-seq data is differential gene expression (DE) analysis. Although well-established tools exist for such analysis in bulk RNA-seq data, methods for scRNA-seq data are just emerging. Given the special characteristics of scRNA-seq data, including generally low library sizes, high noise levels and a large fraction of so-called ‘dropout’ events, it is unclear whether DE methods that have been developed for bulk RNA-seq are suitable also for scRNA-seq.  Check the help page out for the FindMarkers function by using <code> ?FindMarkers </code>

```{r difftest, eval = FALSE}
## Differential expression using t-test
FindMarkers(object = seurat, ident.1 = 0, ident.2 = 1, test.use = "t")
```


**Task:** Try to use different test for diffential expression analysis (hint: `?FindMarkers`)
```{r, eval = FALSE}
## Use the help function and run other tests.  Do they find similar markers?
FindMarkers(object = seurat, ident.1 = 0, ident.2 = 1, test.use = ??)
```


### Check Clusters

How do we test the cell types identified?  How do we know how reliable they are?

Use Classifier to predict cell cluster.  See how it predicts using hold out data.
[reference](https://satijalab.org/seurat/v3.0/integration.html)

```{r check_clusters, eval = FALSE}
# Assign the test object a three level attribute
groups <- sample(c("train", "test"), size = NROW(seurat@meta.data), replace = TRUE, prob = c(0.8, 0.2))
names(groups) <- colnames(seurat)
seurat <- AddMetaData(object = seurat, metadata = groups, col.name = "group")

# Find Anchors
seurat.list <- SplitObject(seurat, split.by = "group")
seurat.anchors <- FindIntegrationAnchors(object.list = seurat.list, dims = 1:30)
seurat.integrated <- IntegrateData(anchorset = seurat.anchors, dims = 1:30)

seurat.query <- seurat.list[["train"]]
seurat.anchors <- FindTransferAnchors(reference = seurat.integrated, 
                                      query = seurat.query, 
                                      dims = 1:30)

predictions <- TransferData(anchorset = seurat.anchors, 
                            refdata = seurat.integrated$ClusterNames_0.6, 
                            dims = 1:30)

seurat.query <- AddMetaData(seurat.query, metadata = predictions)
table(seurat.query@meta.data$ClusterNames_0.6, seurat.query@meta.data$predicted.id)
```


### View Entire Object Structure

Notice all the slots and elements added to the object.  

```{r review_str, eval = FALSE}
str(seurat)
```


#### Probabilistic (LDA)

Another type of clustering we can do is a fuzzy or probablistic clustering.  This is where cells are not assigned to specifically only one cluster.  They get assigned a score for how much the cells belong to each of the clusters (sometimes called topics).  This can be helpful for when your dataset continuous processes and/or cellular states as opposed to distinct cell types.


```{r example_lda, eval = FALSE}
data("MouseDeng2014.FitGoM")
names(MouseDeng2014.FitGoM)
omega <- MouseDeng2014.FitGoM$clust_6$omega


annotation <- data.frame(
  sample_id = paste0("X", c(1:NROW(omega))),
  tissue_label = factor(rownames(omega),
                        levels = rev(c("zy", "early2cell", "mid2cell", "late2cell",
                                        "4cell", "8cell", "16cell", "earlyblast", "midblast", "lateblast")))
  )

rownames(omega) <- annotation$sample_id;

StructureGGplot(omega = omega,
                annotation = annotation,
                palette = RColorBrewer::brewer.pal(8, "Accent"),
                yaxis_label = "Amplification batch",
                order_sample = TRUE,
                axis_tick = list(axis_ticks_length = .1,
                                 axis_ticks_lwd_y = .1,
                                 axis_ticks_lwd_x = .1,
                                 axis_label_size = 7,
                                 axis_label_face = "bold"))
```


```{r pbmc_lda, eval = FALSE}
set.seed(2020)

## Preprocessing Steps
pbmc_small <- NormalizeData(object = pbmc_small, normalization.method = "LogNormalize", scale.factor = 10000)
pbmc_small <- RunPCA(object = pbmc_small)
pbmc_small <- FindClusters(object = pbmc_small, 
                           reduction = "pca", 
                           dims.use = 1:10, 
                           resolution = 1, 
                           print.output = 0)

pbmc_counts <- as.matrix(pbmc_small@assays$RNA@data)
pbmc_meta <- pbmc_small@meta.data
gene_names <- rownames(pbmc_counts)


pbmc_FitGoM <- FitGoM(t(pbmc_counts), K=4)


omega <- data.frame(pbmc_FitGoM$fit$omega)

annotation <- data.frame(sample_id = rownames(omega),
                         tissue_label = paste0("cluster", pbmc_small@active.ident))

colnames(omega) <- paste0("topic", 1:4)
rownames(omega) <- annotation$sample_id;

StructureGGplot(omega = omega,
                annotation = annotation,
                palette = RColorBrewer::brewer.pal(4, "Dark2"),
                yaxis_label = "Cells",
                order_sample = TRUE,
                axis_tick = list(axis_ticks_length = .1,
                                 axis_ticks_lwd_y = .1,
                                 axis_ticks_lwd_x = .1,
                                 axis_label_size = 7,
                                 axis_label_face = "bold"))


# ## Add Topic Scores to Meta Data Part of the Seurat Object
pbmc_small <- AddMetaData(pbmc_small, omega)
```

```{r topic_summary, eval = FALSE}
pbmc_small@meta.data %>% 
  group_by(RNA_snn_res.1) %>% 
  summarise(topic1 = mean(topic1),
            topic2 = mean(topic2),
            topic3 = mean(topic3),
            topic4 = mean(topic4))
```


```{r tsne_plot_pbmc, eval = FALSE}
## ggplot object, you can add layers
p1 <- DimPlot(pbmc_small, reduction = "tsne") + labs(title = "Resolution 1") ## return ggplot object
p1
```

```{r topic_plot, eval = FALSE}
p2 <- FeaturePlot(object = pbmc_small, 
                  features = c("topic1", "topic2", "topic3", "topic4"), 
                  cols = c("grey", "blue"), 
                  reduction = "tsne") ## return ggplot object

p2
```

```{r combine_topic_plot, eval = FALSE}
CombinePlots(list(p1, p2))
```

#### Extract Top Feature

```{r topic_genes, eval = FALSE}

theta_mat <- pbmc_FitGoM$fit$theta

top_features <- ExtractTopFeatures(theta_mat, 
                                   top_features=100,
                                   method="poisson", 
                                   options="min")

gene_list <- do.call(rbind, 
                     lapply(1:dim(top_features$indices)[1],
                            function(x) gene_names[top_features$indices[x,]]))
```


We tabulate the top `5` genes for these `4` topics

```{r topic_genes_by_cluster, eval = FALSE}
out_table <- do.call(rbind, lapply(1:4, function(i) toString(gene_list[i,1:5])))
rownames(out_table) <- paste("Topic", c(1:4))
out_table
```



### Practice Visualizing/Embedding

#### tSNE

Change the parameter settings for tSNE

```{r, eval = FALSE}
RunTSNE()
```

#### UMAP

Change the parameter settings for UMAP

```{r, eval = FALSE}
RunUMAP()
```




## Other Options For Analysis

  - [More Seurat Vignettes](https://satijalab.org/seurat/vignettes.html)
  - [Single Cell Analysis Workshop](https://broadinstitute.github.io/2019_scWorkshop/)
  - [Hemberg Lab Course](https://scrnaseq-course.cog.sanger.ac.uk/website/index.html)
  - [SingleCellExperiment](https://bioconductor.org/packages/release/bioc/vignettes/SingleCellExperiment/inst/doc/intro.html)
  - [Scanpy](https://scanpy-tutorials.readthedocs.io/en/latest/pbmc3k.html)
  - [Pegasus](https://github.com/klarman-cell-observatory/pegasus/blob/master/notebooks/Pegasus%20Tutorial%20in%20Workshop.ipynb)


<!--chapter:end:10-Cluster-Analysis.Rmd-->

---
title: "Psuedotime-trajectories"
author: "Kirk Gosik"
date: "3/22/2020"
output: html_document
---

# Advanced Topics: Pseudotime Cell Trajectories


## Google Slides


<!--chapter:end:14-pseudotime-trajectories.Rmd-->

---
title: "19-resources"
output: html_document
---

# Single Cell Resources

## Comprehensive list of single-cell resources
  - https://github.com/seandavi/awesome-single-cell

## Computational packages for single-cell analysis
  - http://bioconductor.org/packages/devel/workflows/html/simpleSingleCell.html
  - https://satijalab.org/seurat/
  - https://scanpy.readthedocs.io/

## eLife Commentary on the Human Cell Atlas 
[link](https://elifesciences.org/articles/27041)
  - Nature Commentary on the Human Cell Atlas 
  - https://www.nature.com/news/the-human-cell-atlas-from-vision-to-reality-1.22854

## Online courses
  - [Analysis of single cell RNA-seq data](https://scrnaseq-course.cog.sanger.ac.uk/website/index.html)
  - [Single Cell Genomics Day](https://satijalab.org/scgd/)
  - [Simple Single Cell](https://bioconductor.org/packages/release/workflows/html/simpleSingleCell.html)
  - [Single Cell Transcriptomics](https://github.com/SingleCellTranscriptomics)
  - [R for Data Sciences](https://r4ds.had.co.nz/)

<!--chapter:end:19-resources-2019.Rmd-->

`r if (knitr::is_html_output()) '# References {-}'`

<!--chapter:end:99-references.Rmd-->

