<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>16 CITE-Seq | ANALYSIS OF SINGLE CELL RNA-SEQ DATA</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="16 CITE-Seq | ANALYSIS OF SINGLE CELL RNA-SEQ DATA" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://broadinstitute.github.io/2020_scWorkshop/" />
  <meta property="og:image" content="https://broadinstitute.github.io/2020_scWorkshop/images/cover.png" />
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="broadinstitute/2019_scWorkship" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="16 CITE-Seq | ANALYSIS OF SINGLE CELL RNA-SEQ DATA" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="twitter:image" content="https://broadinstitute.github.io/2020_scWorkshop/images/cover.png" />

<meta name="author" content="Orr Ashenberg" />
<meta name="author" content="Dana Silverbush" />
<meta name="author" content="Kirk Gosik" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="mutli-omic-analysis.html"/>
<link rel="next" href="diy-project.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">ANALYSIS OF SINGLE CELL RNA-SEQ DATA</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#course-overview"><i class="fa fa-check"></i><b>1.1</b> COURSE OVERVIEW</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#targeted-audience-assumed-background"><i class="fa fa-check"></i><b>1.2</b> TARGETED AUDIENCE &amp; ASSUMED BACKGROUND</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#course-format"><i class="fa fa-check"></i><b>1.3</b> COURSE FORMAT</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#getting-started"><i class="fa fa-check"></i><b>1.4</b> Getting Started</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#session-content"><i class="fa fa-check"></i><b>1.5</b> SESSION CONTENT</a><ul>
<li class="chapter" data-level="1.5.1" data-path="index.html"><a href="index.html#monday-classes-from-0800-to-1600-lunch-break-1-hr-40-min-of-total-coffee-breaks"><i class="fa fa-check"></i><b>1.5.1</b> Monday – Classes from 08:00 to 16:00 (lunch break-1 hr, 40 min of total coffee breaks)</a></li>
<li class="chapter" data-level="1.5.2" data-path="index.html"><a href="index.html#tuesday-classes-from-0800-to-1600"><i class="fa fa-check"></i><b>1.5.2</b> Tuesday – Classes from 08:00 to 16:00</a></li>
<li class="chapter" data-level="1.5.3" data-path="index.html"><a href="index.html#wednesday-classes-from-0800-to-1600"><i class="fa fa-check"></i><b>1.5.3</b> Wednesday – Classes from 08:00 to 16:00</a></li>
<li class="chapter" data-level="1.5.4" data-path="index.html"><a href="index.html#thursday-classes-from-0800-to-1600"><i class="fa fa-check"></i><b>1.5.4</b> Thursday – Classes from 08:00 to 16:00</a></li>
<li class="chapter" data-level="1.5.5" data-path="index.html"><a href="index.html#friday-classes-from-0800-to-1600"><i class="fa fa-check"></i><b>1.5.5</b> Friday – Classes from 08:00 to 16:00</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction-1.html"><a href="introduction-1.html"><i class="fa fa-check"></i><b>2</b> Introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="introduction-1.html"><a href="introduction-1.html#slides"><i class="fa fa-check"></i><b>2.1</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html"><i class="fa fa-check"></i><b>3</b> Understanding Sequencing Raw Data</a><ul>
<li class="chapter" data-level="3.1" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#class-environment"><i class="fa fa-check"></i><b>3.1</b> Class Environment</a><ul>
<li class="chapter" data-level="3.1.1" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#getting-into-aws-instance"><i class="fa fa-check"></i><b>3.1.1</b> Getting into AWS Instance</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#shell-and-unix-commands"><i class="fa fa-check"></i><b>3.2</b> Shell and Unix commands</a><ul>
<li class="chapter" data-level="3.2.1" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#common-linux-commands"><i class="fa fa-check"></i><b>3.2.1</b> Common Linux Commands</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#file-formats"><i class="fa fa-check"></i><b>3.3</b> File formats</a><ul>
<li class="chapter" data-level="3.3.1" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#view-fastq-files"><i class="fa fa-check"></i><b>3.3.1</b> View FASTQ Files</a></li>
<li class="chapter" data-level="3.3.2" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#view-bam-files"><i class="fa fa-check"></i><b>3.3.2</b> View BAM Files</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#public-data-repositories"><i class="fa fa-check"></i><b>3.4</b> Public data repositories</a><ul>
<li class="chapter" data-level="3.4.1" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#cellranger10x"><i class="fa fa-check"></i><b>3.4.1</b> Cellranger/10x</a></li>
<li class="chapter" data-level="3.4.2" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#geo"><i class="fa fa-check"></i><b>3.4.2</b> GEO</a></li>
<li class="chapter" data-level="3.4.3" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#single-cell-portal"><i class="fa fa-check"></i><b>3.4.3</b> Single Cell Portal</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#docker-commands"><i class="fa fa-check"></i><b>3.5</b> Docker Commands</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="transcriptome-quantification.html"><a href="transcriptome-quantification.html"><i class="fa fa-check"></i><b>4</b> Transcriptome Quantification</a><ul>
<li class="chapter" data-level="4.1" data-path="transcriptome-quantification.html"><a href="transcriptome-quantification.html#google-slides"><i class="fa fa-check"></i><b>4.1</b> Google Slides</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html"><i class="fa fa-check"></i><b>5</b> Processing scRNAseq Data</a><ul>
<li class="chapter" data-level="5.1" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#goal"><i class="fa fa-check"></i><b>5.1</b> Goal</a></li>
<li class="chapter" data-level="5.2" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#further-reading"><i class="fa fa-check"></i><b>5.2</b> Further reading</a></li>
<li class="chapter" data-level="5.3" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#download-data"><i class="fa fa-check"></i><b>5.3</b> Download data</a><ul>
<li class="chapter" data-level="5.3.1" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#fastq-file-format"><i class="fa fa-check"></i><b>5.3.1</b> Fastq file format</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#align-the-reads"><i class="fa fa-check"></i><b>5.4</b> Align the reads</a><ul>
<li class="chapter" data-level="5.4.1" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#star-align"><i class="fa fa-check"></i><b>5.4.1</b> STAR align</a></li>
<li class="chapter" data-level="5.4.2" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#bam-file-format"><i class="fa fa-check"></i><b>5.4.2</b> Bam file format</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#visualization"><i class="fa fa-check"></i><b>5.5</b> Visualization</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html"><i class="fa fa-check"></i><b>6</b> Introduction R/Bioconductor</a><ul>
<li class="chapter" data-level="6.1" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#start-environment"><i class="fa fa-check"></i><b>6.1</b> Start Environment</a><ul>
<li class="chapter" data-level="6.1.1" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#local-command"><i class="fa fa-check"></i><b>6.1.1</b> Local Command</a></li>
<li class="chapter" data-level="6.1.2" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#aws-command"><i class="fa fa-check"></i><b>6.1.2</b> AWS Command</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#installing-packages"><i class="fa fa-check"></i><b>6.2</b> Installing packages</a><ul>
<li class="chapter" data-level="6.2.1" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#cran"><i class="fa fa-check"></i><b>6.2.1</b> CRAN</a></li>
<li class="chapter" data-level="6.2.2" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#github"><i class="fa fa-check"></i><b>6.2.2</b> Github</a></li>
<li class="chapter" data-level="6.2.3" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#bioconductor"><i class="fa fa-check"></i><b>6.2.3</b> Bioconductor</a></li>
<li class="chapter" data-level="6.2.4" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#source"><i class="fa fa-check"></i><b>6.2.4</b> Source</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#installation-instructions"><i class="fa fa-check"></i><b>6.3</b> Installation instructions:</a><ul>
<li class="chapter" data-level="6.3.1" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#classestypes"><i class="fa fa-check"></i><b>6.3.1</b> Classes/Types</a></li>
<li class="chapter" data-level="6.3.2" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#data-structures"><i class="fa fa-check"></i><b>6.3.2</b> Data structures</a></li>
<li class="chapter" data-level="6.3.3" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#detour-to-s3s4"><i class="fa fa-check"></i><b>6.3.3</b> Detour to S3/S4</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#more-information"><i class="fa fa-check"></i><b>6.4</b> More information</a><ul>
<li class="chapter" data-level="6.4.1" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#checking-for-help-for-any-function"><i class="fa fa-check"></i><b>6.4.1</b> Checking for help for any function!</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#grammer-of-graphics-ggplot2"><i class="fa fa-check"></i><b>6.5</b> Grammer of Graphics (ggplot2)</a><ul>
<li class="chapter" data-level="6.5.1" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#what-is-ggplot2"><i class="fa fa-check"></i><b>6.5.1</b> What is ggplot2?</a></li>
<li class="chapter" data-level="6.5.2" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#principles-of-ggplot2"><i class="fa fa-check"></i><b>6.5.2</b> Principles of ggplot2</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#reference"><i class="fa fa-check"></i><b>6.6</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>7</b> Quality Control</a><ul>
<li class="chapter" data-level="7.1" data-path="quality-control.html"><a href="quality-control.html#slides-1"><i class="fa fa-check"></i><b>7.1</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html"><i class="fa fa-check"></i><b>8</b> Data Wrangling scRNAseq</a><ul>
<li class="chapter" data-level="8.1" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#goal-1"><i class="fa fa-check"></i><b>8.1</b> Goal</a></li>
<li class="chapter" data-level="8.2" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#introduction-2"><i class="fa fa-check"></i><b>8.2</b> Introduction</a><ul>
<li class="chapter" data-level="8.2.1" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#load-necessary-packages"><i class="fa fa-check"></i><b>8.2.1</b> Load necessary packages</a></li>
<li class="chapter" data-level="8.2.2" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#read-in-nsclc-counts-matrix."><i class="fa fa-check"></i><b>8.2.2</b> Read in NSCLC counts matrix.</a></li>
<li class="chapter" data-level="8.2.3" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#lets-examine-the-sparse-counts-matrix"><i class="fa fa-check"></i><b>8.2.3</b> Let’s examine the sparse counts matrix</a></li>
<li class="chapter" data-level="8.2.4" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#how-big-is-the-matrix"><i class="fa fa-check"></i><b>8.2.4</b> How big is the matrix?</a></li>
<li class="chapter" data-level="8.2.5" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#how-much-memory-does-a-sparse-matrix-take-up-relative-to-a-dense-matrix"><i class="fa fa-check"></i><b>8.2.5</b> How much memory does a sparse matrix take up relative to a dense matrix?</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#filtering-low-quality-cells"><i class="fa fa-check"></i><b>8.3</b> Filtering low-quality cells</a><ul>
<li class="chapter" data-level="8.3.1" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#look-at-the-summary-counts-for-genes-and-cells"><i class="fa fa-check"></i><b>8.3.1</b> Look at the summary counts for genes and cells</a></li>
<li class="chapter" data-level="8.3.2" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#plot-cells-ranked-by-their-number-of-detected-genes."><i class="fa fa-check"></i><b>8.3.2</b> Plot cells ranked by their number of detected genes.</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#beginning-with-seurat-httpsatijalab.orgseurat"><i class="fa fa-check"></i><b>8.4</b> Beginning with Seurat: <span>http://satijalab.org/seurat/</span></a><ul>
<li class="chapter" data-level="8.4.1" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#creating-a-seurat-object"><i class="fa fa-check"></i><b>8.4.1</b> Creating a seurat object</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#preprocessing-step-1-filter-out-low-quality-cells"><i class="fa fa-check"></i><b>8.5</b> Preprocessing step 1 : Filter out low-quality cells</a></li>
<li class="chapter" data-level="8.6" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#examine-contents-of-seurat-object"><i class="fa fa-check"></i><b>8.6</b> Examine contents of Seurat object</a><ul>
<li class="chapter" data-level="8.6.1" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#preprocessing-step-2-expression-normalization"><i class="fa fa-check"></i><b>8.6.1</b> Preprocessing step 2 : Expression normalization</a></li>
</ul></li>
<li class="chapter" data-level="8.7" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#detection-of-variable-genes-across-the-single-cells"><i class="fa fa-check"></i><b>8.7</b> Detection of variable genes across the single cells</a></li>
<li class="chapter" data-level="8.8" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#gene-set-expression-across-cells"><i class="fa fa-check"></i><b>8.8</b> Gene set expression across cells</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="identifying-cell-populations.html"><a href="identifying-cell-populations.html"><i class="fa fa-check"></i><b>9</b> Identifying Cell Populations</a><ul>
<li class="chapter" data-level="9.1" data-path="identifying-cell-populations.html"><a href="identifying-cell-populations.html#slides-2"><i class="fa fa-check"></i><b>9.1</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html"><i class="fa fa-check"></i><b>10</b> Feature Selection and Cluster Analysis</a><ul>
<li class="chapter" data-level="10.1" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#abstract"><i class="fa fa-check"></i><b>10.1</b> Abstract</a></li>
<li class="chapter" data-level="10.2" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#seurat-tutorial-redo"><i class="fa fa-check"></i><b>10.2</b> Seurat Tutorial Redo</a><ul>
<li class="chapter" data-level="10.2.1" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#preprocessing-steps"><i class="fa fa-check"></i><b>10.2.1</b> Preprocessing Steps</a></li>
<li class="chapter" data-level="10.2.2" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#start-of-identifying-cell-types"><i class="fa fa-check"></i><b>10.2.2</b> Start of Identifying Cell Types</a></li>
<li class="chapter" data-level="10.2.3" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#run-non-linear-dimensional-reduction-umaptsne"><i class="fa fa-check"></i><b>10.2.3</b> Run non-linear dimensional reduction (UMAP/tSNE)</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#feature-selection"><i class="fa fa-check"></i><b>10.3</b> Feature Selection</a><ul>
<li class="chapter" data-level="10.3.1" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#differential-expression-analysis"><i class="fa fa-check"></i><b>10.3.1</b> Differential Expression Analysis</a></li>
<li class="chapter" data-level="10.3.2" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#check-clusters"><i class="fa fa-check"></i><b>10.3.2</b> Check Clusters</a></li>
<li class="chapter" data-level="10.3.3" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#view-entire-object-structure"><i class="fa fa-check"></i><b>10.3.3</b> View Entire Object Structure</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#probabilistic-lda-clustering"><i class="fa fa-check"></i><b>10.4</b> Probabilistic (LDA) Clustering</a><ul>
<li class="chapter" data-level="10.4.1" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#example-lda-in-bulk"><i class="fa fa-check"></i><b>10.4.1</b> Example LDA in Bulk</a></li>
<li class="chapter" data-level="10.4.2" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#pbmc-lda"><i class="fa fa-check"></i><b>10.4.2</b> PBMC LDA</a></li>
<li class="chapter" data-level="10.4.3" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#practice-visualizingembedding"><i class="fa fa-check"></i><b>10.4.3</b> Practice Visualizing/Embedding</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#other-options-for-analysis"><i class="fa fa-check"></i><b>10.5</b> Other Options For Analysis</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="batch-correction-lecture.html"><a href="batch-correction-lecture.html"><i class="fa fa-check"></i><b>11</b> Batch Correction Lecture</a><ul>
<li class="chapter" data-level="11.1" data-path="batch-correction-lecture.html"><a href="batch-correction-lecture.html#slides-3"><i class="fa fa-check"></i><b>11.1</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html"><i class="fa fa-check"></i><b>12</b> Batch Correction Lab</a><ul>
<li class="chapter" data-level="12.1" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#load-settings-and-packages"><i class="fa fa-check"></i><b>12.1</b> Load settings and packages</a></li>
<li class="chapter" data-level="12.2" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#read-in-pancreas-expression-matrices"><i class="fa fa-check"></i><b>12.2</b> Read in pancreas expression matrices</a></li>
<li class="chapter" data-level="12.3" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#preparing-the-individual-seurat-objects-for-each-pancreas-dataset-without-batch-correction"><i class="fa fa-check"></i><b>12.3</b> Preparing the individual Seurat objects for each pancreas dataset without batch correction</a></li>
<li class="chapter" data-level="12.4" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#cluster-pancreatic-datasets-without-batch-correction"><i class="fa fa-check"></i><b>12.4</b> Cluster pancreatic datasets without batch correction</a><ul>
<li class="chapter" data-level="12.4.1" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#batch-correction-canonical-correlation-analysis-cca-mutual-nearest-neighbors-mnn-using-seurat-v3"><i class="fa fa-check"></i><b>12.4.1</b> Batch correction: canonical correlation analysis (CCA) + mutual nearest neighbors (MNN) using Seurat v3</a></li>
<li class="chapter" data-level="12.4.2" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#batch-correction-integrative-non-negative-matrix-factorization-nmf-using-liger"><i class="fa fa-check"></i><b>12.4.2</b> Batch correction: integrative non-negative matrix factorization (NMF) using LIGER</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#additional-exploration-regressing-out-unwanted-covariates"><i class="fa fa-check"></i><b>12.5</b> Additional exploration: Regressing out unwanted covariates</a></li>
<li class="chapter" data-level="12.6" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#additional-exploration-kbet"><i class="fa fa-check"></i><b>12.6</b> Additional exploration: kBET</a></li>
<li class="chapter" data-level="12.7" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#additional-exploration-seurat-3"><i class="fa fa-check"></i><b>12.7</b> Additional exploration: Seurat 3</a></li>
<li class="chapter" data-level="12.8" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#acknowledgements"><i class="fa fa-check"></i><b>12.8</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="trajectory-inference.html"><a href="trajectory-inference.html"><i class="fa fa-check"></i><b>13</b> Trajectory Inference</a><ul>
<li class="chapter" data-level="13.1" data-path="trajectory-inference.html"><a href="trajectory-inference.html#slides-4"><i class="fa fa-check"></i><b>13.1</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html"><i class="fa fa-check"></i><b>14</b> Trajectory Analysis</a><ul>
<li class="chapter" data-level="14.1" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#load-settings-and-packages-1"><i class="fa fa-check"></i><b>14.1</b> Load settings and packages</a></li>
<li class="chapter" data-level="14.2" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#first-look-at-the-differentiation-data-from-deng-et-al."><i class="fa fa-check"></i><b>14.2</b> First look at the differentiation data from Deng et al.</a></li>
<li class="chapter" data-level="14.3" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#principle-components-analysis"><i class="fa fa-check"></i><b>14.3</b> Principle Components Analysis</a></li>
<li class="chapter" data-level="14.4" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#diffusion-map-pseudotime"><i class="fa fa-check"></i><b>14.4</b> Diffusion map pseudotime</a></li>
<li class="chapter" data-level="14.5" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#slingshot-map-pseudotime"><i class="fa fa-check"></i><b>14.5</b> Slingshot map pseudotime</a></li>
<li class="chapter" data-level="14.6" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#find-temporally-expressed-genes"><i class="fa fa-check"></i><b>14.6</b> Find temporally expressed genes</a></li>
<li class="chapter" data-level="14.7" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#comparison-of-the-different-trajectory-inference-methods"><i class="fa fa-check"></i><b>14.7</b> Comparison of the different trajectory inference methods</a></li>
<li class="chapter" data-level="14.8" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#plots-of-gene-expression-over-time."><i class="fa fa-check"></i><b>14.8</b> Plots of gene expression over time.</a></li>
<li class="chapter" data-level="14.9" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#acknowledgements-1"><i class="fa fa-check"></i><b>14.9</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="mutli-omic-analysis.html"><a href="mutli-omic-analysis.html"><i class="fa fa-check"></i><b>15</b> Mutli-omic Analysis</a><ul>
<li class="chapter" data-level="15.1" data-path="mutli-omic-analysis.html"><a href="mutli-omic-analysis.html#slides-5"><i class="fa fa-check"></i><b>15.1</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="cite-seq.html"><a href="cite-seq.html"><i class="fa fa-check"></i><b>16</b> CITE-Seq</a><ul>
<li class="chapter" data-level="16.1" data-path="cite-seq.html"><a href="cite-seq.html#load-settings-and-packages-2"><i class="fa fa-check"></i><b>16.1</b> Load settings and packages</a></li>
<li class="chapter" data-level="16.2" data-path="cite-seq.html"><a href="cite-seq.html#load-in-the-data"><i class="fa fa-check"></i><b>16.2</b> Load in the data</a></li>
<li class="chapter" data-level="16.3" data-path="cite-seq.html"><a href="cite-seq.html#setup-a-seurat-object-and-cluster-cells-based-on-rna-expression"><i class="fa fa-check"></i><b>16.3</b> Setup a Seurat object, and cluster cells based on RNA expression</a></li>
<li class="chapter" data-level="16.4" data-path="cite-seq.html"><a href="cite-seq.html#add-the-protein-expression-levels-to-the-seurat-object"><i class="fa fa-check"></i><b>16.4</b> Add the protein expression levels to the Seurat object</a></li>
<li class="chapter" data-level="16.5" data-path="cite-seq.html"><a href="cite-seq.html#visualize-protein-levels-on-rna-clusters"><i class="fa fa-check"></i><b>16.5</b> Visualize protein levels on RNA clusters</a></li>
<li class="chapter" data-level="16.6" data-path="cite-seq.html"><a href="cite-seq.html#identify-differentially-expressed-proteins-between-clusters"><i class="fa fa-check"></i><b>16.6</b> Identify differentially expressed proteins between clusters</a></li>
<li class="chapter" data-level="16.7" data-path="cite-seq.html"><a href="cite-seq.html#cluster-directly-on-protein-levels"><i class="fa fa-check"></i><b>16.7</b> Cluster directly on protein levels</a></li>
<li class="chapter" data-level="16.8" data-path="cite-seq.html"><a href="cite-seq.html#additional-exploration-another-example-of-multi-modal-analysis"><i class="fa fa-check"></i><b>16.8</b> Additional exploration: another example of multi-modal analysis</a></li>
<li class="chapter" data-level="16.9" data-path="cite-seq.html"><a href="cite-seq.html#acknowledgements-2"><i class="fa fa-check"></i><b>16.9</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="diy-project.html"><a href="diy-project.html"><i class="fa fa-check"></i><b>17</b> DIY Project</a><ul>
<li class="chapter" data-level="17.1" data-path="diy-project.html"><a href="diy-project.html#slides-6"><i class="fa fa-check"></i><b>17.1</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="diy-lab.html"><a href="diy-lab.html"><i class="fa fa-check"></i><b>18</b> DIY Lab</a><ul>
<li class="chapter" data-level="18.1" data-path="diy-lab.html"><a href="diy-lab.html#diy-lab-1"><i class="fa fa-check"></i><b>18.1</b> DIY Lab</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="single-cell-resources.html"><a href="single-cell-resources.html"><i class="fa fa-check"></i><b>19</b> Single Cell Resources</a><ul>
<li class="chapter" data-level="19.1" data-path="single-cell-resources.html"><a href="single-cell-resources.html#comprehensive-list-of-single-cell-resources"><i class="fa fa-check"></i><b>19.1</b> Comprehensive list of single-cell resources</a></li>
<li class="chapter" data-level="19.2" data-path="single-cell-resources.html"><a href="single-cell-resources.html#computational-packages-for-single-cell-analysis"><i class="fa fa-check"></i><b>19.2</b> Computational packages for single-cell analysis</a></li>
<li class="chapter" data-level="19.3" data-path="single-cell-resources.html"><a href="single-cell-resources.html#elife-commentary-on-the-human-cell-atlas"><i class="fa fa-check"></i><b>19.3</b> eLife Commentary on the Human Cell Atlas</a></li>
<li class="chapter" data-level="19.4" data-path="single-cell-resources.html"><a href="single-cell-resources.html#online-courses"><i class="fa fa-check"></i><b>19.4</b> Online courses</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ANALYSIS OF SINGLE CELL RNA-SEQ DATA</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="cite-seq" class="section level1">
<h1><span class="header-section-number">16</span> CITE-Seq</h1>
<p>In this lab, we will look at how single cell RNA-seq and single cell protein expression measurement datasets can be jointly analyzed, as part of a CITE-Seq experiment. To learn more about how the antibody barcode matrix is computationally generated from the sequencing data, please visit <a href="https://hoohm.github.io/CITE-seq-Count/">CITE-seq-Count</a>. To learn more about CITE-Seq and feature barcoding, please visit the <a href="https://cite-seq.com/">CITE-seq site</a>.</p>
<p>This lab closely follows the official vignette available at <a href="https://satijalab.org/seurat/multimodal_vignette.html">Using Seurat with multi-modal data</a>.</p>
<div id="load-settings-and-packages-2" class="section level2">
<h2><span class="header-section-number">16.1</span> Load settings and packages</h2>
<div class="sourceCode" id="cb633"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb633-1" data-line-number="1">knitr<span class="op">::</span>opts_chunk<span class="op">$</span><span class="kw">set</span>(<span class="dt">echo =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb633-2" data-line-number="2"></a>
<a class="sourceLine" id="cb633-3" data-line-number="3"><span class="kw">library</span>(Seurat)</a>
<a class="sourceLine" id="cb633-4" data-line-number="4"><span class="kw">library</span>(Matrix)</a>
<a class="sourceLine" id="cb633-5" data-line-number="5"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb633-6" data-line-number="6"><span class="kw">library</span>(patchwork)</a>
<a class="sourceLine" id="cb633-7" data-line-number="7"><span class="kw">library</span>(dplyr)</a>
<a class="sourceLine" id="cb633-8" data-line-number="8"><span class="kw">library</span>(plyr)</a>
<a class="sourceLine" id="cb633-9" data-line-number="9"></a>
<a class="sourceLine" id="cb633-10" data-line-number="10"><span class="co"># Set folder location for saving output files. This is also the same location as input data.</span></a>
<a class="sourceLine" id="cb633-11" data-line-number="11">mydir &lt;-<span class="st"> &quot;data/citeseq/&quot;</span></a>
<a class="sourceLine" id="cb633-12" data-line-number="12"><span class="kw">setwd</span>(<span class="st">&quot;/home/rstudio/materials/&quot;</span>) </a>
<a class="sourceLine" id="cb633-13" data-line-number="13"></a>
<a class="sourceLine" id="cb633-14" data-line-number="14"><span class="co"># Objects to save.</span></a>
<a class="sourceLine" id="cb633-15" data-line-number="15">Rda.quickload.path &lt;-<span class="st"> </span><span class="kw">paste0</span>(mydir, <span class="st">&quot;citeseq_quickload.Rda&quot;</span>)  <span class="co"># datasets saved as sparse objects</span></a>
<a class="sourceLine" id="cb633-16" data-line-number="16">Rda.RNA.path &lt;-<span class="st"> </span><span class="kw">paste0</span>(mydir, <span class="st">&quot;citeseq_RNA.Rda&quot;</span>)  <span class="co"># cbmc clustered using RNA</span></a>
<a class="sourceLine" id="cb633-17" data-line-number="17">Rda.multi.path &lt;-<span class="st"> </span><span class="kw">paste0</span>(mydir, <span class="st">&quot;citesq_cbmc_multi.rda&quot;</span>)  <span class="co"># cbmc clustered and ADT added as an assay</span></a>
<a class="sourceLine" id="cb633-18" data-line-number="18">Rda.protein.path &lt;-<span class="st"> </span><span class="kw">paste0</span>(mydir, <span class="st">&quot;citeseq_protein.Rda&quot;</span>)  <span class="co"># cbmc clustered using protein</span></a></code></pre></div>
</div>
<div id="load-in-the-data" class="section level2">
<h2><span class="header-section-number">16.2</span> Load in the data</h2>
<p>Here, we analyze a dataset of 8,617 cord blood mononuclear cells (CBMCs), produced with CITE-seq, where we simultaneously measure the single cell transcriptomes alongside the expression of 11 surface proteins, whose levels are quantified with DNA-barcoded antibodies. First, we load in two count matrices : one for the RNA measurements, and one for the antibody-derived tags (ADT).</p>
<div class="sourceCode" id="cb634"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb634-1" data-line-number="1"><span class="co"># Load in the RNA UMI matrix</span></a>
<a class="sourceLine" id="cb634-2" data-line-number="2"></a>
<a class="sourceLine" id="cb634-3" data-line-number="3"><span class="co"># Note that this dataset also contains ~5% of mouse cells, which we can use</span></a>
<a class="sourceLine" id="cb634-4" data-line-number="4"><span class="co"># as negative controls for the protein measurements. For this reason, the</span></a>
<a class="sourceLine" id="cb634-5" data-line-number="5"><span class="co"># gene expression matrix has HUMAN_ or MOUSE_ appended to the beginning of</span></a>
<a class="sourceLine" id="cb634-6" data-line-number="6"><span class="co"># each gene.</span></a>
<a class="sourceLine" id="cb634-7" data-line-number="7">cbmc.rna &lt;-<span class="st"> </span><span class="kw">as.sparse</span>(<span class="kw">read.csv</span>(<span class="kw">paste0</span>(mydir, <span class="st">&quot;GSE100866_CBMC_8K_13AB_10X-RNA_umi.csv.gz&quot;</span>), <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">row.names =</span> <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb634-8" data-line-number="8">cbmc.rna[<span class="dv">20400</span><span class="op">:</span><span class="dv">20403</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]</a>
<a class="sourceLine" id="cb634-9" data-line-number="9"></a>
<a class="sourceLine" id="cb634-10" data-line-number="10"><span class="co"># To make life a bit easier going forward, we&#39;re going to discard all but</span></a>
<a class="sourceLine" id="cb634-11" data-line-number="11"><span class="co"># the top 100 most highly expressed mouse genes, and remove the &#39;HUMAN_&#39;</span></a>
<a class="sourceLine" id="cb634-12" data-line-number="12"><span class="co"># from the CITE-seq prefix</span></a>
<a class="sourceLine" id="cb634-13" data-line-number="13">cbmc.rna &lt;-<span class="st"> </span><span class="kw">CollapseSpeciesExpressionMatrix</span>(cbmc.rna, <span class="dt">prefix =</span> <span class="st">&quot;HUMAN_&quot;</span>, <span class="dt">controls =</span> <span class="st">&quot;MOUSE_&quot;</span>)</a>
<a class="sourceLine" id="cb634-14" data-line-number="14"></a>
<a class="sourceLine" id="cb634-15" data-line-number="15"><span class="co"># Load in the ADT UMI matrix</span></a>
<a class="sourceLine" id="cb634-16" data-line-number="16">cbmc.adt &lt;-<span class="st"> </span><span class="kw">as.sparse</span>(<span class="kw">read.csv</span>(<span class="kw">paste0</span>(mydir, <span class="st">&quot;GSE100866_CBMC_8K_13AB_10X-ADT_umi.csv.gz&quot;</span>), <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">row.names =</span> <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb634-17" data-line-number="17"></a>
<a class="sourceLine" id="cb634-18" data-line-number="18"><span class="co"># When adding multimodal data to Seurat, it&#39;s okay to have duplicate feature names. Each set of</span></a>
<a class="sourceLine" id="cb634-19" data-line-number="19"><span class="co"># modal data (eg. RNA, ADT, etc.) is stored in its own Assay object.  One of these Assay objects</span></a>
<a class="sourceLine" id="cb634-20" data-line-number="20"><span class="co"># is called the &#39;default assay&#39;, meaning it&#39;s used for all analyses and visualization.  To pull</span></a>
<a class="sourceLine" id="cb634-21" data-line-number="21"><span class="co"># data from an assay that isn&#39;t the default, you can specify a key that&#39;s linked to an assay for</span></a>
<a class="sourceLine" id="cb634-22" data-line-number="22"><span class="co"># feature pulling.  To see all keys for all objects, use the Key function.  </span></a>
<a class="sourceLine" id="cb634-23" data-line-number="23"></a>
<a class="sourceLine" id="cb634-24" data-line-number="24"><span class="co"># Lastly, we observed poor enrichments for CCR5, CCR7, and CD10 - and therefore </span></a>
<a class="sourceLine" id="cb634-25" data-line-number="25"><span class="co"># remove them from the matrix (optional)</span></a>
<a class="sourceLine" id="cb634-26" data-line-number="26">cbmc.adt &lt;-<span class="st"> </span>cbmc.adt[<span class="kw">setdiff</span>(<span class="kw">rownames</span>(<span class="dt">x =</span> cbmc.adt), <span class="kw">c</span>(<span class="st">&quot;CCR5&quot;</span>, <span class="st">&quot;CCR7&quot;</span>, <span class="st">&quot;CD10&quot;</span>)), ]</a>
<a class="sourceLine" id="cb634-27" data-line-number="27"></a>
<a class="sourceLine" id="cb634-28" data-line-number="28"><span class="co"># Look at structure of ADT matrix.</span></a>
<a class="sourceLine" id="cb634-29" data-line-number="29">cbmc.adt[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</a>
<a class="sourceLine" id="cb634-30" data-line-number="30"></a>
<a class="sourceLine" id="cb634-31" data-line-number="31"><span class="co"># What fraction of cells in the ADT and RNA matrix overlap?</span></a>
<a class="sourceLine" id="cb634-32" data-line-number="32"><span class="kw">length</span>(<span class="kw">intersect</span>(<span class="kw">colnames</span>(cbmc.rna), <span class="kw">colnames</span>(cbmc.adt))) <span class="op">/</span><span class="st"> </span><span class="kw">length</span>(<span class="kw">union</span>(<span class="kw">colnames</span>(cbmc.rna), <span class="kw">colnames</span>(cbmc.adt)))</a>
<a class="sourceLine" id="cb634-33" data-line-number="33"></a>
<a class="sourceLine" id="cb634-34" data-line-number="34"><span class="co"># Save current progress.</span></a>
<a class="sourceLine" id="cb634-35" data-line-number="35"><span class="kw">save</span>(cbmc.rna, cbmc.adt, <span class="dt">file =</span> Rda.quickload.path)</a></code></pre></div>
</div>
<div id="setup-a-seurat-object-and-cluster-cells-based-on-rna-expression" class="section level2">
<h2><span class="header-section-number">16.3</span> Setup a Seurat object, and cluster cells based on RNA expression</h2>
<p>The steps below represent a quick clustering of the PBMCs based on the scRNA-seq data. For more detail on individual steps or more advanced options, see our PBMC clustering guided tutorial here</p>
<div class="sourceCode" id="cb635"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb635-1" data-line-number="1"><span class="kw">load</span>(Rda.quickload.path)</a>
<a class="sourceLine" id="cb635-2" data-line-number="2">cbmc &lt;-<span class="st"> </span><span class="kw">CreateSeuratObject</span>(<span class="dt">counts =</span> cbmc.rna)</a></code></pre></div>
<pre><code>## Warning: Feature names cannot have underscores (&#39;_&#39;), replacing with dashes
## (&#39;-&#39;)</code></pre>
<div class="sourceCode" id="cb637"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb637-1" data-line-number="1"><span class="co"># This code sub-samples the data in order to speed up calculations and not use too much memory.</span></a>
<a class="sourceLine" id="cb637-2" data-line-number="2"><span class="co"># Idents(cbmc) &lt;- &quot;orig.ident&quot;</span></a>
<a class="sourceLine" id="cb637-3" data-line-number="3"><span class="co"># cbmc &lt;- subset(cbmc, downsample = 2000, seed = 1)</span></a>
<a class="sourceLine" id="cb637-4" data-line-number="4"><span class="co"># cbmc.adt &lt;- cbmc.adt[, colnames(cbmc)]</span></a>
<a class="sourceLine" id="cb637-5" data-line-number="5"></a>
<a class="sourceLine" id="cb637-6" data-line-number="6"><span class="co"># standard log-normalization</span></a>
<a class="sourceLine" id="cb637-7" data-line-number="7">cbmc &lt;-<span class="st"> </span><span class="kw">NormalizeData</span>(cbmc)</a>
<a class="sourceLine" id="cb637-8" data-line-number="8"></a>
<a class="sourceLine" id="cb637-9" data-line-number="9"><span class="co"># choose ~1k variable features</span></a>
<a class="sourceLine" id="cb637-10" data-line-number="10">cbmc &lt;-<span class="st"> </span><span class="kw">FindVariableFeatures</span>(cbmc)</a>
<a class="sourceLine" id="cb637-11" data-line-number="11"></a>
<a class="sourceLine" id="cb637-12" data-line-number="12"><span class="co"># standard scaling (no regression)</span></a>
<a class="sourceLine" id="cb637-13" data-line-number="13">cbmc &lt;-<span class="st"> </span><span class="kw">ScaleData</span>(cbmc)</a></code></pre></div>
<pre><code>## Centering and scaling data matrix</code></pre>
<div class="sourceCode" id="cb639"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb639-1" data-line-number="1"><span class="co"># Run PCA, select PCs for tSNE visualization and graph-based clustering</span></a>
<a class="sourceLine" id="cb639-2" data-line-number="2">cbmc &lt;-<span class="st"> </span><span class="kw">RunPCA</span>(cbmc, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb639-3" data-line-number="3"><span class="kw">ElbowPlot</span>(cbmc, <span class="dt">ndims =</span> <span class="dv">50</span>)</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/cbmc_obj-1.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb640"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb640-1" data-line-number="1"><span class="co"># Cluster the cells using the first 25 principal components.</span></a>
<a class="sourceLine" id="cb640-2" data-line-number="2">cbmc &lt;-<span class="st"> </span><span class="kw">FindNeighbors</span>(cbmc, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">25</span>)</a></code></pre></div>
<pre><code>## Computing nearest neighbor graph</code></pre>
<pre><code>## Computing SNN</code></pre>
<div class="sourceCode" id="cb643"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb643-1" data-line-number="1">cbmc &lt;-<span class="st"> </span><span class="kw">FindClusters</span>(cbmc, <span class="dt">resolution =</span> <span class="fl">0.8</span>)</a></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 8617
## Number of edges: 347548
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8592
## Number of communities: 19
## Elapsed time: 1 seconds</code></pre>
<div class="sourceCode" id="cb645"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb645-1" data-line-number="1">cbmc &lt;-<span class="st"> </span><span class="kw">RunTSNE</span>(cbmc, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">25</span>, <span class="dt">method =</span> <span class="st">&quot;FIt-SNE&quot;</span>)</a>
<a class="sourceLine" id="cb645-2" data-line-number="2"></a>
<a class="sourceLine" id="cb645-3" data-line-number="3"><span class="co"># Find the markers that define each cluster, and use these to annotate the</span></a>
<a class="sourceLine" id="cb645-4" data-line-number="4"><span class="co"># clusters, we use max.cells.per.ident to speed up the process</span></a>
<a class="sourceLine" id="cb645-5" data-line-number="5">cbmc.rna.markers &lt;-<span class="st"> </span><span class="kw">FindAllMarkers</span>(cbmc, <span class="dt">max.cells.per.ident =</span> <span class="dv">100</span>, <span class="dt">logfc.threshold =</span> <span class="kw">log</span>(<span class="dv">2</span>), <span class="dt">only.pos =</span> <span class="ot">TRUE</span>, <span class="dt">min.diff.pct =</span> <span class="fl">0.3</span>)</a></code></pre></div>
<pre><code>## Calculating cluster 0</code></pre>
<pre><code>## Calculating cluster 1</code></pre>
<pre><code>## Calculating cluster 2</code></pre>
<pre><code>## Calculating cluster 3</code></pre>
<pre><code>## Calculating cluster 4</code></pre>
<pre><code>## Calculating cluster 5</code></pre>
<pre><code>## Calculating cluster 6</code></pre>
<pre><code>## Calculating cluster 7</code></pre>
<pre><code>## Calculating cluster 8</code></pre>
<pre><code>## Calculating cluster 9</code></pre>
<pre><code>## Calculating cluster 10</code></pre>
<pre><code>## Calculating cluster 11</code></pre>
<pre><code>## Calculating cluster 12</code></pre>
<pre><code>## Calculating cluster 13</code></pre>
<pre><code>## Calculating cluster 14</code></pre>
<pre><code>## Calculating cluster 15</code></pre>
<pre><code>## Calculating cluster 16</code></pre>
<pre><code>## Calculating cluster 17</code></pre>
<pre><code>## Calculating cluster 18</code></pre>
<div class="sourceCode" id="cb665"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb665-1" data-line-number="1"><span class="co"># Which cluster consists of mouse cells?</span></a>
<a class="sourceLine" id="cb665-2" data-line-number="2">cbmc.rna.markers <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(cluster <span class="op">==</span><span class="st"> </span><span class="dv">5</span>)</a></code></pre></div>
<pre><code>##            p_val avg_logFC pct.1 pct.2    p_val_adj cluster            gene
## 1   2.589983e-38 3.0688532 1.000 0.082 5.309724e-34       5     MOUSE-Shfm1
## 2   3.343974e-38 3.3214918 1.000 0.065 6.855481e-34       5     MOUSE-Eef1g
## 3   4.576942e-38 3.0566728 1.000 0.087 9.383188e-34       5      MOUSE-Rtn4
## 4   5.060526e-38 3.3736882 1.000 0.092 1.037459e-33       5      MOUSE-Npm1
## 5   5.853288e-38 3.1075365 1.000 0.075 1.199983e-33       5   MOUSE-S100a11
## 6   7.309965e-38 3.1210730 1.000 0.137 1.498616e-33       5       MOUSE-Bgn
## 7   7.854611e-38 3.1823668 1.000 0.048 1.610274e-33       5  MOUSE-Atp6v1g1
## 8   8.078132e-38 3.1757854 1.000 0.079 1.656098e-33       5    MOUSE-Serbp1
## 9   1.205463e-37 3.5136647 1.000 0.081 2.471320e-33       5   MOUSE-Gm10260
## 10  1.309584e-37 3.0016168 1.000 0.065 2.684778e-33       5     MOUSE-Rpl22
## 11  1.358667e-37 3.2116968 1.000 0.081 2.785404e-33       5    MOUSE-Ndufa4
## 12  1.451626e-37 3.4523548 1.000 0.121 2.975979e-33       5     MOUSE-Rpl32
## 13  1.548045e-37 3.0793534 1.000 0.055 3.173646e-33       5    MOUSE-Rps27l
## 14  1.891767e-37 3.2763202 1.000 0.073 3.878312e-33       5     MOUSE-Rplp2
## 15  1.918593e-37 3.2224149 1.000 0.131 3.933307e-33       5   MOUSE-Slc25a4
## 16  2.230479e-37 3.4298554 1.000 0.137 4.572705e-33       5      MOUSE-Rps9
## 17  2.262278e-37 3.2949243 1.000 0.144 4.637896e-33       5  MOUSE-Hsp90ab1
## 18  2.436807e-37 3.0609232 1.000 0.128 4.995698e-33       5      MOUSE-Nme1
## 19  2.576793e-37 3.3457217 1.000 0.096 5.282684e-33       5    MOUSE-Tmsb10
## 20  2.599760e-37 3.1809764 1.000 0.149 5.329768e-33       5     MOUSE-Tubb5
## 21  2.835723e-37 2.9497869 1.000 0.067 5.813516e-33       5      MOUSE-Calr
## 22  2.845096e-37 3.5141515 1.000 0.085 5.832732e-33       5      MOUSE-Ybx1
## 23  2.861704e-37 3.0508286 1.000 0.098 5.866779e-33       5    MOUSE-Arpc1b
## 24  2.943740e-37 3.0414266 1.000 0.113 6.034962e-33       5      MOUSE-Lmna
## 25  3.141150e-37 2.8791432 1.000 0.104 6.439672e-33       5      MOUSE-Ctsl
## 26  3.205286e-37 2.9776496 1.000 0.103 6.571158e-33       5      MOUSE-Gas5
## 27  3.248892e-37 2.9783581 1.000 0.141 6.660554e-33       5     MOUSE-Atp5b
## 28  3.248892e-37 2.9437600 1.000 0.102 6.660554e-33       5   MOUSE-Slc25a3
## 29  3.373363e-37 3.3672873 1.000 0.095 6.915732e-33       5     MOUSE-Hmga2
## 30  3.486852e-37 3.1325081 1.000 0.045 7.148395e-33       5      MOUSE-Eno1
## 31  3.709120e-37 3.3965827 1.000 0.113 7.604067e-33       5     MOUSE-H3f3b
## 32  3.731361e-37 3.3445044 1.000 0.109 7.649664e-33       5     MOUSE-Eif5a
## 33  3.766307e-37 3.2020434 0.998 0.060 7.721307e-33       5    MOUSE-mt-Nd2
## 34  4.115629e-37 3.1411789 1.000 0.099 8.437452e-33       5       MOUSE-Pkm
## 35  4.300905e-37 2.8335714 1.000 0.064 8.817285e-33       5     MOUSE-Rps26
## 36  4.511666e-37 3.2189211 1.000 0.077 9.249367e-33       5     MOUSE-Rps24
## 37  4.661889e-37 3.0882543 1.000 0.107 9.557339e-33       5      MOUSE-Ldha
## 38  4.914987e-37 3.2250975 1.000 0.103 1.007622e-32       5     MOUSE-Calm1
## 39  5.008380e-37 3.7123031 1.000 0.157 1.026768e-32       5     MOUSE-Rplp0
## 40  5.175830e-37 2.9014521 1.000 0.089 1.061097e-32       5       MOUSE-Esd
## 41  5.244907e-37 3.4640366 1.000 0.094 1.075258e-32       5     MOUSE-Rpl26
## 42  5.524036e-37 3.4322609 1.000 0.134 1.132483e-32       5     MOUSE-Rpl14
## 43  5.604675e-37 3.1056839 1.000 0.085 1.149014e-32       5    MOUSE-Eef1b2
## 44  5.709515e-37 3.2070806 1.000 0.111 1.170508e-32       5      MOUSE-Eef2
## 45  6.051582e-37 2.8562423 1.000 0.099 1.240635e-32       5    MOUSE-Cox6b1
## 46  6.276767e-37 2.9754960 1.000 0.076 1.286800e-32       5     MOUSE-Arpc2
## 47  6.294522e-37 2.9709200 1.000 0.102 1.290440e-32       5    MOUSE-Ppp1ca
## 48  6.603459e-37 3.4895906 1.000 0.087 1.353775e-32       5      MOUSE-Cfl1
## 49  6.726146e-37 3.1145355 0.998 0.048 1.378927e-32       5   MOUSE-Slc25a5
## 50  7.176881e-37 3.5936267 1.000 0.126 1.471332e-32       5     MOUSE-Rps11
## 51  8.111934e-37 3.1933176 1.000 0.093 1.663028e-32       5     MOUSE-Rps19
## 52  8.378419e-37 2.9673217 1.000 0.109 1.717660e-32       5 MOUSE-Tnfrsf12a
## 53  8.654506e-37 3.5547613 1.000 0.154 1.774260e-32       5      MOUSE-Pfn1
## 54  8.654506e-37 3.2416146 1.000 0.136 1.774260e-32       5      MOUSE-Txn1
## 55  9.162208e-37 3.2749753 1.000 0.077 1.878344e-32       5      MOUSE-Naca
## 56  9.605768e-37 3.3640993 1.000 0.118 1.969279e-32       5     MOUSE-Rplp1
## 57  9.781641e-37 3.0361351 1.000 0.098 2.005334e-32       5    MOUSE-Tm4sf1
## 58  1.121385e-36 3.2047609 1.000 0.112 2.298952e-32       5    MOUSE-Ranbp1
## 59  1.238798e-36 2.9237221 1.000 0.122 2.539659e-32       5     MOUSE-Anxa2
## 60  1.374845e-36 3.0152863 1.000 0.125 2.818570e-32       5       MOUSE-B2m
## 61  1.486569e-36 3.0540241 1.000 0.164 3.047615e-32       5     MOUSE-Hint1
## 62  1.709022e-36 3.1125583 1.000 0.165 3.503666e-32       5   MOUSE-S100a10
## 63  2.391987e-36 3.2185589 1.000 0.110 4.903812e-32       5     MOUSE-Cox8a
## 64  2.546516e-36 3.4041833 1.000 0.197 5.220612e-32       5      MOUSE-Rpl4
## 65  2.694724e-36 3.4068076 1.000 0.113 5.524454e-32       5     MOUSE-Rpl23
## 66  2.782618e-36 3.0687226 1.000 0.167 5.704646e-32       5    MOUSE-Gnb2l1
## 67  2.803277e-36 2.9466599 1.000 0.071 5.746997e-32       5     MOUSE-Cald1
## 68  3.060359e-36 3.0983460 1.000 0.120 6.274042e-32       5       MOUSE-Mt1
## 69  3.061720e-36 3.5504399 1.000 0.085 6.276832e-32       5    MOUSE-Rps3a1
## 70  3.089689e-36 3.0538578 1.000 0.110 6.334172e-32       5      MOUSE-Cdk4
## 71  3.162101e-36 2.9919371 1.000 0.107 6.482624e-32       5     MOUSE-Gsto1
## 72  3.194002e-36 3.2865393 1.000 0.187 6.548023e-32       5      MOUSE-Dlk1
## 73  3.483292e-36 3.1971727 1.000 0.123 7.141097e-32       5     MOUSE-Prdx1
## 74  3.818679e-36 3.2985227 1.000 0.224 7.828673e-32       5      MOUSE-Rps3
## 75  4.225164e-36 3.3243282 1.000 0.209 8.662009e-32       5       MOUSE-Vim
## 76  4.389534e-36 3.5227154 1.000 0.236 8.998984e-32       5    MOUSE-Rpl13a
## 77  4.799768e-36 1.9191170 0.979 0.024 9.840005e-32       5            MYL3
## 78  4.817738e-36 3.2396729 1.000 0.146 9.876844e-32       5     MOUSE-Anxa1
## 79  5.263533e-36 3.4278589 1.000 0.126 1.079077e-31       5     MOUSE-Rps15
## 80  5.521946e-36 2.8645123 1.000 0.114 1.132054e-31       5      MOUSE-Cstb
## 81  5.736456e-36 3.1156562 1.000 0.148 1.176031e-31       5    MOUSE-Myl12a
## 82  7.180006e-36 2.9175274 1.000 0.183 1.471973e-31       5    MOUSE-Cox4i1
## 83  8.161440e-36 3.0344876 1.000 0.108 1.673177e-31       5     MOUSE-Lamp1
## 84  8.360053e-36 3.2601292 1.000 0.205 1.713894e-31       5      MOUSE-Actb
## 85  9.018525e-36 3.2825215 1.000 0.269 1.848888e-31       5   MOUSE-mt-Cytb
## 86  1.661694e-35 3.3308664 1.000 0.295 3.406639e-31       5      MOUSE-Rpl8
## 87  1.755974e-35 3.1615591 1.000 0.124 3.599922e-31       5      MOUSE-Gnas
## 88  1.844947e-35 3.2151653 0.998 0.145 3.782325e-31       5    MOUSE-mt-Nd4
## 89  2.160878e-35 3.5516310 1.000 0.144 4.430015e-31       5      MOUSE-Cd63
## 90  2.272852e-35 3.3854796 1.000 0.178 4.659574e-31       5     MOUSE-Rps14
## 91  2.334724e-35 2.9695641 1.000 0.171 4.786418e-31       5       MOUSE-Bsg
## 92  3.138487e-35 3.3949580 1.000 0.378 6.434213e-31       5      MOUSE-Rps5
## 93  4.384772e-35 3.1478765 1.000 0.237 8.989221e-31       5    MOUSE-Lgals3
## 94  7.608819e-35 3.5809744 0.995 0.238 1.559884e-30       5    MOUSE-Malat1
## 95  1.535971e-34 3.1143600 1.000 0.331 3.148895e-30       5    MOUSE-S100a4
## 96  3.969846e-34 3.2843640 1.000 0.549 8.138582e-30       5    MOUSE-S100a6
## 97  7.554328e-34 3.3319006 1.000 0.355 1.548713e-29       5    MOUSE-mt-Nd1
## 98  8.353188e-34 1.4166049 0.984 0.093 1.712487e-29       5          DYNLL2
## 99  1.504607e-33 3.3243385 1.000 0.446 3.084595e-29       5      MOUSE-Fth1
## 100 2.099049e-33 3.1753011 1.000 0.400 4.303261e-29       5    MOUSE-Tmsb4x
## 101 7.464377e-32 2.6376435 0.963 0.098 1.530272e-27       5     MOUSE-Acta2
## 102 3.566992e-29 1.3863978 0.972 0.062 7.312689e-25       5   CTD-2090I13.1
## 103 1.946493e-23 0.9060487 0.810 0.013 3.990505e-19       5       HIST2H2AB</code></pre>
<div class="sourceCode" id="cb667"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb667-1" data-line-number="1">cbmc.rna.markers <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(cluster <span class="op">==</span><span class="st"> </span><span class="dv">13</span>)</a></code></pre></div>
<pre><code>##            p_val avg_logFC pct.1 pct.2    p_val_adj cluster            gene
## 1   5.694937e-32  2.261027 1.000 0.146 1.167519e-27      13    MOUSE-Ranbp1
## 2   6.122860e-32  2.499240 0.991 0.159 1.255248e-27      13     MOUSE-Rps15
## 3   7.231469e-32  2.668083 0.991 0.152 1.482523e-27      13     MOUSE-Rplp1
## 4   1.174562e-31  2.778360 1.000 0.357 2.407970e-27      13    MOUSE-S100a4
## 5   1.209176e-31  2.730176 1.000 0.423 2.478932e-27      13    MOUSE-Tmsb4x
## 6   1.797643e-31  2.408277 0.991 0.156 3.685348e-27      13     MOUSE-Anxa2
## 7   1.936705e-31  2.542341 0.972 0.109 3.970440e-27      13     MOUSE-Rplp2
## 8   2.159518e-31  2.481118 0.991 0.145 4.427228e-27      13     MOUSE-Cox8a
## 9   2.985211e-31  2.723078 0.981 0.157 6.119982e-27      13     MOUSE-Prdx1
## 10  3.172531e-31  2.901085 1.000 0.566 6.504006e-27      13    MOUSE-S100a6
## 11  7.106704e-31  3.160532 0.991 0.236 1.456945e-26      13      MOUSE-Actb
## 12  8.246008e-31  2.495074 0.981 0.147 1.690514e-26      13     MOUSE-Rpl23
## 13  9.223015e-31  2.471669 0.981 0.117 1.890810e-26      13   MOUSE-Gm10260
## 14  9.371755e-31  2.452599 0.981 0.160 1.921303e-26      13     MOUSE-Rps11
## 15  9.779046e-31  2.620856 0.953 0.113 2.004802e-26      13     MOUSE-Rps24
## 16  1.251003e-30  2.465479 0.972 0.129 2.564682e-26      13     MOUSE-Rpl26
## 17  1.756273e-30  2.535067 0.953 0.101 3.600535e-26      13     MOUSE-Rpl22
## 18  2.034570e-30  2.482089 0.991 0.155 4.171071e-26      13     MOUSE-Rpl32
## 19  2.379570e-30  2.559327 0.972 0.167 4.878356e-26      13     MOUSE-Rpl14
## 20  2.548928e-30  2.753512 0.943 0.111 5.225558e-26      13   MOUSE-S100a11
## 21  2.681390e-30  2.301315 0.981 0.197 5.497117e-26      13   MOUSE-S100a10
## 22  3.546009e-30  2.573062 0.991 0.186 7.269673e-26      13      MOUSE-Pfn1
## 23  3.734094e-30  2.449508 0.991 0.209 7.655267e-26      13     MOUSE-Rps14
## 24  4.606642e-30  2.482054 0.981 0.266 9.444077e-26      13    MOUSE-Rpl13a
## 25  5.136327e-30  2.335140 0.972 0.121 1.052998e-25      13    MOUSE-Rps3a1
## 26  5.280799e-30  2.215450 0.981 0.197 1.082617e-25      13     MOUSE-Hint1
## 27  5.595216e-30  2.314418 0.953 0.113 1.147075e-25      13      MOUSE-Naca
## 28  7.937246e-30  2.479286 0.953 0.135 1.627215e-25      13    MOUSE-Cox6b1
## 29  1.203382e-29  2.444119 0.953 0.169 2.467054e-25      13      MOUSE-Txn1
## 30  1.260502e-29  2.399652 1.000 0.189 2.584156e-25      13     MOUSE-Rplp0
## 31  1.296573e-29  2.369831 0.962 0.149 2.658103e-25      13      MOUSE-Cstb
## 32  1.409398e-29  2.398371 0.991 0.323 2.889407e-25      13      MOUSE-Rpl8
## 33  1.501281e-29  2.156191 0.953 0.133 3.077776e-25      13    MOUSE-Arpc1b
## 34  1.713709e-29  2.675714 0.925 0.101 3.513275e-25      13     MOUSE-Rps26
## 35  1.713709e-29  2.368067 0.953 0.117 3.513275e-25      13    MOUSE-Ndufa4
## 36  1.781970e-29  2.380462 0.953 0.102 3.653216e-25      13     MOUSE-Eef1g
## 37  2.110199e-29  2.402753 0.991 0.402 4.326119e-25      13      MOUSE-Rps5
## 38  2.136048e-29  2.620474 0.962 0.177 4.379112e-25      13  MOUSE-Hsp90ab1
## 39  2.270318e-29  2.698158 0.962 0.131 4.654380e-25      13    MOUSE-Tmsb10
## 40  2.298364e-29  2.360640 0.981 0.254 4.711877e-25      13      MOUSE-Rps3
## 41  2.774051e-29  2.405205 0.953 0.120 5.687082e-25      13    MOUSE-Eef1b2
## 42  2.971166e-29  2.403258 0.962 0.146 6.091187e-25      13      MOUSE-Eef2
## 43  3.217869e-29  2.255929 0.972 0.143 6.596953e-25      13     MOUSE-Eif5a
## 44  6.052367e-29  2.390698 0.962 0.129 1.240796e-24      13     MOUSE-Rps19
## 45  6.446107e-29  2.381902 0.943 0.118 1.321516e-24      13     MOUSE-Shfm1
## 46  8.392692e-29  2.522999 0.991 0.468 1.720586e-24      13      MOUSE-Fth1
## 47  1.039199e-28  2.490092 0.953 0.181 2.130462e-24      13    MOUSE-Myl12a
## 48  1.059617e-28  2.363141 0.962 0.174 2.172321e-24      13     MOUSE-Atp5b
## 49  1.136848e-28  2.420211 0.962 0.170 2.330652e-24      13      MOUSE-Rps9
## 50  1.677680e-28  2.231629 0.953 0.121 3.439412e-24      13      MOUSE-Ybx1
## 51  1.797144e-28  2.131666 0.953 0.122 3.684326e-24      13      MOUSE-Cfl1
## 52  2.217540e-28  3.140499 0.972 0.297 4.546179e-24      13   MOUSE-mt-Cytb
## 53  2.236480e-28  2.200847 0.953 0.127 4.585009e-24      13      MOUSE-Npm1
## 54  3.684323e-28  2.280569 0.972 0.240 7.553230e-24      13       MOUSE-Vim
## 55  4.076600e-28  2.343247 0.953 0.134 8.357438e-24      13       MOUSE-Pkm
## 56  4.636754e-28  2.218386 0.981 0.228 9.505810e-24      13      MOUSE-Rpl4
## 57  6.749065e-28  2.132095 0.962 0.200 1.383626e-23      13    MOUSE-Gnb2l1
## 58  7.778351e-28  2.210371 0.934 0.163 1.594640e-23      13      MOUSE-Nme1
## 59  8.741573e-28  2.341123 0.943 0.215 1.792110e-23      13    MOUSE-Cox4i1
## 60  2.222504e-27  2.114141 0.962 0.138 4.556355e-23      13     MOUSE-Calm1
## 61  2.860745e-27  2.205576 0.962 0.158 5.864813e-23      13      MOUSE-Gnas
## 62  4.016601e-27  2.146927 0.962 0.267 8.234433e-23      13    MOUSE-Lgals3
## 63  4.338376e-27  3.106415 0.925 0.179 8.894105e-23      13    MOUSE-mt-Nd4
## 64  4.581561e-27  2.038114 0.896 0.086 9.392658e-23      13   MOUSE-Slc25a5
## 65  5.536735e-27  2.718182 0.915 0.097 1.135086e-22      13    MOUSE-mt-Nd2
## 66  1.000179e-26  2.182786 0.915 0.137 2.050467e-22      13   MOUSE-Slc25a3
## 67  2.126742e-26  3.054986 0.953 0.380 4.360033e-22      13    MOUSE-mt-Nd1
## 68  2.555881e-26  2.306473 0.915 0.160 5.239812e-22      13       MOUSE-B2m
## 69  2.753783e-26  2.265517 0.915 0.108 5.645531e-22      13     MOUSE-Cald1
## 70  5.021616e-26  2.158415 0.887 0.083 1.029482e-21      13      MOUSE-Eno1
## 71  6.477556e-26  1.876609 0.934 0.145 1.327964e-21      13      MOUSE-Cdk4
## 72  9.132195e-26  1.929906 0.925 0.148 1.872191e-21      13      MOUSE-Lmna
## 73  1.443189e-25  2.028266 0.896 0.142 2.958682e-21      13     MOUSE-Gsto1
## 74  2.187135e-25  2.573234 0.906 0.139 4.483845e-21      13      MOUSE-Gas5
## 75  9.676047e-25  2.057407 0.896 0.125 1.983686e-20      13       MOUSE-Esd
## 76  1.163687e-24  1.963349 0.915 0.180 2.385675e-20      13     MOUSE-Anxa1
## 77  1.249671e-24  1.814887 0.934 0.165 2.561951e-20      13   MOUSE-Slc25a4
## 78  5.351638e-24  2.017758 0.906 0.178 1.097139e-19      13      MOUSE-Cd63
## 79  9.051252e-24  1.716038 0.858 0.116 1.855597e-19      13    MOUSE-Serbp1
## 80  1.035274e-23  1.886684 0.877 0.143 2.122415e-19      13      MOUSE-Ldha
## 81  6.074209e-23  2.021360 0.858 0.093 1.245274e-18      13    MOUSE-Rps27l
## 82  1.256351e-22  1.796896 0.877 0.113 2.575645e-18      13     MOUSE-Arpc2
## 83  7.286297e-22  1.796396 0.821 0.087 1.493764e-17      13  MOUSE-Atp6v1g1
## 84  4.131820e-21  2.009562 0.849 0.156 8.470644e-17      13       MOUSE-Mt1
## 85  4.918820e-20  1.713497 0.830 0.139 1.008407e-15      13    MOUSE-Ppp1ca
## 86  5.324099e-20  2.763313 0.755 0.134 1.091493e-15      13     MOUSE-Acta2
## 87  4.312371e-19  1.870316 0.802 0.173 8.840792e-15      13       MOUSE-Bgn
## 88  8.315943e-19  1.833178 0.811 0.221 1.704851e-14      13      MOUSE-Dlk1
## 89  3.248015e-18  1.842319 0.764 0.145 6.658756e-14      13     MOUSE-Lamp1
## 90  4.798035e-18  1.818716 0.821 0.205 9.836452e-14      13       MOUSE-Bsg
## 91  5.757352e-18  2.126146 0.792 0.184 1.180315e-13      13     MOUSE-Tubb5
## 92  1.243481e-16  1.983483 0.717 0.142 2.549260e-12      13      MOUSE-Ctsl
## 93  2.451551e-16  1.886574 0.717 0.106 5.025925e-12      13      MOUSE-Calr
## 94  7.392626e-14  1.794081 0.585 0.065 1.515562e-09      13            MYL3
## 95  7.932037e-13  1.738518 0.632 0.135 1.626147e-08      13     MOUSE-Hmga2
## 96  4.127263e-12  1.526832 0.660 0.137 8.461302e-08      13    MOUSE-Tm4sf1
## 97  1.305564e-11  1.661696 0.623 0.127 2.676536e-07      13      MOUSE-Rtn4
## 98  5.333310e-09  1.413978 0.585 0.152 1.093382e-04      13     MOUSE-H3f3b
## 99  7.207313e-08  1.132159 0.434 0.103 1.477571e-03      13   CTD-2090I13.1
## 100 9.026752e-07  1.263586 0.509 0.149 1.850574e-02      13 MOUSE-Tnfrsf12a</code></pre>
<div class="sourceCode" id="cb669"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb669-1" data-line-number="1"><span class="co"># Note, for simplicity we are merging two CD14+ Monocyte clusters (that differ in expression of</span></a>
<a class="sourceLine" id="cb669-2" data-line-number="2"><span class="co"># HLA-DR genes) and NK clusters (that differ in cell cycle stage)</span></a>
<a class="sourceLine" id="cb669-3" data-line-number="3">new.cluster.ids &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Memory CD4 T&quot;</span>, <span class="st">&quot;CD14+ Mono&quot;</span>, <span class="st">&quot;Naive CD4 T&quot;</span>, <span class="st">&quot;NK&quot;</span>, <span class="st">&quot;CD14+ Mono&quot;</span>, <span class="st">&quot;Mouse&quot;</span>, <span class="st">&quot;B&quot;</span>, <span class="st">&quot;CD8 T&quot;</span>, <span class="st">&quot;CD16+ Mono&quot;</span>, <span class="st">&quot;T/Mono doublets&quot;</span>, <span class="st">&quot;NK&quot;</span>, <span class="st">&quot;CD34+&quot;</span>, <span class="st">&quot;Multiplets&quot;</span>, <span class="st">&quot;Mouse&quot;</span>, <span class="st">&quot;Eryth&quot;</span>, <span class="st">&quot;Mk&quot;</span>, <span class="st">&quot;Mouse&quot;</span>, <span class="st">&quot;DC&quot;</span>, <span class="st">&quot;pDCs&quot;</span>)</a>
<a class="sourceLine" id="cb669-4" data-line-number="4"><span class="kw">names</span>(new.cluster.ids) &lt;-<span class="st"> </span><span class="kw">levels</span>(cbmc)</a>
<a class="sourceLine" id="cb669-5" data-line-number="5">cbmc &lt;-<span class="st"> </span><span class="kw">RenameIdents</span>(cbmc, new.cluster.ids)</a>
<a class="sourceLine" id="cb669-6" data-line-number="6"></a>
<a class="sourceLine" id="cb669-7" data-line-number="7"><span class="co"># Visualize clustering based on RNA.</span></a>
<a class="sourceLine" id="cb669-8" data-line-number="8"><span class="kw">DimPlot</span>(cbmc, <span class="dt">label =</span> <span class="ot">TRUE</span>, <span class="dt">reduction =</span> <span class="st">&quot;tsne&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">NoLegend</span>()</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/cbmc_obj-2.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb670"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb670-1" data-line-number="1"><span class="co"># Save current progress.</span></a>
<a class="sourceLine" id="cb670-2" data-line-number="2"><span class="kw">save</span>(cbmc, cbmc.rna.markers, cbmc.adt, <span class="dt">file =</span> Rda.RNA.path)</a>
<a class="sourceLine" id="cb670-3" data-line-number="3"><span class="co"># To load the data, run the following command.</span></a>
<a class="sourceLine" id="cb670-4" data-line-number="4"><span class="co"># load(Rda.RNA.path)</span></a></code></pre></div>
</div>
<div id="add-the-protein-expression-levels-to-the-seurat-object" class="section level2">
<h2><span class="header-section-number">16.4</span> Add the protein expression levels to the Seurat object</h2>
<p>Seurat v3.0 allows you to store information from multiple assays in the same object, as long as the data is multi-modal (collected on the same set of cells). You can use the <code>SetAssayData</code> and <code>GetAssayData</code> accessor functions to add and fetch data from additional assays.</p>
<div class="sourceCode" id="cb671"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb671-1" data-line-number="1"><span class="co"># We will define an ADT assay, and store raw counts for it</span></a>
<a class="sourceLine" id="cb671-2" data-line-number="2"></a>
<a class="sourceLine" id="cb671-3" data-line-number="3"><span class="co"># If you are interested in how these data are internally stored, you can check out the Assay</span></a>
<a class="sourceLine" id="cb671-4" data-line-number="4"><span class="co"># class, which is defined in objects.R; note that all single-cell expression data, including RNA</span></a>
<a class="sourceLine" id="cb671-5" data-line-number="5"><span class="co"># data, are still stored in Assay objects, and can also be accessed using GetAssayData</span></a>
<a class="sourceLine" id="cb671-6" data-line-number="6">cbmc[[<span class="st">&quot;ADT&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">CreateAssayObject</span>(<span class="dt">counts =</span> cbmc.adt)</a>
<a class="sourceLine" id="cb671-7" data-line-number="7"></a>
<a class="sourceLine" id="cb671-8" data-line-number="8"><span class="kw">GetAssayData</span>(cbmc, <span class="dt">slot =</span> <span class="st">&quot;counts&quot;</span>, <span class="dt">assay =</span> <span class="st">&quot;ADT&quot;</span>)[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</a></code></pre></div>
<pre><code>## 3 x 3 sparse Matrix of class &quot;dgCMatrix&quot;
##     CTGTTTACACCGCTAG CTCTACGGTGTGGCTC AGCAGCCAGGCTCATT
## CD3               60               52               89
## CD4               72               49              112
## CD8               76               59               61</code></pre>
<div class="sourceCode" id="cb673"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb673-1" data-line-number="1">cbmc[[<span class="st">&quot;ADT&quot;</span>]]<span class="op">@</span>counts[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</a></code></pre></div>
<pre><code>## 3 x 3 sparse Matrix of class &quot;dgCMatrix&quot;
##     CTGTTTACACCGCTAG CTCTACGGTGTGGCTC AGCAGCCAGGCTCATT
## CD3               60               52               89
## CD4               72               49              112
## CD8               76               59               61</code></pre>
<div class="sourceCode" id="cb675"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb675-1" data-line-number="1"><span class="co"># Now we can repeat the preprocessing (normalization and scaling) steps that we typically run</span></a>
<a class="sourceLine" id="cb675-2" data-line-number="2"><span class="co"># with RNA, but modifying the &#39;assay&#39; argument.  For CITE-seq data, we do not recommend typical</span></a>
<a class="sourceLine" id="cb675-3" data-line-number="3"><span class="co"># LogNormalization. Instead, we use a centered log-ratio (CLR) normalization, computed</span></a>
<a class="sourceLine" id="cb675-4" data-line-number="4"><span class="co"># independently for each feature.  This is a slightly improved procedure from the original</span></a>
<a class="sourceLine" id="cb675-5" data-line-number="5"><span class="co"># publication, and we will release more advanced versions of CITE-seq normalizations soon.</span></a>
<a class="sourceLine" id="cb675-6" data-line-number="6">cbmc &lt;-<span class="st"> </span><span class="kw">NormalizeData</span>(cbmc, <span class="dt">assay =</span> <span class="st">&quot;ADT&quot;</span>, <span class="dt">normalization.method =</span> <span class="st">&quot;CLR&quot;</span>)</a></code></pre></div>
<pre><code>## Normalizing across features</code></pre>
<div class="sourceCode" id="cb677"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb677-1" data-line-number="1">cbmc &lt;-<span class="st"> </span><span class="kw">ScaleData</span>(cbmc, <span class="dt">assay =</span> <span class="st">&quot;ADT&quot;</span>)</a></code></pre></div>
<pre><code>## Centering and scaling data matrix</code></pre>
</div>
<div id="visualize-protein-levels-on-rna-clusters" class="section level2">
<h2><span class="header-section-number">16.5</span> Visualize protein levels on RNA clusters</h2>
<p>You can use the names of any ADT markers, (i.e. “adt_CD4”), in FetchData, FeaturePlot, RidgePlot, FeatureScatter, DoHeatmap, or any other visualization features</p>
<div class="sourceCode" id="cb679"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb679-1" data-line-number="1"><span class="kw">DefaultAssay</span>(cbmc) &lt;-<span class="st"> &quot;RNA&quot;</span></a>
<a class="sourceLine" id="cb679-2" data-line-number="2"></a>
<a class="sourceLine" id="cb679-3" data-line-number="3"><span class="co"># In this plot, protein (ADT) levels are on top, and RNA levels are on the bottom</span></a>
<a class="sourceLine" id="cb679-4" data-line-number="4"><span class="kw">FeaturePlot</span>(cbmc, <span class="dt">features =</span> <span class="kw">c</span>(<span class="st">&quot;adt_CD3&quot;</span>, <span class="st">&quot;adt_CD11c&quot;</span>, <span class="st">&quot;adt_CD8&quot;</span>, <span class="st">&quot;adt_CD16&quot;</span>, <span class="st">&quot;CD3E&quot;</span>, <span class="st">&quot;ITGAX&quot;</span>, <span class="st">&quot;CD8A&quot;</span>, <span class="st">&quot;FCGR3A&quot;</span>), <span class="dt">min.cutoff =</span> <span class="st">&quot;q05&quot;</span>, <span class="dt">max.cutoff =</span> <span class="st">&quot;q95&quot;</span>, <span class="dt">ncol =</span> <span class="dv">4</span>)</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/viz_protein-1.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb680"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb680-1" data-line-number="1"><span class="co"># How do the gene and protein expression levels compare to one another?</span></a>
<a class="sourceLine" id="cb680-2" data-line-number="2"></a>
<a class="sourceLine" id="cb680-3" data-line-number="3"><span class="co"># Compare gene and protein expression levels for the other 6 antibodies.</span></a>
<a class="sourceLine" id="cb680-4" data-line-number="4"><span class="kw">FeaturePlot</span>(cbmc, <span class="dt">features =</span> <span class="kw">c</span>(<span class="st">&quot;adt_CD4&quot;</span>, <span class="st">&quot;adt_CD45RA&quot;</span>, <span class="st">&quot;adt_CD56&quot;</span>, <span class="st">&quot;adt_CD14&quot;</span>, <span class="st">&quot;adt_CD19&quot;</span>, <span class="st">&quot;adt_CD34&quot;</span>, <span class="st">&quot;CD4&quot;</span>, <span class="st">&quot;PTPRC&quot;</span>, <span class="st">&quot;NCAM1&quot;</span>, <span class="st">&quot;CD14&quot;</span>, <span class="st">&quot;CD19&quot;</span>, <span class="st">&quot;CD34&quot;</span>), <span class="dt">min.cutoff =</span> <span class="st">&quot;q05&quot;</span>, <span class="dt">max.cutoff =</span> <span class="st">&quot;q95&quot;</span>, <span class="dt">ncol =</span> <span class="dv">6</span>)</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/viz_protein-2.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb681"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb681-1" data-line-number="1"><span class="co"># Ridge plots are another useful visualization.</span></a>
<a class="sourceLine" id="cb681-2" data-line-number="2"><span class="kw">RidgePlot</span>(cbmc, <span class="dt">features =</span> <span class="kw">c</span>(<span class="st">&quot;adt_CD3&quot;</span>, <span class="st">&quot;adt_CD11c&quot;</span>, <span class="st">&quot;adt_CD8&quot;</span>, <span class="st">&quot;adt_CD16&quot;</span>), <span class="dt">ncol =</span> <span class="dv">2</span>)</a></code></pre></div>
<pre><code>## Picking joint bandwidth of 0.0848</code></pre>
<pre><code>## Picking joint bandwidth of 0.1</code></pre>
<pre><code>## Picking joint bandwidth of 0.142</code></pre>
<pre><code>## Picking joint bandwidth of 0.0862</code></pre>
<p><img src="2020_scWorkshop_files/figure-html/viz_protein-3.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb686"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb686-1" data-line-number="1"><span class="co"># Draw ADT scatter plots (like biaxial plots for FACS). Note that you can even &#39;gate&#39; cells if</span></a>
<a class="sourceLine" id="cb686-2" data-line-number="2"><span class="co"># desired by using HoverLocator and CellSelector</span></a>
<a class="sourceLine" id="cb686-3" data-line-number="3"><span class="kw">FeatureScatter</span>(cbmc, <span class="dt">feature1 =</span> <span class="st">&quot;adt_CD19&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;adt_CD3&quot;</span>)</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/viz_protein-4.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb687"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb687-1" data-line-number="1"><span class="co"># HoverLocator(FeatureScatter(cbmc, feature1 = &quot;adt_CD19&quot;, feature2 = &quot;adt_CD3&quot;))</span></a>
<a class="sourceLine" id="cb687-2" data-line-number="2"><span class="co"># CellSelector(FeatureScatter(cbmc, feature1 = &quot;adt_CD19&quot;, feature2 = &quot;adt_CD3&quot;))</span></a>
<a class="sourceLine" id="cb687-3" data-line-number="3"></a>
<a class="sourceLine" id="cb687-4" data-line-number="4"><span class="co"># View relationship between protein and RNA</span></a>
<a class="sourceLine" id="cb687-5" data-line-number="5"><span class="kw">FeatureScatter</span>(cbmc, <span class="dt">feature1 =</span> <span class="st">&quot;adt_CD3&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;CD3E&quot;</span>)</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/viz_protein-5.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb688"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb688-1" data-line-number="1"><span class="co"># Let&#39;s plot CD4 vs CD8 levels in T cells</span></a>
<a class="sourceLine" id="cb688-2" data-line-number="2">tcells &lt;-<span class="st"> </span><span class="kw">subset</span>(cbmc, <span class="dt">idents =</span> <span class="kw">c</span>(<span class="st">&quot;Naive CD4 T&quot;</span>, <span class="st">&quot;Memory CD4 T&quot;</span>, <span class="st">&quot;CD8 T&quot;</span>))</a>
<a class="sourceLine" id="cb688-3" data-line-number="3"><span class="kw">FeatureScatter</span>(tcells, <span class="dt">feature1 =</span> <span class="st">&quot;adt_CD4&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;adt_CD8&quot;</span>)</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/viz_protein-6.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb689"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb689-1" data-line-number="1"><span class="co"># Let&#39;s look at the raw (non-normalized) ADT counts. You can see the values are quite high,</span></a>
<a class="sourceLine" id="cb689-2" data-line-number="2"><span class="co"># particularly in comparison to RNA values. This is due to the significantly higher protein copy</span></a>
<a class="sourceLine" id="cb689-3" data-line-number="3"><span class="co"># number in cells, which significantly reduces &#39;drop-out&#39; in ADT data</span></a>
<a class="sourceLine" id="cb689-4" data-line-number="4"><span class="kw">FeatureScatter</span>(tcells, <span class="dt">feature1 =</span> <span class="st">&quot;adt_CD4&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;adt_CD8&quot;</span>, <span class="dt">slot =</span> <span class="st">&quot;counts&quot;</span>)</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/viz_protein-7.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb690"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb690-1" data-line-number="1"><span class="co"># If you look a bit more closely, you&#39;ll see that our CD8 T cell cluster is</span></a>
<a class="sourceLine" id="cb690-2" data-line-number="2"><span class="co"># enriched for CD8 T cells, but still contains many CD4+ CD8- T cells.  This</span></a>
<a class="sourceLine" id="cb690-3" data-line-number="3"><span class="co"># is because Naive CD4 and CD8 T cells are quite similar transcriptomically,</span></a>
<a class="sourceLine" id="cb690-4" data-line-number="4"><span class="co"># and the RNA dropout levels for CD4 and CD8 are quite high.  This</span></a>
<a class="sourceLine" id="cb690-5" data-line-number="5"><span class="co"># demonstrates the challenge of defining subtle immune cell differences from</span></a>
<a class="sourceLine" id="cb690-6" data-line-number="6"><span class="co"># scRNA-seq data alone.</span></a>
<a class="sourceLine" id="cb690-7" data-line-number="7"></a>
<a class="sourceLine" id="cb690-8" data-line-number="8"><span class="co"># What fraction of T cells are double negative in gene expression? (CD4- and CD8-)</span></a>
<a class="sourceLine" id="cb690-9" data-line-number="9"><span class="co"># You can use an interactive plot to gate on the cells (do.identify = T) or use </span></a>
<a class="sourceLine" id="cb690-10" data-line-number="10"><span class="co"># Boolean conditions on CD4 and CD8A expression to find double negative cells.</span></a>
<a class="sourceLine" id="cb690-11" data-line-number="11"><span class="kw">FeatureScatter</span>(tcells, <span class="dt">feature1 =</span> <span class="st">&quot;CD4&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;CD8A&quot;</span>)</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/viz_protein-8.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb691"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb691-1" data-line-number="1"><span class="kw">ncol</span>(<span class="kw">subset</span>(tcells, <span class="dt">subset =</span> CD4 <span class="op">==</span><span class="st"> </span><span class="dv">0</span> <span class="op">&amp;</span><span class="st"> </span>CD8A <span class="op">==</span><span class="st"> </span><span class="dv">0</span>)) <span class="op">/</span><span class="st"> </span><span class="kw">ncol</span>(tcells)</a></code></pre></div>
<pre><code>## [1] 0.8291063</code></pre>
<div class="sourceCode" id="cb693"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb693-1" data-line-number="1"><span class="co"># What fraction of T cells are double negative in protein expression? (CD4- and CD8-)</span></a>
<a class="sourceLine" id="cb693-2" data-line-number="2"><span class="co"># length(cells) / length(tcells@cell.names)</span></a>
<a class="sourceLine" id="cb693-3" data-line-number="3"><span class="kw">DefaultAssay</span>(tcells) &lt;-<span class="st"> &quot;ADT&quot;</span>  <span class="co"># work with ADT count matrix</span></a>
<a class="sourceLine" id="cb693-4" data-line-number="4"><span class="kw">FeatureScatter</span>(tcells, <span class="dt">feature1 =</span> <span class="st">&quot;adt_CD4&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;adt_CD8&quot;</span>)</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/viz_protein-9.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb694"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb694-1" data-line-number="1"><span class="kw">ncol</span>(<span class="kw">subset</span>(tcells, <span class="dt">subset =</span> adt_CD4 <span class="op">&lt;</span><span class="st"> </span><span class="dv">1</span> <span class="op">&amp;</span><span class="st"> </span>adt_CD8 <span class="op">&lt;</span><span class="st"> </span><span class="dv">1</span>)) <span class="op">/</span><span class="st"> </span><span class="kw">ncol</span>(tcells)</a></code></pre></div>
<pre><code>## [1] 0.0102657</code></pre>
<div class="sourceCode" id="cb696"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb696-1" data-line-number="1"><span class="co"># Save current progress.</span></a>
<a class="sourceLine" id="cb696-2" data-line-number="2"><span class="kw">save</span>(cbmc, <span class="dt">file =</span> Rda.multi.path)</a>
<a class="sourceLine" id="cb696-3" data-line-number="3"><span class="co"># To load the data, run the following command.</span></a>
<a class="sourceLine" id="cb696-4" data-line-number="4"><span class="co"># load(Rda.multi.path)</span></a></code></pre></div>
</div>
<div id="identify-differentially-expressed-proteins-between-clusters" class="section level2">
<h2><span class="header-section-number">16.6</span> Identify differentially expressed proteins between clusters</h2>
<div class="sourceCode" id="cb697"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb697-1" data-line-number="1"><span class="co"># Downsample the clusters to a maximum of 300 cells each (makes the heatmap easier to see for</span></a>
<a class="sourceLine" id="cb697-2" data-line-number="2"><span class="co"># small clusters)</span></a>
<a class="sourceLine" id="cb697-3" data-line-number="3">cbmc.small &lt;-<span class="st"> </span><span class="kw">subset</span>(cbmc, <span class="dt">downsample =</span> <span class="dv">300</span>)</a>
<a class="sourceLine" id="cb697-4" data-line-number="4"></a>
<a class="sourceLine" id="cb697-5" data-line-number="5"><span class="co"># Find protein markers for all clusters, and draw a heatmap</span></a>
<a class="sourceLine" id="cb697-6" data-line-number="6">adt.markers &lt;-<span class="st"> </span><span class="kw">FindAllMarkers</span>(cbmc.small, <span class="dt">assay =</span> <span class="st">&quot;ADT&quot;</span>, <span class="dt">only.pos =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## Calculating cluster Memory CD4 T</code></pre>
<pre><code>## Calculating cluster CD14+ Mono</code></pre>
<pre><code>## Calculating cluster Naive CD4 T</code></pre>
<pre><code>## Calculating cluster NK</code></pre>
<pre><code>## Calculating cluster Mouse</code></pre>
<pre><code>## Calculating cluster B</code></pre>
<pre><code>## Calculating cluster CD8 T</code></pre>
<pre><code>## Calculating cluster CD16+ Mono</code></pre>
<pre><code>## Calculating cluster T/Mono doublets</code></pre>
<pre><code>## Calculating cluster CD34+</code></pre>
<pre><code>## Calculating cluster Multiplets</code></pre>
<pre><code>## Calculating cluster Eryth</code></pre>
<pre><code>## Calculating cluster Mk</code></pre>
<pre><code>## Calculating cluster DC</code></pre>
<pre><code>## Calculating cluster pDCs</code></pre>
<div class="sourceCode" id="cb713"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb713-1" data-line-number="1"><span class="kw">DoHeatmap</span>(cbmc.small, <span class="dt">features =</span> <span class="kw">unique</span>(adt.markers<span class="op">$</span>gene), <span class="dt">assay =</span> <span class="st">&quot;ADT&quot;</span>, <span class="dt">angle =</span> <span class="dv">90</span>) <span class="op">+</span><span class="st"> </span><span class="kw">NoLegend</span>()</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/ident_diff-1.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb714"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb714-1" data-line-number="1"><span class="co"># You can see that our unknown cells co-express both myeloid and lymphoid markers (true at the</span></a>
<a class="sourceLine" id="cb714-2" data-line-number="2"><span class="co"># RNA level as well). They are likely cell clumps (multiplets) that should be discarded. We&#39;ll</span></a>
<a class="sourceLine" id="cb714-3" data-line-number="3"><span class="co"># remove the mouse cells now as well</span></a>
<a class="sourceLine" id="cb714-4" data-line-number="4">cbmc &lt;-<span class="st"> </span><span class="kw">subset</span>(cbmc, <span class="dt">idents =</span> <span class="kw">c</span>(<span class="st">&quot;Multiplets&quot;</span>, <span class="st">&quot;Mouse&quot;</span>), <span class="dt">invert =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
</div>
<div id="cluster-directly-on-protein-levels" class="section level2">
<h2><span class="header-section-number">16.7</span> Cluster directly on protein levels</h2>
<p>You can also run dimensional reduction and graph-based clustering directly on CITE-seq data</p>
<div class="sourceCode" id="cb715"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb715-1" data-line-number="1"><span class="co"># Because we&#39;re going to be working with the ADT data extensively, we&#39;re going to switch the</span></a>
<a class="sourceLine" id="cb715-2" data-line-number="2"><span class="co"># default assay to the &#39;CITE&#39; assay.  This will cause all functions to use ADT data by default,</span></a>
<a class="sourceLine" id="cb715-3" data-line-number="3"><span class="co"># rather than requiring us to specify it each time</span></a>
<a class="sourceLine" id="cb715-4" data-line-number="4"><span class="kw">DefaultAssay</span>(cbmc) &lt;-<span class="st"> &quot;ADT&quot;</span></a>
<a class="sourceLine" id="cb715-5" data-line-number="5">cbmc &lt;-<span class="st"> </span><span class="kw">RunPCA</span>(cbmc, <span class="dt">features =</span> <span class="kw">rownames</span>(cbmc), <span class="dt">reduction.name =</span> <span class="st">&quot;pca_adt&quot;</span>, <span class="dt">reduction.key =</span> <span class="st">&quot;pca_adt_&quot;</span>, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<pre><code>## Warning in irlba(A = t(x = object), nv = npcs, ...): You&#39;re computing too large
## a percentage of total singular values, use a standard svd instead.</code></pre>
<pre><code>## Warning: Keys should be one or more alphanumeric characters followed by an
## underscore, setting key from pca_adt_ to pcaadt_</code></pre>
<pre><code>## Warning: All keys should be one or more alphanumeric characters followed by an
## underscore &#39;_&#39;, setting key to pcaadt_</code></pre>
<div class="sourceCode" id="cb719"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb719-1" data-line-number="1"><span class="kw">DimPlot</span>(cbmc, <span class="dt">reduction =</span> <span class="st">&quot;pca_adt&quot;</span>)</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/clust_proteins-1.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb720"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb720-1" data-line-number="1"><span class="co"># Why do we not use PCA to do dimensionality reduction here?</span></a>
<a class="sourceLine" id="cb720-2" data-line-number="2"><span class="co"># Is Euclidean distance a good distance metric in this case?</span></a>
<a class="sourceLine" id="cb720-3" data-line-number="3"><span class="kw">ElbowPlot</span>(cbmc)</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/clust_proteins-2.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb721"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb721-1" data-line-number="1"><span class="co"># Since we only have 10 markers, instead of doing PCA, we&#39;ll just use a standard euclidean</span></a>
<a class="sourceLine" id="cb721-2" data-line-number="2"><span class="co"># distance matrix here.  Also, this provides a good opportunity to demonstrate how to do</span></a>
<a class="sourceLine" id="cb721-3" data-line-number="3"><span class="co"># visualization and clustering using a custom distance matrix in Seurat</span></a>
<a class="sourceLine" id="cb721-4" data-line-number="4">adt.data &lt;-<span class="st"> </span><span class="kw">GetAssayData</span>(cbmc, <span class="dt">slot =</span> <span class="st">&quot;data&quot;</span>)</a>
<a class="sourceLine" id="cb721-5" data-line-number="5">adt.dist &lt;-<span class="st"> </span><span class="kw">dist</span>(<span class="kw">t</span>(adt.data))</a>
<a class="sourceLine" id="cb721-6" data-line-number="6"></a>
<a class="sourceLine" id="cb721-7" data-line-number="7"><span class="co"># Before we recluster the data on ADT levels, we&#39;ll stash the RNA cluster IDs for later</span></a>
<a class="sourceLine" id="cb721-8" data-line-number="8">cbmc[[<span class="st">&quot;rnaClusterID&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">Idents</span>(cbmc)</a>
<a class="sourceLine" id="cb721-9" data-line-number="9"></a>
<a class="sourceLine" id="cb721-10" data-line-number="10"><span class="co"># Now, we rerun tSNE using our distance matrix defined only on ADT (protein) levels.</span></a>
<a class="sourceLine" id="cb721-11" data-line-number="11">cbmc[[<span class="st">&quot;tsne_adt&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">RunTSNE</span>(adt.dist, <span class="dt">assay =</span> <span class="st">&quot;ADT&quot;</span>, <span class="dt">reduction.key =</span> <span class="st">&quot;adtTSNE_&quot;</span>)</a>
<a class="sourceLine" id="cb721-12" data-line-number="12">cbmc[[<span class="st">&quot;adt_snn&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">FindNeighbors</span>(adt.dist)<span class="op">$</span>snn</a></code></pre></div>
<pre><code>## Building SNN based on a provided distance matrix</code></pre>
<pre><code>## Computing SNN</code></pre>
<pre><code>## Warning: Adding a Graph without an assay associated with it</code></pre>
<div class="sourceCode" id="cb725"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb725-1" data-line-number="1">cbmc &lt;-<span class="st"> </span><span class="kw">FindClusters</span>(cbmc, <span class="dt">resolution =</span> <span class="fl">0.2</span>, <span class="dt">graph.name =</span> <span class="st">&quot;adt_snn&quot;</span>)</a></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 7895
## Number of edges: 258146
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.9491
## Number of communities: 11
## Elapsed time: 0 seconds</code></pre>
<pre><code>## Warning: Adding a command log without an assay associated with it</code></pre>
<div class="sourceCode" id="cb728"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb728-1" data-line-number="1"><span class="co"># We can compare the RNA and protein clustering, and use this to annotate the protein clustering</span></a>
<a class="sourceLine" id="cb728-2" data-line-number="2"><span class="co"># (we could also of course use FindMarkers)</span></a>
<a class="sourceLine" id="cb728-3" data-line-number="3">clustering.table &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="kw">Idents</span>(cbmc), cbmc<span class="op">$</span>rnaClusterID)</a>
<a class="sourceLine" id="cb728-4" data-line-number="4">clustering.table</a></code></pre></div>
<pre><code>##     
##      Memory CD4 T CD14+ Mono Naive CD4 T   NK    B CD8 T CD16+ Mono
##   0          1754          0        1217   29    0    27          0
##   1             0       2189           0    4    0     0         30
##   2             3          0           2  890    3     1          0
##   3             0          4           0    2  319     0          2
##   4            24          0          18    4    1   243          0
##   5             1         27           4  157    2     2         10
##   6             4          5           0    1    0     0          0
##   7             4         59           4    0    0     0          9
##   8             0          9           0    2    0     0        179
##   9             0          0           1    0    0     0          0
##   10            1          0           2    0   25     0          0
##     
##      T/Mono doublets CD34+ Eryth   Mk   DC pDCs
##   0                5     2     4   24    1    2
##   1                1     1     5   25   55    0
##   2                0     1     3    7    2    1
##   3                0     2     2    3    0    0
##   4                0     0     1    2    0    0
##   5               56     0     9   16    6    2
##   6                1   113    81   16    5    0
##   7              117     0     0    2    0    1
##   8                0     0     0    1    0    0
##   9                0     0     0    0    1   43
##   10               2     0     0    0    0    0</code></pre>
<div class="sourceCode" id="cb730"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb730-1" data-line-number="1">new.cluster.ids &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;CD4 T&quot;</span>, <span class="st">&quot;CD14+ Mono&quot;</span>, <span class="st">&quot;NK&quot;</span>, <span class="st">&quot;B&quot;</span>, <span class="st">&quot;CD8 T&quot;</span>, <span class="st">&quot;NK&quot;</span>, <span class="st">&quot;CD34+&quot;</span>, <span class="st">&quot;T/Mono doublets&quot;</span>, <span class="st">&quot;CD16+ Mono&quot;</span>, <span class="st">&quot;pDCs&quot;</span>, <span class="st">&quot;B&quot;</span>)</a>
<a class="sourceLine" id="cb730-2" data-line-number="2"><span class="kw">names</span>(new.cluster.ids) &lt;-<span class="st"> </span><span class="kw">levels</span>(cbmc)</a>
<a class="sourceLine" id="cb730-3" data-line-number="3">cbmc &lt;-<span class="st"> </span><span class="kw">RenameIdents</span>(cbmc, new.cluster.ids)</a>
<a class="sourceLine" id="cb730-4" data-line-number="4"></a>
<a class="sourceLine" id="cb730-5" data-line-number="5">tsne_rnaClusters &lt;-<span class="st"> </span><span class="kw">DimPlot</span>(cbmc, <span class="dt">reduction =</span> <span class="st">&quot;tsne_adt&quot;</span>, <span class="dt">group.by =</span> <span class="st">&quot;rnaClusterID&quot;</span>, <span class="dt">pt.size =</span> <span class="fl">0.5</span>) <span class="op">+</span><span class="st"> </span><span class="kw">NoLegend</span>()</a>
<a class="sourceLine" id="cb730-6" data-line-number="6">tsne_rnaClusters &lt;-<span class="st"> </span>tsne_rnaClusters <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Clustering based on scRNA-seq&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust =</span> <span class="fl">0.5</span>))</a>
<a class="sourceLine" id="cb730-7" data-line-number="7">tsne_rnaClusters &lt;-<span class="st"> </span><span class="kw">LabelClusters</span>(<span class="dt">plot =</span> tsne_rnaClusters, <span class="dt">id =</span> <span class="st">&quot;rnaClusterID&quot;</span>, <span class="dt">size =</span> <span class="dv">4</span>)</a>
<a class="sourceLine" id="cb730-8" data-line-number="8"></a>
<a class="sourceLine" id="cb730-9" data-line-number="9">tsne_adtClusters &lt;-<span class="st"> </span><span class="kw">DimPlot</span>(cbmc, <span class="dt">reduction =</span> <span class="st">&quot;tsne_adt&quot;</span>, <span class="dt">pt.size =</span> <span class="fl">0.5</span>) <span class="op">+</span><span class="st"> </span><span class="kw">NoLegend</span>()</a>
<a class="sourceLine" id="cb730-10" data-line-number="10">tsne_adtClusters &lt;-<span class="st"> </span>tsne_adtClusters <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Clustering based on ADT signal&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust =</span> <span class="fl">0.5</span>))</a>
<a class="sourceLine" id="cb730-11" data-line-number="11">tsne_adtClusters &lt;-<span class="st"> </span><span class="kw">LabelClusters</span>(<span class="dt">plot =</span> tsne_adtClusters, <span class="dt">id =</span> <span class="st">&quot;ident&quot;</span>, <span class="dt">size =</span> <span class="dv">4</span>)</a>
<a class="sourceLine" id="cb730-12" data-line-number="12"></a>
<a class="sourceLine" id="cb730-13" data-line-number="13"><span class="co"># Note: for this comparison, both the RNA and protein clustering are visualized on a tSNE</span></a>
<a class="sourceLine" id="cb730-14" data-line-number="14"><span class="co"># generated using the ADT distance matrix.</span></a>
<a class="sourceLine" id="cb730-15" data-line-number="15"><span class="kw">wrap_plots</span>(<span class="kw">list</span>(tsne_rnaClusters, tsne_adtClusters), <span class="dt">ncol =</span> <span class="dv">2</span>)</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/clust_proteins-3.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb731"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb731-1" data-line-number="1"><span class="co"># What differences if any do you see between the clustering based on scRNA-seq</span></a>
<a class="sourceLine" id="cb731-2" data-line-number="2"><span class="co"># and the clustering based on ADT signal?</span></a>
<a class="sourceLine" id="cb731-3" data-line-number="3"><span class="co"># How could we combine these datasets in a joint, integrative analysis?</span></a>
<a class="sourceLine" id="cb731-4" data-line-number="4"></a>
<a class="sourceLine" id="cb731-5" data-line-number="5"><span class="co"># Save current progress.</span></a>
<a class="sourceLine" id="cb731-6" data-line-number="6"><span class="kw">save</span>(cbmc, <span class="dt">file =</span> Rda.protein.path)</a>
<a class="sourceLine" id="cb731-7" data-line-number="7"><span class="co"># To load the data, run the following command.</span></a>
<a class="sourceLine" id="cb731-8" data-line-number="8"><span class="co"># load(Rda.protein.path)</span></a></code></pre></div>
<p>The ADT-based clustering yields similar results, but with a few differences</p>
<ul>
<li>Clustering is improved for CD4/CD8 T cell populations, based on the robust ADT data for CD4, CD8, CD14, and CD45RA</li>
<li>However, some clusters for which the ADT data does not contain good distinguishing protein markers (i.e. Mk/Ery/DC) lose separation</li>
<li>You can verify this using FindMarkers at the RNA level, as well</li>
</ul>
</div>
<div id="additional-exploration-another-example-of-multi-modal-analysis" class="section level2">
<h2><span class="header-section-number">16.8</span> Additional exploration: another example of multi-modal analysis</h2>
<p>For another nice example of multi-modal analysis, please explore this <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/cicero/inst/doc/website.html">single cell ATAC-Seq vignette</a> and this <a href="https://satijalab.org/signac/articles/integration.html">scATAC-seq data integration</a>.</p>
</div>
<div id="acknowledgements-2" class="section level2">
<h2><span class="header-section-number">16.9</span> Acknowledgements</h2>
<p>This document is largely a tutorial from Seurat website, with some small modifications. The official vignette is available at <a href="https://satijalab.org/seurat/multimodal_vignette.html">Using Seurat with multi-modal data</a>.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="mutli-omic-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="diy-project.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["2020_scWorkshop.pdf", "2020_scWorkshop.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
