<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>10 Feature Selection and Cluster Analysis | ANALYSIS OF SINGLE CELL RNA-SEQ DATA</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="10 Feature Selection and Cluster Analysis | ANALYSIS OF SINGLE CELL RNA-SEQ DATA" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://broadinstitute.github.io/2020_scWorkshop/" />
  <meta property="og:image" content="https://broadinstitute.github.io/2020_scWorkshop/images/cover.png" />
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="broadinstitute/2019_scWorkship" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="10 Feature Selection and Cluster Analysis | ANALYSIS OF SINGLE CELL RNA-SEQ DATA" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="twitter:image" content="https://broadinstitute.github.io/2020_scWorkshop/images/cover.png" />

<meta name="author" content="Orr Ashenberg" />
<meta name="author" content="Dana Silverbush" />
<meta name="author" content="Kirk Gosik" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="identifying-cell-populations.html"/>
<link rel="next" href="batch-correction-lecture.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">ANALYSIS OF SINGLE CELL RNA-SEQ DATA</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#course-overview"><i class="fa fa-check"></i><b>1.1</b> COURSE OVERVIEW</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#targeted-audience-assumed-background"><i class="fa fa-check"></i><b>1.2</b> TARGETED AUDIENCE &amp; ASSUMED BACKGROUND</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#course-format"><i class="fa fa-check"></i><b>1.3</b> COURSE FORMAT</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#getting-started"><i class="fa fa-check"></i><b>1.4</b> Getting Started</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#session-content"><i class="fa fa-check"></i><b>1.5</b> SESSION CONTENT</a><ul>
<li class="chapter" data-level="1.5.1" data-path="index.html"><a href="index.html#monday-classes-from-0800-to-1600-lunch-break-1-hr-40-min-of-total-coffee-breaks"><i class="fa fa-check"></i><b>1.5.1</b> Monday – Classes from 08:00 to 16:00 (lunch break-1 hr, 40 min of total coffee breaks)</a></li>
<li class="chapter" data-level="1.5.2" data-path="index.html"><a href="index.html#tuesday-classes-from-0800-to-1600"><i class="fa fa-check"></i><b>1.5.2</b> Tuesday – Classes from 08:00 to 16:00</a></li>
<li class="chapter" data-level="1.5.3" data-path="index.html"><a href="index.html#wednesday-classes-from-0800-to-1600"><i class="fa fa-check"></i><b>1.5.3</b> Wednesday – Classes from 08:00 to 16:00</a></li>
<li class="chapter" data-level="1.5.4" data-path="index.html"><a href="index.html#thursday-classes-from-0800-to-1600"><i class="fa fa-check"></i><b>1.5.4</b> Thursday – Classes from 08:00 to 16:00</a></li>
<li class="chapter" data-level="1.5.5" data-path="index.html"><a href="index.html#friday-classes-from-0800-to-1600"><i class="fa fa-check"></i><b>1.5.5</b> Friday – Classes from 08:00 to 16:00</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction-1.html"><a href="introduction-1.html"><i class="fa fa-check"></i><b>2</b> Introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="introduction-1.html"><a href="introduction-1.html#slides"><i class="fa fa-check"></i><b>2.1</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html"><i class="fa fa-check"></i><b>3</b> Understanding Sequencing Raw Data</a><ul>
<li class="chapter" data-level="3.1" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#class-environment"><i class="fa fa-check"></i><b>3.1</b> Class Environment</a><ul>
<li class="chapter" data-level="3.1.1" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#getting-into-aws-instance"><i class="fa fa-check"></i><b>3.1.1</b> Getting into AWS Instance</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#shell-and-unix-commands"><i class="fa fa-check"></i><b>3.2</b> Shell and Unix commands</a><ul>
<li class="chapter" data-level="3.2.1" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#common-linux-commands"><i class="fa fa-check"></i><b>3.2.1</b> Common Linux Commands</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#file-formats"><i class="fa fa-check"></i><b>3.3</b> File formats</a><ul>
<li class="chapter" data-level="3.3.1" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#view-fastq-files"><i class="fa fa-check"></i><b>3.3.1</b> View FASTQ Files</a></li>
<li class="chapter" data-level="3.3.2" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#view-bam-files"><i class="fa fa-check"></i><b>3.3.2</b> View BAM Files</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#public-data-repositories"><i class="fa fa-check"></i><b>3.4</b> Public data repositories</a><ul>
<li class="chapter" data-level="3.4.1" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#cellranger10x"><i class="fa fa-check"></i><b>3.4.1</b> Cellranger/10x</a></li>
<li class="chapter" data-level="3.4.2" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#geo"><i class="fa fa-check"></i><b>3.4.2</b> GEO</a></li>
<li class="chapter" data-level="3.4.3" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#single-cell-portal"><i class="fa fa-check"></i><b>3.4.3</b> Single Cell Portal</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#docker-commands"><i class="fa fa-check"></i><b>3.5</b> Docker Commands</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="transcriptome-quantification.html"><a href="transcriptome-quantification.html"><i class="fa fa-check"></i><b>4</b> Transcriptome Quantification</a><ul>
<li class="chapter" data-level="4.1" data-path="transcriptome-quantification.html"><a href="transcriptome-quantification.html#google-slides"><i class="fa fa-check"></i><b>4.1</b> Google Slides</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html"><i class="fa fa-check"></i><b>5</b> Processing scRNAseq Data</a><ul>
<li class="chapter" data-level="5.1" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#goal"><i class="fa fa-check"></i><b>5.1</b> Goal</a></li>
<li class="chapter" data-level="5.2" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#further-reading"><i class="fa fa-check"></i><b>5.2</b> Further reading</a></li>
<li class="chapter" data-level="5.3" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#download-data"><i class="fa fa-check"></i><b>5.3</b> Download data</a><ul>
<li class="chapter" data-level="5.3.1" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#fastq-file-format"><i class="fa fa-check"></i><b>5.3.1</b> Fastq file format</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#align-the-reads"><i class="fa fa-check"></i><b>5.4</b> Align the reads</a><ul>
<li class="chapter" data-level="5.4.1" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#star-align"><i class="fa fa-check"></i><b>5.4.1</b> STAR align</a></li>
<li class="chapter" data-level="5.4.2" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#bam-file-format"><i class="fa fa-check"></i><b>5.4.2</b> Bam file format</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#visualization"><i class="fa fa-check"></i><b>5.5</b> Visualization</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html"><i class="fa fa-check"></i><b>6</b> Introduction R/Bioconductor</a><ul>
<li class="chapter" data-level="6.1" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#start-environment"><i class="fa fa-check"></i><b>6.1</b> Start Environment</a><ul>
<li class="chapter" data-level="6.1.1" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#local-command"><i class="fa fa-check"></i><b>6.1.1</b> Local Command</a></li>
<li class="chapter" data-level="6.1.2" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#aws-command"><i class="fa fa-check"></i><b>6.1.2</b> AWS Command</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#installing-packages"><i class="fa fa-check"></i><b>6.2</b> Installing packages</a><ul>
<li class="chapter" data-level="6.2.1" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#cran"><i class="fa fa-check"></i><b>6.2.1</b> CRAN</a></li>
<li class="chapter" data-level="6.2.2" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#github"><i class="fa fa-check"></i><b>6.2.2</b> Github</a></li>
<li class="chapter" data-level="6.2.3" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#bioconductor"><i class="fa fa-check"></i><b>6.2.3</b> Bioconductor</a></li>
<li class="chapter" data-level="6.2.4" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#source"><i class="fa fa-check"></i><b>6.2.4</b> Source</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#installation-instructions"><i class="fa fa-check"></i><b>6.3</b> Installation instructions:</a><ul>
<li class="chapter" data-level="6.3.1" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#classestypes"><i class="fa fa-check"></i><b>6.3.1</b> Classes/Types</a></li>
<li class="chapter" data-level="6.3.2" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#data-structures"><i class="fa fa-check"></i><b>6.3.2</b> Data structures</a></li>
<li class="chapter" data-level="6.3.3" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#detour-to-s3s4"><i class="fa fa-check"></i><b>6.3.3</b> Detour to S3/S4</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#more-information"><i class="fa fa-check"></i><b>6.4</b> More information</a><ul>
<li class="chapter" data-level="6.4.1" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#checking-for-help-for-any-function"><i class="fa fa-check"></i><b>6.4.1</b> Checking for help for any function!</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#grammer-of-graphics-ggplot2"><i class="fa fa-check"></i><b>6.5</b> Grammer of Graphics (ggplot2)</a><ul>
<li class="chapter" data-level="6.5.1" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#what-is-ggplot2"><i class="fa fa-check"></i><b>6.5.1</b> What is ggplot2?</a></li>
<li class="chapter" data-level="6.5.2" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#principles-of-ggplot2"><i class="fa fa-check"></i><b>6.5.2</b> Principles of ggplot2</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#reference"><i class="fa fa-check"></i><b>6.6</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>7</b> Quality Control</a><ul>
<li class="chapter" data-level="7.1" data-path="quality-control.html"><a href="quality-control.html#slides-1"><i class="fa fa-check"></i><b>7.1</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html"><i class="fa fa-check"></i><b>8</b> Data Wrangling scRNAseq</a><ul>
<li class="chapter" data-level="8.1" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#goal-1"><i class="fa fa-check"></i><b>8.1</b> Goal</a></li>
<li class="chapter" data-level="8.2" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#introduction-2"><i class="fa fa-check"></i><b>8.2</b> Introduction</a><ul>
<li class="chapter" data-level="8.2.1" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#load-necessary-packages"><i class="fa fa-check"></i><b>8.2.1</b> Load necessary packages</a></li>
<li class="chapter" data-level="8.2.2" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#read-in-nsclc-counts-matrix."><i class="fa fa-check"></i><b>8.2.2</b> Read in NSCLC counts matrix.</a></li>
<li class="chapter" data-level="8.2.3" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#lets-examine-the-sparse-counts-matrix"><i class="fa fa-check"></i><b>8.2.3</b> Let’s examine the sparse counts matrix</a></li>
<li class="chapter" data-level="8.2.4" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#how-big-is-the-matrix"><i class="fa fa-check"></i><b>8.2.4</b> How big is the matrix?</a></li>
<li class="chapter" data-level="8.2.5" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#how-much-memory-does-a-sparse-matrix-take-up-relative-to-a-dense-matrix"><i class="fa fa-check"></i><b>8.2.5</b> How much memory does a sparse matrix take up relative to a dense matrix?</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#filtering-low-quality-cells"><i class="fa fa-check"></i><b>8.3</b> Filtering low-quality cells</a><ul>
<li class="chapter" data-level="8.3.1" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#look-at-the-summary-counts-for-genes-and-cells"><i class="fa fa-check"></i><b>8.3.1</b> Look at the summary counts for genes and cells</a></li>
<li class="chapter" data-level="8.3.2" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#plot-cells-ranked-by-their-number-of-detected-genes."><i class="fa fa-check"></i><b>8.3.2</b> Plot cells ranked by their number of detected genes.</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#beginning-with-seurat-httpsatijalab.orgseurat"><i class="fa fa-check"></i><b>8.4</b> Beginning with Seurat: <span>http://satijalab.org/seurat/</span></a><ul>
<li class="chapter" data-level="8.4.1" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#creating-a-seurat-object"><i class="fa fa-check"></i><b>8.4.1</b> Creating a seurat object</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#preprocessing-step-1-filter-out-low-quality-cells"><i class="fa fa-check"></i><b>8.5</b> Preprocessing step 1 : Filter out low-quality cells</a></li>
<li class="chapter" data-level="8.6" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#examine-contents-of-seurat-object"><i class="fa fa-check"></i><b>8.6</b> Examine contents of Seurat object</a><ul>
<li class="chapter" data-level="8.6.1" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#preprocessing-step-2-expression-normalization"><i class="fa fa-check"></i><b>8.6.1</b> Preprocessing step 2 : Expression normalization</a></li>
</ul></li>
<li class="chapter" data-level="8.7" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#detection-of-variable-genes-across-the-single-cells"><i class="fa fa-check"></i><b>8.7</b> Detection of variable genes across the single cells</a></li>
<li class="chapter" data-level="8.8" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#gene-set-expression-across-cells"><i class="fa fa-check"></i><b>8.8</b> Gene set expression across cells</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="identifying-cell-populations.html"><a href="identifying-cell-populations.html"><i class="fa fa-check"></i><b>9</b> Identifying Cell Populations</a><ul>
<li class="chapter" data-level="9.1" data-path="identifying-cell-populations.html"><a href="identifying-cell-populations.html#slides-2"><i class="fa fa-check"></i><b>9.1</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html"><i class="fa fa-check"></i><b>10</b> Feature Selection and Cluster Analysis</a><ul>
<li class="chapter" data-level="10.1" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#abstract"><i class="fa fa-check"></i><b>10.1</b> Abstract</a></li>
<li class="chapter" data-level="10.2" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#seurat-tutorial-redo"><i class="fa fa-check"></i><b>10.2</b> Seurat Tutorial Redo</a><ul>
<li class="chapter" data-level="10.2.1" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#preprocessing-steps"><i class="fa fa-check"></i><b>10.2.1</b> Preprocessing Steps</a></li>
<li class="chapter" data-level="10.2.2" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#start-of-identifying-cell-types"><i class="fa fa-check"></i><b>10.2.2</b> Start of Identifying Cell Types</a></li>
<li class="chapter" data-level="10.2.3" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#run-non-linear-dimensional-reduction-umaptsne"><i class="fa fa-check"></i><b>10.2.3</b> Run non-linear dimensional reduction (UMAP/tSNE)</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#feature-selection"><i class="fa fa-check"></i><b>10.3</b> Feature Selection</a><ul>
<li class="chapter" data-level="10.3.1" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#differential-expression-analysis"><i class="fa fa-check"></i><b>10.3.1</b> Differential Expression Analysis</a></li>
<li class="chapter" data-level="10.3.2" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#check-clusters"><i class="fa fa-check"></i><b>10.3.2</b> Check Clusters</a></li>
<li class="chapter" data-level="10.3.3" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#view-entire-object-structure"><i class="fa fa-check"></i><b>10.3.3</b> View Entire Object Structure</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#probabilistic-lda-clustering"><i class="fa fa-check"></i><b>10.4</b> Probabilistic (LDA) Clustering</a><ul>
<li class="chapter" data-level="10.4.1" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#example-lda-in-bulk"><i class="fa fa-check"></i><b>10.4.1</b> Example LDA in Bulk</a></li>
<li class="chapter" data-level="10.4.2" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#pbmc-lda"><i class="fa fa-check"></i><b>10.4.2</b> PBMC LDA</a></li>
<li class="chapter" data-level="10.4.3" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#practice-visualizingembedding"><i class="fa fa-check"></i><b>10.4.3</b> Practice Visualizing/Embedding</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#other-options-for-analysis"><i class="fa fa-check"></i><b>10.5</b> Other Options For Analysis</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="batch-correction-lecture.html"><a href="batch-correction-lecture.html"><i class="fa fa-check"></i><b>11</b> Batch Correction Lecture</a><ul>
<li class="chapter" data-level="11.1" data-path="batch-correction-lecture.html"><a href="batch-correction-lecture.html#slides-3"><i class="fa fa-check"></i><b>11.1</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html"><i class="fa fa-check"></i><b>12</b> Batch Correction Lab</a><ul>
<li class="chapter" data-level="12.1" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#load-settings-and-packages"><i class="fa fa-check"></i><b>12.1</b> Load settings and packages</a></li>
<li class="chapter" data-level="12.2" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#read-in-pancreas-expression-matrices"><i class="fa fa-check"></i><b>12.2</b> Read in pancreas expression matrices</a></li>
<li class="chapter" data-level="12.3" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#preparing-the-individual-seurat-objects-for-each-pancreas-dataset-without-batch-correction"><i class="fa fa-check"></i><b>12.3</b> Preparing the individual Seurat objects for each pancreas dataset without batch correction</a></li>
<li class="chapter" data-level="12.4" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#cluster-pancreatic-datasets-without-batch-correction"><i class="fa fa-check"></i><b>12.4</b> Cluster pancreatic datasets without batch correction</a><ul>
<li class="chapter" data-level="12.4.1" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#batch-correction-canonical-correlation-analysis-cca-mutual-nearest-neighbors-mnn-using-seurat-v3"><i class="fa fa-check"></i><b>12.4.1</b> Batch correction: canonical correlation analysis (CCA) + mutual nearest neighbors (MNN) using Seurat v3</a></li>
<li class="chapter" data-level="12.4.2" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#batch-correction-integrative-non-negative-matrix-factorization-nmf-using-liger"><i class="fa fa-check"></i><b>12.4.2</b> Batch correction: integrative non-negative matrix factorization (NMF) using LIGER</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#additional-exploration-regressing-out-unwanted-covariates"><i class="fa fa-check"></i><b>12.5</b> Additional exploration: Regressing out unwanted covariates</a></li>
<li class="chapter" data-level="12.6" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#additional-exploration-kbet"><i class="fa fa-check"></i><b>12.6</b> Additional exploration: kBET</a></li>
<li class="chapter" data-level="12.7" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#additional-exploration-seurat-3"><i class="fa fa-check"></i><b>12.7</b> Additional exploration: Seurat 3</a></li>
<li class="chapter" data-level="12.8" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#acknowledgements"><i class="fa fa-check"></i><b>12.8</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="trajectory-inference.html"><a href="trajectory-inference.html"><i class="fa fa-check"></i><b>13</b> Trajectory Inference</a><ul>
<li class="chapter" data-level="13.1" data-path="trajectory-inference.html"><a href="trajectory-inference.html#slides-4"><i class="fa fa-check"></i><b>13.1</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html"><i class="fa fa-check"></i><b>14</b> Trajectory Analysis</a><ul>
<li class="chapter" data-level="14.1" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#load-settings-and-packages-1"><i class="fa fa-check"></i><b>14.1</b> Load settings and packages</a></li>
<li><a href="trajectory-analysis.html#first-look-at-the-differentiation-data-from-deng-et-al.we-will-use-a-nice-smart-seq2-single-cell-rna-seq-data-from-single-cell-rna-seq-reveals-dynamic-random-monoallelic-gene-expression-in-mammalian-cells.-here-is-one-relevant-detail-from-their-paper-to-investigate-allele-specific-gene-expression-at-single-cell-resolution-we-isolated-269-individual-cells-dissociated-from-in-vivo-f1-embryos-casteij-c57bl6j-hereafter-abbreviated-as-cast-and-c57-respectively-from-oocyte-to-blastocyst-stages-of-mouse-preimplantation-development-pd"><span class="toc-section-number">14.2</span> First look at the differentiation data from Deng et
al. We will use a nice SMART-Seq2 single cell RNA-seq data from <span>Single-Cell RNA-Seq Reveals Dynamic, Random Monoallelic Gene Expression in Mammalian Cells</span>. Here is one relevant detail from their paper: “To investigate allele-specific gene expression at single-cell resolution, we isolated 269 individual cells dissociated from in vivo F1 embryos (CAST/EiJ × C57BL/6J, hereafter abbreviated as CAST and C57, respectively) from oocyte to blastocyst stages of mouse preimplantation development (PD)”</a></li>
<li class="chapter" data-level="14.3" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#principle-components-analysis"><i class="fa fa-check"></i><b>14.3</b> Principle Components Analysis</a></li>
<li class="chapter" data-level="14.4" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#diffusion-map-pseudotime"><i class="fa fa-check"></i><b>14.4</b> Diffusion map pseudotime</a></li>
<li class="chapter" data-level="14.5" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#slingshot-map-pseudotime"><i class="fa fa-check"></i><b>14.5</b> Slingshot map pseudotime</a></li>
<li class="chapter" data-level="14.6" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#find-temporally-expressed-genes"><i class="fa fa-check"></i><b>14.6</b> Find temporally expressed genes</a></li>
<li class="chapter" data-level="14.7" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#comparison-of-the-different-trajectory-inference-methods"><i class="fa fa-check"></i><b>14.7</b> Comparison of the different trajectory inference methods</a></li>
<li class="chapter" data-level="14.8" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#plots-of-gene-expression-over-time."><i class="fa fa-check"></i><b>14.8</b> Plots of gene expression over time.</a></li>
<li class="chapter" data-level="14.9" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#acknowledgements-1"><i class="fa fa-check"></i><b>14.9</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="mutli-omic-analysis.html"><a href="mutli-omic-analysis.html"><i class="fa fa-check"></i><b>15</b> Mutli-omic Analysis</a><ul>
<li class="chapter" data-level="15.1" data-path="mutli-omic-analysis.html"><a href="mutli-omic-analysis.html#slides-5"><i class="fa fa-check"></i><b>15.1</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="cite-seq.html"><a href="cite-seq.html"><i class="fa fa-check"></i><b>16</b> Cite-Seq</a><ul>
<li class="chapter" data-level="16.1" data-path="cite-seq.html"><a href="cite-seq.html#load-settings-and-packages-2"><i class="fa fa-check"></i><b>16.1</b> Load settings and packages</a></li>
<li class="chapter" data-level="16.2" data-path="cite-seq.html"><a href="cite-seq.html#load-in-the-data"><i class="fa fa-check"></i><b>16.2</b> Load in the data</a></li>
<li class="chapter" data-level="16.3" data-path="cite-seq.html"><a href="cite-seq.html#setup-a-seurat-object-and-cluster-cells-based-on-rna-expression"><i class="fa fa-check"></i><b>16.3</b> Setup a Seurat object, and cluster cells based on RNA expression</a></li>
<li class="chapter" data-level="16.4" data-path="cite-seq.html"><a href="cite-seq.html#add-the-protein-expression-levels-to-the-seurat-object"><i class="fa fa-check"></i><b>16.4</b> Add the protein expression levels to the Seurat object</a></li>
<li class="chapter" data-level="16.5" data-path="cite-seq.html"><a href="cite-seq.html#visualize-protein-levels-on-rna-clusters"><i class="fa fa-check"></i><b>16.5</b> Visualize protein levels on RNA clusters</a></li>
<li class="chapter" data-level="16.6" data-path="cite-seq.html"><a href="cite-seq.html#identify-differentially-expressed-proteins-between-clusters"><i class="fa fa-check"></i><b>16.6</b> Identify differentially expressed proteins between clusters</a></li>
<li class="chapter" data-level="16.7" data-path="cite-seq.html"><a href="cite-seq.html#cluster-directly-on-protein-levels"><i class="fa fa-check"></i><b>16.7</b> Cluster directly on protein levels</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="additional-exploration-another-example-of-multi-modal-analysis.html"><a href="additional-exploration-another-example-of-multi-modal-analysis.html"><i class="fa fa-check"></i><b>17</b> Additional exploration: another example of multi-modal analysis</a><ul>
<li class="chapter" data-level="17.1" data-path="additional-exploration-another-example-of-multi-modal-analysis.html"><a href="additional-exploration-another-example-of-multi-modal-analysis.html#acknowledgements-2"><i class="fa fa-check"></i><b>17.1</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="single-cell-resources.html"><a href="single-cell-resources.html"><i class="fa fa-check"></i><b>18</b> Single Cell Resources</a><ul>
<li class="chapter" data-level="18.1" data-path="single-cell-resources.html"><a href="single-cell-resources.html#comprehensive-list-of-single-cell-resources"><i class="fa fa-check"></i><b>18.1</b> Comprehensive list of single-cell resources</a></li>
<li class="chapter" data-level="18.2" data-path="single-cell-resources.html"><a href="single-cell-resources.html#computational-packages-for-single-cell-analysis"><i class="fa fa-check"></i><b>18.2</b> Computational packages for single-cell analysis</a></li>
<li class="chapter" data-level="18.3" data-path="single-cell-resources.html"><a href="single-cell-resources.html#elife-commentary-on-the-human-cell-atlas"><i class="fa fa-check"></i><b>18.3</b> eLife Commentary on the Human Cell Atlas</a></li>
<li class="chapter" data-level="18.4" data-path="single-cell-resources.html"><a href="single-cell-resources.html#online-courses"><i class="fa fa-check"></i><b>18.4</b> Online courses</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ANALYSIS OF SINGLE CELL RNA-SEQ DATA</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="feature-selection-and-cluster-analysis" class="section level1">
<h1><span class="header-section-number">10</span> Feature Selection and Cluster Analysis</h1>
<div id="abstract" class="section level2">
<h2><span class="header-section-number">10.1</span> Abstract</h2>
<p>Many methods have been used to determine differential gene expression from single-cell RNA (scRNA)-seq data. We evaluated 36 approaches using experimental and synthetic data and found considerable differences in the number and characteristics of the genes that are called differentially expressed. Prefiltering of lowly expressed genes has important effects, particularly for some of the methods developed for bulk RNA-seq data analysis. However, we found that bulk RNA-seq analysis methods do not generally perform worse than those developed specifically for scRNA-seq. We also present conquer, a repository of consistently processed, analysis-ready public scRNA-seq data sets that is aimed at simplifying method evaluation and reanalysis of published results. Each data set provides abundance estimates for both genes and transcripts, as well as quality control and exploratory analysis reports. <span class="citation">(<span class="citeproc-not-found" data-reference-id="soneson2018bias"><strong>???</strong></span>)</span></p>
<p>Cells are the basic building blocks of organisms and each cell is unique. Single-cell RNA sequencing has emerged as an indispensable tool to dissect the cellular heterogeneity and decompose tissues into cell types and/or cell states, which offers enormous potential for de novo discovery. Single-cell transcriptomic atlases provide unprecedented resolution to reveal complex cellular events and deepen our understanding of biological systems. In this review, we summarize and compare single-cell RNA sequencing technologies, that were developed since 2009, to facilitate a well-informed choice of method. The applications of these methods in different biological contexts are also discussed. We anticipate an ever-increasing role of single-cell RNA sequencing in biology with further improvement in providing spatial information and coupling to other cellular modalities. In the future, such biological findings will greatly benefit medical research. <span class="citation">(<span class="citeproc-not-found" data-reference-id="hedlund2018single"><strong>???</strong></span>)</span></p>
</div>
<div id="seurat-tutorial-redo" class="section level2">
<h2><span class="header-section-number">10.2</span> Seurat Tutorial Redo</h2>
<p>For this tutorial, we will be analyzing the a dataset of Non-Small Cell Lung Cancer Cells (NSCLC) freely available from 10X Genomics (<a href="https://support.10xgenomics.com/single-cell-vdj/datasets/2.2.0/vdj_v1_hs_nsclc_5gex" class="uri">https://support.10xgenomics.com/single-cell-vdj/datasets/2.2.0/vdj_v1_hs_nsclc_5gex</a>), using the Seurat R package (<a href="http://satijalab.org/seurat/" class="uri">http://satijalab.org/seurat/</a>), a popular and powerful set of tools to conduct scRNA-seq analysis in R. In this dataset, there are 7802 single cells that were sequenced on the Illumina NovaSeq 6000. Please note this tutorial borrows heavily from Seurat’s tutorials, so feel free to go through them in more detail.</p>
<p><strong>Task: Check the dirname to directory where you saved your data</strong></p>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb194-1" data-line-number="1">knitr<span class="op">::</span>opts_knit<span class="op">$</span><span class="kw">set</span>(<span class="dt">progress=</span><span class="ot">FALSE</span>, <span class="dt">verbose=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb194-2" data-line-number="2"></a>
<a class="sourceLine" id="cb194-3" data-line-number="3"><span class="kw">library</span>(Seurat)</a>
<a class="sourceLine" id="cb194-4" data-line-number="4"><span class="kw">library</span>(dplyr)</a></code></pre></div>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb198-1" data-line-number="1"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb198-2" data-line-number="2"><span class="kw">library</span>(CountClust)</a>
<a class="sourceLine" id="cb198-3" data-line-number="3"></a>
<a class="sourceLine" id="cb198-4" data-line-number="4">dirname &lt;-<span class="st"> &quot;data/&quot;</span></a>
<a class="sourceLine" id="cb198-5" data-line-number="5"></a>
<a class="sourceLine" id="cb198-6" data-line-number="6">counts_matrix_filename =<span class="st"> </span><span class="kw">paste0</span>(dirname, <span class="st">&quot;filtered_gene_bc_matrices/GRCh38/&quot;</span>)</a>
<a class="sourceLine" id="cb198-7" data-line-number="7">counts &lt;-<span class="st"> </span><span class="kw">Read10X</span>(<span class="dt">data.dir =</span> counts_matrix_filename)  <span class="co"># Seurat function to read in 10x count data</span></a></code></pre></div>
<div class="sourceCode" id="cb199"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb199-1" data-line-number="1"><span class="co">## Using the first 1000 cells for computational efficiency</span></a>
<a class="sourceLine" id="cb199-2" data-line-number="2">seurat &lt;-<span class="st"> </span><span class="kw">CreateSeuratObject</span>(<span class="dt">counts =</span> counts, <span class="dt">min.cells =</span> <span class="dv">3</span>, <span class="dt">min.features =</span> <span class="dv">350</span>, <span class="dt">project =</span> <span class="st">&quot;10X_NSCLC&quot;</span>)</a></code></pre></div>
<pre><code>## Warning: Feature names cannot have underscores (&#39;_&#39;), replacing with dashes (&#39;-&#39;)</code></pre>
<div id="preprocessing-steps" class="section level3">
<h3><span class="header-section-number">10.2.1</span> Preprocessing Steps</h3>
<p>This was all covered in Dana’s Lab!</p>
<div class="sourceCode" id="cb201"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb201-1" data-line-number="1"><span class="co"># The number of genes and UMIs (nFeature_RNA nCount_RNA) are automatically calculated</span></a>
<a class="sourceLine" id="cb201-2" data-line-number="2"><span class="co"># for every object by Seurat.  For non-UMI data, nCount_RNA represents the sum of</span></a>
<a class="sourceLine" id="cb201-3" data-line-number="3"><span class="co"># the non-normalized values within a cell We calculate the percentage of</span></a>
<a class="sourceLine" id="cb201-4" data-line-number="4"><span class="co"># mitochondrial genes here and store it in percent.mito using AddMetaData.</span></a>
<a class="sourceLine" id="cb201-5" data-line-number="5"><span class="co"># We use object@raw.data since this represents non-transformed and</span></a>
<a class="sourceLine" id="cb201-6" data-line-number="6"><span class="co"># non-log-normalized counts The % of UMI mapping to MT-genes is a common</span></a>
<a class="sourceLine" id="cb201-7" data-line-number="7"><span class="co"># scRNA-seq QC metric.</span></a>
<a class="sourceLine" id="cb201-8" data-line-number="8"><span class="co"># Load the the list of house keeping genes</span></a>
<a class="sourceLine" id="cb201-9" data-line-number="9">hkgenes &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;data/resources/tirosh_house_keeping.txt&quot;</span>, <span class="dt">skip =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb201-10" data-line-number="10">hkgenes &lt;-<span class="st"> </span><span class="kw">as.vector</span>(hkgenes<span class="op">$</span>V1)</a>
<a class="sourceLine" id="cb201-11" data-line-number="11"></a>
<a class="sourceLine" id="cb201-12" data-line-number="12"><span class="co"># remove hkgenes that were not found</span></a>
<a class="sourceLine" id="cb201-13" data-line-number="13">hkgenes.found &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">toupper</span>(<span class="kw">rownames</span>(seurat<span class="op">@</span>assays<span class="op">$</span>RNA<span class="op">@</span>data)) <span class="op">%in%</span><span class="st"> </span>hkgenes)</a>
<a class="sourceLine" id="cb201-14" data-line-number="14">n.expressed.hkgenes &lt;-<span class="st"> </span><span class="kw">sum</span>(seurat<span class="op">@</span>assays<span class="op">$</span>RNA<span class="op">@</span>data[hkgenes.found, ] <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)</a>
<a class="sourceLine" id="cb201-15" data-line-number="15"></a>
<a class="sourceLine" id="cb201-16" data-line-number="16"><span class="co">## Add to Seurat Meta Data</span></a>
<a class="sourceLine" id="cb201-17" data-line-number="17">seurat &lt;-<span class="st"> </span><span class="kw">AddMetaData</span>(<span class="dt">object =</span> seurat, <span class="dt">metadata =</span> n.expressed.hkgenes, <span class="dt">col.name =</span> <span class="st">&quot;n.exp.hkgenes&quot;</span>)</a>
<a class="sourceLine" id="cb201-18" data-line-number="18"></a>
<a class="sourceLine" id="cb201-19" data-line-number="19">seurat[[<span class="st">&quot;percent.mito&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">PercentageFeatureSet</span>(<span class="dt">object =</span> seurat, <span class="dt">pattern =</span> <span class="st">&quot;^MT-&quot;</span>)</a>
<a class="sourceLine" id="cb201-20" data-line-number="20"><span class="co">##VlnPlot(object = seurat, features = c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;, &quot;percent.mt&quot;), ncol = 3)</span></a></code></pre></div>
<div class="sourceCode" id="cb202"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb202-1" data-line-number="1"><span class="co"># FeatureScatter is typically used to visualize gene-gene relationships, but can</span></a>
<a class="sourceLine" id="cb202-2" data-line-number="2"><span class="co"># be used for anything calculated by the object, i.e. columns in</span></a>
<a class="sourceLine" id="cb202-3" data-line-number="3"><span class="co"># object@meta.data, PC scores etc.  Since there is a rare subset of cells</span></a>
<a class="sourceLine" id="cb202-4" data-line-number="4"><span class="co"># with an outlier level of high mitochondrial percentage and also low UMI</span></a>
<a class="sourceLine" id="cb202-5" data-line-number="5"><span class="co"># content, we filter these as well</span></a>
<a class="sourceLine" id="cb202-6" data-line-number="6"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</a>
<a class="sourceLine" id="cb202-7" data-line-number="7"><span class="kw">FeatureScatter</span>(<span class="dt">object =</span> seurat, <span class="dt">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;percent.mito&quot;</span>)</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/scatter_plot_v3-1.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb203"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb203-1" data-line-number="1"><span class="kw">FeatureScatter</span>(<span class="dt">object =</span> seurat, <span class="dt">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;nFeature_RNA&quot;</span>)</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/scatter_plot_v3-2.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb204"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb204-1" data-line-number="1"><span class="co"># We filter out cells that have unique gene counts over 4,000 or less than</span></a>
<a class="sourceLine" id="cb204-2" data-line-number="2"><span class="co"># 350 Note that low thresholds and high thresholds are used to define a</span></a>
<a class="sourceLine" id="cb204-3" data-line-number="3"><span class="co"># &#39;gate&#39;.  -Inf and Inf should be used if you don&#39;t want a lower or upper</span></a>
<a class="sourceLine" id="cb204-4" data-line-number="4"><span class="co"># threshold.</span></a>
<a class="sourceLine" id="cb204-5" data-line-number="5"></a>
<a class="sourceLine" id="cb204-6" data-line-number="6"></a>
<a class="sourceLine" id="cb204-7" data-line-number="7">seurat &lt;-<span class="st"> </span><span class="kw">subset</span>(seurat, <span class="dt">subset =</span> nFeature_RNA <span class="op">&gt;</span><span class="st"> </span><span class="dv">350</span> <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb204-8" data-line-number="8"><span class="st">                   </span>nFeature_RNA <span class="op">&lt;</span><span class="st"> </span><span class="dv">4000</span> <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb204-9" data-line-number="9"><span class="st">                   </span>percent.mito <span class="op">&lt;</span><span class="st"> </span><span class="dv">15</span> <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb204-10" data-line-number="10"><span class="st">                   </span>n.exp.hkgenes <span class="op">&gt;</span><span class="st"> </span><span class="dv">55</span>)</a></code></pre></div>
<div class="sourceCode" id="cb205"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb205-1" data-line-number="1">seurat &lt;-<span class="st"> </span><span class="kw">NormalizeData</span>(<span class="dt">object =</span> seurat, <span class="dt">normalization.method =</span> <span class="st">&quot;LogNormalize&quot;</span>, <span class="dt">scale.factor =</span> <span class="dv">10000</span>)</a></code></pre></div>
<div class="sourceCode" id="cb206"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb206-1" data-line-number="1"><span class="co"># Read in a list of cell cycle markers, from Tirosh et al, 2015.</span></a>
<a class="sourceLine" id="cb206-2" data-line-number="2"><span class="co"># We can segregate this list into markers of G2/M phase and markers of S phase.</span></a>
<a class="sourceLine" id="cb206-3" data-line-number="3">s.genes &lt;-<span class="st"> </span>Seurat<span class="op">::</span>cc.genes<span class="op">$</span>s.genes</a>
<a class="sourceLine" id="cb206-4" data-line-number="4">s.genes &lt;-<span class="st"> </span>s.genes[s.genes <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(seurat)] <span class="co"># genes in dataset</span></a>
<a class="sourceLine" id="cb206-5" data-line-number="5">g2m.genes &lt;-<span class="st"> </span>Seurat<span class="op">::</span>cc.genes<span class="op">$</span>g2m.genes</a>
<a class="sourceLine" id="cb206-6" data-line-number="6">g2m.genes &lt;-<span class="st"> </span>g2m.genes[g2m.genes <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(seurat)] <span class="co"># genes in dataset</span></a>
<a class="sourceLine" id="cb206-7" data-line-number="7"></a>
<a class="sourceLine" id="cb206-8" data-line-number="8">seurat &lt;-<span class="st"> </span><span class="kw">CellCycleScoring</span>(<span class="dt">object =</span> seurat, </a>
<a class="sourceLine" id="cb206-9" data-line-number="9">                           <span class="dt">s.features =</span> s.genes, </a>
<a class="sourceLine" id="cb206-10" data-line-number="10">                           <span class="dt">g2m.features =</span> g2m.genes, </a>
<a class="sourceLine" id="cb206-11" data-line-number="11">                           <span class="dt">set.ident =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<div class="sourceCode" id="cb207"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb207-1" data-line-number="1">seurat &lt;-<span class="st"> </span><span class="kw">FindVariableFeatures</span>(<span class="dt">object =</span> seurat, <span class="dt">selection.method =</span> <span class="st">&quot;vst&quot;</span>, <span class="dt">nfeatures =</span> <span class="dv">2000</span>)</a></code></pre></div>
</div>
<div id="start-of-identifying-cell-types" class="section level3">
<h3><span class="header-section-number">10.2.2</span> Start of Identifying Cell Types</h3>
<div id="scaling" class="section level4">
<h4><span class="header-section-number">10.2.2.1</span> Scaling</h4>
<p>This part is where you mean center the data, substract the mean. You also divide by the standard deviation to make everything to a ‘standard normal’, where the mean is zero and the standard deviation is 1.</p>
<div class="sourceCode" id="cb208"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb208-1" data-line-number="1">seurat &lt;-<span class="st"> </span><span class="kw">ScaleData</span>(<span class="dt">object =</span> seurat, <span class="dt">vars.to.regress =</span> <span class="kw">c</span>(<span class="st">&quot;percent.mito&quot;</span>))</a></code></pre></div>
<pre><code>## Regressing out percent.mito</code></pre>
<pre><code>## Centering and scaling data matrix</code></pre>
<p><strong>Task: Try Regressing Other Variables</strong></p>
<div class="sourceCode" id="cb211"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb211-1" data-line-number="1"><span class="kw">set.seed</span>(<span class="dv">2020</span>) <span class="co">## used for reporducibility</span></a>
<a class="sourceLine" id="cb211-2" data-line-number="2"></a>
<a class="sourceLine" id="cb211-3" data-line-number="3"><span class="co">## randomly making a batch id data.frame</span></a>
<a class="sourceLine" id="cb211-4" data-line-number="4">batch_ids &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">barcode =</span> <span class="kw">rownames</span>(seurat<span class="op">@</span>meta.data), </a>
<a class="sourceLine" id="cb211-5" data-line-number="5">                        <span class="dt">batch_id =</span> <span class="kw">sample</span>(<span class="dv">0</span><span class="op">:</span><span class="dv">2</span>, <span class="kw">NROW</span>(seurat<span class="op">@</span>meta.data), <span class="dt">replace =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb211-6" data-line-number="6">                        <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb211-7" data-line-number="7"></a>
<a class="sourceLine" id="cb211-8" data-line-number="8"><span class="co">## naming batch ids by cell names</span></a>
<a class="sourceLine" id="cb211-9" data-line-number="9"><span class="kw">row.names</span>(batch_ids) &lt;-<span class="st"> </span><span class="kw">row.names</span>(seurat<span class="op">@</span>meta.data)</a>
<a class="sourceLine" id="cb211-10" data-line-number="10"></a>
<a class="sourceLine" id="cb211-11" data-line-number="11"><span class="co">## adding batch ids to the meta data</span></a>
<a class="sourceLine" id="cb211-12" data-line-number="12">seurat &lt;-<span class="st"> </span><span class="kw">AddMetaData</span>(<span class="dt">object =</span> seurat, <span class="dt">metadata =</span> batch_ids, <span class="dt">col.name =</span> <span class="ot">NULL</span>)</a>
<a class="sourceLine" id="cb211-13" data-line-number="13"></a>
<a class="sourceLine" id="cb211-14" data-line-number="14"></a>
<a class="sourceLine" id="cb211-15" data-line-number="15"><span class="co">## try different variables to regress</span></a>
<a class="sourceLine" id="cb211-16" data-line-number="16">seurat &lt;-<span class="st"> </span><span class="kw">ScaleData</span>(<span class="dt">object =</span> seurat, <span class="dt">vars.to.regress =</span> <span class="st">&#39;??&#39;</span>)</a></code></pre></div>
</div>
<div id="perform-linear-dimensional-reduction-pca" class="section level4">
<h4><span class="header-section-number">10.2.2.2</span> Perform linear dimensional reduction (PCA)</h4>
<p>This will run pca on the just the variable features found.</p>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb212-1" data-line-number="1">seurat &lt;-<span class="st"> </span><span class="kw">RunPCA</span>(<span class="dt">object =</span> seurat, </a>
<a class="sourceLine" id="cb212-2" data-line-number="2">               <span class="dt">features =</span> seurat<span class="op">@</span>assays<span class="op">$</span>RNA<span class="op">@</span>var.features, </a>
<a class="sourceLine" id="cb212-3" data-line-number="3">               <span class="dt">ndims.print =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, </a>
<a class="sourceLine" id="cb212-4" data-line-number="4">               <span class="dt">nfeatures.print =</span> <span class="dv">5</span>)</a></code></pre></div>
<pre><code>## PC_ 1 
## Positive:  CD74, IRF8, IGHA1, CCR7, TNF 
## Negative:  IFITM3, CST3, KRT7, S100A11, S100A9 
## PC_ 2 
## Positive:  LYZ, AIF1, TYROBP, FCER1G, SERPINA1 
## Negative:  KRT7, KRT17, SLPI, FXYD3, C19orf33 
## PC_ 3 
## Positive:  HLA-DRA, CD74, HLA-DRB1, HLA-DQB1, HLA-DPA1 
## Negative:  TPSD1, CPA3, TPSAB1, TPSB2, MS4A2 
## PC_ 4 
## Positive:  HLA-DQA1, HLA-DRA, HLA-DQB1, HLA-DPA1, HLA-DPB1 
## Negative:  IL32, CD7, GZMA, NKG7, PRF1 
## PC_ 5 
## Positive:  S100A12, SPRR1B, S100A8, SPRR2A, RDH10 
## Negative:  IGFBP7, GAS6, SFRP1, CALD1, HTRA1</code></pre>
</div>
<div id="visualizing-pca-in-different-ways" class="section level4">
<h4><span class="header-section-number">10.2.2.3</span> Visualizing PCA in Different Ways</h4>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb214-1" data-line-number="1"><span class="kw">DimPlot</span>(seurat, <span class="dt">reduction =</span> <span class="st">&quot;pca&quot;</span>)</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/plot_pca-1.png" width="90%" style="display: block; margin: auto;" /></p>
</div>
<div id="perform-linear-dimensional-reduction-ica" class="section level4">
<h4><span class="header-section-number">10.2.2.4</span> Perform linear dimensional reduction (ICA)</h4>
<p><strong>Task:</strong> Try running Independent Component Analysis. If you need help with the inputs try using the ?RunICA menu.</p>
<div class="sourceCode" id="cb215"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb215-1" data-line-number="1">seurat &lt;-<span class="st"> </span><span class="kw">RunICA</span>()</a></code></pre></div>
</div>
<div id="visualizing-ica-in-different-ways" class="section level4">
<h4><span class="header-section-number">10.2.2.5</span> Visualizing ICA in Different Ways</h4>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb216-1" data-line-number="1"><span class="kw">DimPlot</span>()</a></code></pre></div>
<div class="sourceCode" id="cb217"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb217-1" data-line-number="1"><span class="co"># ProjectDim scores each gene in the dataset (including genes not included</span></a>
<a class="sourceLine" id="cb217-2" data-line-number="2"><span class="co"># in the PCA) based on their correlation with the calculated components.</span></a>
<a class="sourceLine" id="cb217-3" data-line-number="3"><span class="co"># Though we don&#39;t use this further here, it can be used to identify markers</span></a>
<a class="sourceLine" id="cb217-4" data-line-number="4"><span class="co"># that are strongly correlated with cellular heterogeneity, but may not have</span></a>
<a class="sourceLine" id="cb217-5" data-line-number="5"><span class="co"># passed through variable gene selection.  The results of the projected PCA</span></a>
<a class="sourceLine" id="cb217-6" data-line-number="6"><span class="co"># can be explored by setting use.full=T in the functions above</span></a>
<a class="sourceLine" id="cb217-7" data-line-number="7">seurat &lt;-<span class="st"> </span><span class="kw">ProjectDim</span>(<span class="dt">object =</span> seurat, <span class="dt">reduction =</span> <span class="st">&quot;pca&quot;</span>)</a></code></pre></div>
<pre><code>## PC_ 1 
## Positive:  CD74, IRF8, IGHA1, CCR7, TNF, ITM2A, CCL5, RP5-887A10.1, RP11-731F5.2, PIM2 
##     CD7, HERPUD1, HLA-DQA1, KIAA0125, STAG3, GZMA, KLRB1, HLA-DPA1, AC079767.4, IGKC 
## Negative:  IFITM3, CST3, KRT7, S100A11, S100A9, TACSTD2, SLPI, GSTP1, LGALS3, APLP2 
##     CLDN7, ELF3, GLUL, KRT19, SERINC2, PTGES, S100P, RAC1, FXYD3, S100A6 
## PC_ 2 
## Positive:  LYZ, AIF1, TYROBP, FCER1G, SERPINA1, FCGR2A, MS4A6A, CD14, CSF1R, CPVL 
##     SPI1, CD68, FPR1, FTL, FCGRT, CD163, PILRA, C1orf162, CLEC7A, LILRB2 
## Negative:  KRT7, KRT17, SLPI, FXYD3, C19orf33, PERP, ELF3, S100P, TACSTD2, PTGES 
##     KRT19, RPS18, DSP, LCN2, KRT18, KLF5, CLDN7, SERPINB3, AQP3, GPRC5A 
## PC_ 3 
## Positive:  HLA-DRA, CD74, HLA-DRB1, HLA-DQB1, HLA-DPA1, HLA-DQA1, HLA-DPB1, IRF8, HLA-DRB5, CD24 
##     CYBB, RAB31, FCGR2B, RNASE6, KYNU, S100A12, FCN1, S100A8, S100A9, BCL2A1 
## Negative:  TPSD1, CPA3, TPSAB1, TPSB2, MS4A2, GATA2, HPGDS, SLC18A2, KIT, HDC 
##     RP11-354E11.2, IL1RL1, VWA5A, CNRIP1, RP11-32B5.7, MAOB, CLU, LTC4S, HPGD, RGS13 
## PC_ 4 
## Positive:  HLA-DQA1, HLA-DRA, HLA-DQB1, HLA-DPA1, HLA-DPB1, HLA-DRB1, IRF8, CD74, HLA-DRB5, CD24 
##     C1orf186, ALOX5, FCGR2B, CD1C, IGHA1, RP5-887A10.1, RP11-731F5.2, HPGDS, DERL3, SLC18A2 
## Negative:  IL32, CD7, GZMA, NKG7, PRF1, CCL5, CTSW, CST7, KLRB1, CD8A 
##     IL2RB, CCL4, KLRD1, GNLY, ID2, MT2A, TRBC1, TIGIT, SIRPG, GZMH 
## PC_ 5 
## Positive:  S100A12, SPRR1B, S100A8, SPRR2A, RDH10, MUC4, PDZK1IP1, ERO1A, ADGRF1, PRSS22 
##     MALAT1, RNF39, S100A9, SLC6A14, SPRR3, SCEL, DHRS9, SLC26A9, ELF3, TMPRSS11D 
## Negative:  IGFBP7, GAS6, SFRP1, CALD1, HTRA1, SOD3, NNMT, IGFBP4, MSLN, SERPINF1 
##     IGFBP2, CAV1, PCSK1, FBLN1, BCAM, DST, ADIRF, CPE, TUSC3, PPP1R14A</code></pre>
</div>
<div id="genes-by-pcs" class="section level4">
<h4><span class="header-section-number">10.2.2.6</span> Genes by PCs</h4>
<div class="sourceCode" id="cb219"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb219-1" data-line-number="1"><span class="kw">DimHeatmap</span>(<span class="dt">object =</span> seurat, </a>
<a class="sourceLine" id="cb219-2" data-line-number="2">          <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">6</span>, </a>
<a class="sourceLine" id="cb219-3" data-line-number="3">          <span class="dt">cells =</span> <span class="dv">50</span>, </a>
<a class="sourceLine" id="cb219-4" data-line-number="4">          <span class="dt">reduction =</span> <span class="st">&quot;pca&quot;</span>,</a>
<a class="sourceLine" id="cb219-5" data-line-number="5">          <span class="dt">balanced =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/pca_heatmap-1.png" width="90%" style="display: block; margin: auto;" /></p>
<p>Check other PCs to plot</p>
<p><strong>Task: Check other PCs</strong></p>
<div class="sourceCode" id="cb220"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb220-1" data-line-number="1"><span class="kw">DimHeatmap</span>()</a></code></pre></div>
<div class="sourceCode" id="cb221"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb221-1" data-line-number="1">seurat &lt;-<span class="st"> </span><span class="kw">JackStraw</span>(<span class="dt">object =</span> seurat, <span class="dt">reduction =</span> <span class="st">&quot;pca&quot;</span>)</a>
<a class="sourceLine" id="cb221-2" data-line-number="2">seurat &lt;-<span class="st"> </span><span class="kw">ScoreJackStraw</span>(seurat, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">20</span>)</a></code></pre></div>
<div class="sourceCode" id="cb222"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb222-1" data-line-number="1"><span class="kw">JackStrawPlot</span>(seurat, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">20</span>)</a></code></pre></div>
<pre><code>## Warning: Removed 28000 rows containing missing values (geom_point).</code></pre>
<p><img src="2020_scWorkshop_files/figure-html/jack_straw_plot-1.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb224"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb224-1" data-line-number="1"><span class="kw">PCASigGenes</span>(<span class="dt">object =</span> seurat, <span class="dt">pcs.use =</span> <span class="dv">1</span>, <span class="dt">pval.cut =</span> <span class="fl">0.001</span>)[<span class="dv">1</span><span class="op">:</span><span class="dv">20</span>]</a></code></pre></div>
<pre><code>##  [1] &quot;JCHAIN&quot;   &quot;IGHA1&quot;    &quot;IGLC2&quot;    &quot;HLA-DQB2&quot; &quot;APOE&quot;     &quot;S100A7&quot;   &quot;S100B&quot;    &quot;IGHG1&quot;    &quot;APOC1&quot;   
## [10] &quot;SAA1&quot;     &quot;SPP1&quot;     &quot;IGKC&quot;     &quot;IGHV3-23&quot; &quot;IGHM&quot;     &quot;MSMB&quot;     &quot;SERPINB4&quot; &quot;MSLN&quot;     &quot;IGLC3&quot;   
## [19] &quot;PI3&quot;      &quot;SPRR2A&quot;</code></pre>
<div class="sourceCode" id="cb226"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb226-1" data-line-number="1"><span class="kw">ElbowPlot</span>(<span class="dt">object =</span> seurat, <span class="dt">ndims =</span> <span class="dv">30</span>, <span class="dt">reduction =</span> <span class="st">&quot;pca&quot;</span>)</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/pca_elbow-1.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb227"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb227-1" data-line-number="1"><span class="co"># save.SNN = T saves the SNN so that the clustering algorithm can be rerun</span></a>
<a class="sourceLine" id="cb227-2" data-line-number="2"><span class="co"># using the same graph but with a different resolution value (see docs for</span></a>
<a class="sourceLine" id="cb227-3" data-line-number="3"><span class="co"># full details)</span></a>
<a class="sourceLine" id="cb227-4" data-line-number="4"><span class="kw">set.seed</span>(<span class="dv">2020</span>)</a>
<a class="sourceLine" id="cb227-5" data-line-number="5">seurat &lt;-<span class="st"> </span><span class="kw">FindNeighbors</span>(<span class="dt">object =</span> seurat, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>)</a></code></pre></div>
<pre><code>## Computing nearest neighbor graph</code></pre>
<pre><code>## Computing SNN</code></pre>
<div class="sourceCode" id="cb230"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb230-1" data-line-number="1">seurat &lt;-<span class="st"> </span><span class="kw">FindClusters</span>(<span class="dt">object =</span> seurat, </a>
<a class="sourceLine" id="cb230-2" data-line-number="2">                       <span class="dt">reduction =</span> <span class="st">&quot;pca&quot;</span>, </a>
<a class="sourceLine" id="cb230-3" data-line-number="3">                       <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, </a>
<a class="sourceLine" id="cb230-4" data-line-number="4">                       <span class="dt">resolution =</span> <span class="fl">0.6</span>,</a>
<a class="sourceLine" id="cb230-5" data-line-number="5">                       <span class="dt">random.seed =</span> <span class="dv">2020</span>)</a></code></pre></div>
<pre><code>## Warning: The following arguments are not used: reduction, dims

## Warning: The following arguments are not used: reduction, dims</code></pre>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 6187
## Number of edges: 202352
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8875
## Number of communities: 15
## Elapsed time: 0 seconds</code></pre>
</div>
</div>
<div id="run-non-linear-dimensional-reduction-umaptsne" class="section level3">
<h3><span class="header-section-number">10.2.3</span> Run non-linear dimensional reduction (UMAP/tSNE)</h3>
<p>Seurat offers several non-linear dimensional reduction techniques, such as tSNE and UMAP, to visualize and explore these datasets. The goal of these algorithms is to learn the underlying manifold of the data in order to place similar cells together in low-dimensional space. Cells within the graph-based clusters determined above should co-localize on these dimension reduction plots. As input to the UMAP and tSNE, we suggest using the same PCs as input to the clustering analysis.</p>
<p><strong>Task: Look up and tune hyperparameters of the tSNE. (hint: <code>?RunTSNE</code>)</strong></p>
<div class="sourceCode" id="cb233"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb233-1" data-line-number="1">seurat &lt;-<span class="st"> </span><span class="kw">RunTSNE</span>(seurat, <span class="dt">reduction.use =</span> <span class="st">&quot;pca&quot;</span>, <span class="dt">dims.use =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="dt">perplexity=</span><span class="dv">10</span>)</a>
<a class="sourceLine" id="cb233-2" data-line-number="2"></a>
<a class="sourceLine" id="cb233-3" data-line-number="3"><span class="co"># note that you can set do.label=T to help label individual clusters</span></a>
<a class="sourceLine" id="cb233-4" data-line-number="4"><span class="kw">DimPlot</span>(<span class="dt">object =</span> seurat, <span class="dt">reduction =</span> <span class="st">&quot;tsne&quot;</span>)</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/run_tsne-1.png" width="90%" style="display: block; margin: auto;" /></p>
<p><strong>Task: Try using UMAP for the non-linear dimension reduction technique (hint: <code>?RunUMAP</code>)</strong></p>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb234-1" data-line-number="1"><span class="kw">set.seed</span>(<span class="dv">2020</span>)</a>
<a class="sourceLine" id="cb234-2" data-line-number="2">seurat &lt;-<span class="st"> </span><span class="kw">RunUMAP</span>()</a>
<a class="sourceLine" id="cb234-3" data-line-number="3"></a>
<a class="sourceLine" id="cb234-4" data-line-number="4"><span class="co"># note that you can set label=TRUE to help label individual clusters</span></a>
<a class="sourceLine" id="cb234-5" data-line-number="5"><span class="kw">DimPlot</span>()</a></code></pre></div>
<div id="finding-differentially-expressed-features-cluster-biomarkers" class="section level4">
<h4><span class="header-section-number">10.2.3.1</span> Finding differentially expressed features (cluster biomarkers)</h4>
<p>Seurat can help you find markers that define clusters via differential expression. By default, it identifes positive and negative markers of a single cluster (specified in ident.1), compared to all other cells. FindAllMarkers automates this process for all clusters, but you can also test groups of clusters vs. each other, or against all cells.</p>
<p>The min.pct argument requires a feature to be detected at a minimum percentage in either of the two groups of cells, and the thresh.test argument requires a feature to be differentially expressed (on average) by some amount between the two groups. You can set both of these to 0, but with a dramatic increase in time - since this will test a large number of features that are unlikely to be highly discriminatory. As another option to speed up these computations, max.cells.per.ident can be set. This will downsample each identity class to have no more cells than whatever this is set to. While there is generally going to be a loss in power, the speed increases can be significiant and the most highly differentially expressed features will likely still rise to the top.</p>
<div class="sourceCode" id="cb235"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb235-1" data-line-number="1"><span class="co"># find all markers of cluster 1 using default parameters</span></a>
<a class="sourceLine" id="cb235-2" data-line-number="2">cluster1.markers &lt;-<span class="st"> </span><span class="kw">FindMarkers</span>(<span class="dt">object =</span> seurat, </a>
<a class="sourceLine" id="cb235-3" data-line-number="3">                                <span class="dt">ident.1 =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb235-4" data-line-number="4">                                <span class="dt">min.pct =</span> <span class="fl">0.1</span>)</a>
<a class="sourceLine" id="cb235-5" data-line-number="5"></a>
<a class="sourceLine" id="cb235-6" data-line-number="6"><span class="kw">head</span>(cluster1.markers)</a></code></pre></div>
<pre><code>##                p_val avg_logFC pct.1 pct.2     p_val_adj
## IL7R    0.000000e+00 1.7849385 0.913 0.201  0.000000e+00
## CD3E    0.000000e+00 1.1829350 0.945 0.215  0.000000e+00
## IL32    0.000000e+00 1.1492149 0.953 0.274  0.000000e+00
## CD3D    0.000000e+00 1.0278812 0.858 0.177  0.000000e+00
## CD40LG 3.373243e-278 0.8890083 0.401 0.036 6.818336e-274
## GIMAP4 3.823325e-256 1.0141440 0.655 0.160 7.728087e-252</code></pre>
<p><strong>Task: Try tuning different parameters. How does that affect results?</strong></p>
<div class="sourceCode" id="cb237"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb237-1" data-line-number="1"><span class="co"># find all markers distinguishing cluster 5 from clusters 0 and 1</span></a>
<a class="sourceLine" id="cb237-2" data-line-number="2">cluster5.markers &lt;-<span class="st"> </span><span class="kw">FindMarkers</span>(<span class="dt">object =</span> seurat, </a>
<a class="sourceLine" id="cb237-3" data-line-number="3">                                <span class="dt">ident.1 =</span> <span class="dv">5</span>, <span class="dt">ident.2 =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), </a>
<a class="sourceLine" id="cb237-4" data-line-number="4">                                <span class="dt">min.pct =</span> ?? </a>
<a class="sourceLine" id="cb237-5" data-line-number="5">                                <span class="dt">only.pos =</span> ??)</a>
<a class="sourceLine" id="cb237-6" data-line-number="6"></a>
<a class="sourceLine" id="cb237-7" data-line-number="7"><span class="kw">head</span>(cluster5.markers)</a></code></pre></div>
<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb238-1" data-line-number="1">cluster3.markers &lt;-<span class="st"> </span><span class="kw">FindMarkers</span>(<span class="dt">object =</span> seurat, </a>
<a class="sourceLine" id="cb238-2" data-line-number="2">                                <span class="dt">ident.1 =</span> <span class="dv">3</span>, </a>
<a class="sourceLine" id="cb238-3" data-line-number="3">                                <span class="dt">thresh.use =</span> <span class="fl">0.25</span>, </a>
<a class="sourceLine" id="cb238-4" data-line-number="4">                                <span class="dt">only.pos =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb238-5" data-line-number="5"></a>
<a class="sourceLine" id="cb238-6" data-line-number="6"><span class="kw">head</span>(cluster3.markers)</a></code></pre></div>
<div class="sourceCode" id="cb239"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb239-1" data-line-number="1"><span class="kw">VlnPlot</span>(<span class="dt">object =</span> seurat, <span class="dt">features =</span> <span class="kw">c</span>(<span class="st">&quot;MS4A1&quot;</span>, <span class="st">&quot;CD79A&quot;</span>))</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/plot_cluster3_markers-1.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb240"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb240-1" data-line-number="1"><span class="co"># you can plot raw UMI counts as well</span></a>
<a class="sourceLine" id="cb240-2" data-line-number="2"><span class="kw">VlnPlot</span>(<span class="dt">object =</span> seurat, </a>
<a class="sourceLine" id="cb240-3" data-line-number="3">        <span class="dt">features =</span> <span class="kw">c</span>(<span class="st">&quot;NKG7&quot;</span>, <span class="st">&quot;PF4&quot;</span>), </a>
<a class="sourceLine" id="cb240-4" data-line-number="4">        <span class="dt">log =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/plot_by_umi-1.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb241"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb241-1" data-line-number="1"><span class="co"># find markers for every cluster compared to all remaining cells, report</span></a>
<a class="sourceLine" id="cb241-2" data-line-number="2"><span class="co"># only the positive ones</span></a>
<a class="sourceLine" id="cb241-3" data-line-number="3">nsclc.markers &lt;-<span class="st"> </span><span class="kw">FindAllMarkers</span>(<span class="dt">object =</span> seurat, <span class="dt">only.pos =</span> <span class="ot">TRUE</span>, <span class="dt">min.pct =</span> <span class="fl">0.25</span>, <span class="dt">thresh.use =</span> <span class="fl">0.25</span>)</a></code></pre></div>
<pre><code>## Calculating cluster 0</code></pre>
<pre><code>## Calculating cluster 1</code></pre>
<pre><code>## Calculating cluster 2</code></pre>
<pre><code>## Calculating cluster 3</code></pre>
<pre><code>## Calculating cluster 4</code></pre>
<pre><code>## Calculating cluster 5</code></pre>
<pre><code>## Calculating cluster 6</code></pre>
<pre><code>## Calculating cluster 7</code></pre>
<pre><code>## Calculating cluster 8</code></pre>
<pre><code>## Calculating cluster 9</code></pre>
<pre><code>## Calculating cluster 10</code></pre>
<pre><code>## Calculating cluster 11</code></pre>
<pre><code>## Calculating cluster 12</code></pre>
<pre><code>## Calculating cluster 13</code></pre>
<pre><code>## Calculating cluster 14</code></pre>
<div class="sourceCode" id="cb257"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb257-1" data-line-number="1">nsclc.markers <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(cluster) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">top_n</span>(<span class="dv">2</span>, avg_logFC)</a></code></pre></div>
<pre><code>## # A tibble: 30 x 7
## # Groups:   cluster [15]
##        p_val avg_logFC pct.1 pct.2 p_val_adj cluster gene 
##        &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;fct&gt;   &lt;chr&gt;
##  1 0.            1.42  0.98  0.189 0.        0       MS4A1
##  2 0.            1.36  0.981 0.219 0.        0       CD79A
##  3 0.            1.78  0.913 0.201 0.        1       IL7R 
##  4 1.48e-245     1.20  0.568 0.11  2.99e-241 1       KLRB1
##  5 7.48e-218     1.09  0.921 0.321 1.51e-213 2       MS4A1
##  6 1.51e-189     0.961 0.907 0.35  3.05e-185 2       CD79A
##  7 5.82e-263     2.00  0.468 0.054 1.18e-258 3       GZMK 
##  8 2.59e-236     1.81  0.692 0.178 5.24e-232 3       CCL5 
##  9 0.            3.39  0.909 0.128 0.        4       KRT17
## 10 0.            3.37  0.834 0.047 0.        4       SLPI 
## # … with 20 more rows</code></pre>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb259-1" data-line-number="1"><span class="kw">FeaturePlot</span>(<span class="dt">object =</span> seurat, </a>
<a class="sourceLine" id="cb259-2" data-line-number="2">            <span class="dt">features =</span> <span class="kw">c</span>(<span class="st">&quot;MS4A1&quot;</span>, <span class="st">&quot;GNLY&quot;</span>, <span class="st">&quot;CD3E&quot;</span>, <span class="st">&quot;CD14&quot;</span>, <span class="st">&quot;FCER1A&quot;</span>, <span class="st">&quot;FCGR3A&quot;</span>, <span class="st">&quot;LYZ&quot;</span>, <span class="st">&quot;PPBP&quot;</span>, <span class="st">&quot;CD8A&quot;</span>),</a>
<a class="sourceLine" id="cb259-3" data-line-number="3">            <span class="dt">cols =</span> <span class="kw">c</span>(<span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>), </a>
<a class="sourceLine" id="cb259-4" data-line-number="4">            <span class="dt">reduction =</span> <span class="st">&quot;tsne&quot;</span>)</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/gene_tsne-1.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb260"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb260-1" data-line-number="1">top10 &lt;-<span class="st"> </span>nsclc.markers <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(cluster) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">top_n</span>(<span class="dv">10</span>, avg_logFC)</a>
<a class="sourceLine" id="cb260-2" data-line-number="2"><span class="co"># setting slim.col.label to TRUE will print just the cluster IDS instead of</span></a>
<a class="sourceLine" id="cb260-3" data-line-number="3"><span class="co"># every cell name</span></a>
<a class="sourceLine" id="cb260-4" data-line-number="4"><span class="kw">DoHeatmap</span>(<span class="dt">object =</span> seurat, <span class="dt">features =</span> top10<span class="op">$</span>gene, <span class="dt">label =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## Warning in DoHeatmap(object = seurat, features = top10$gene, label = TRUE): The following features were
## omitted as they were not found in the scale.data slot for the RNA assay: PTPRCAP, SEPT6, ENTPD1, DUSP2, RCSD1,
## PCSK7, AIM1, TNFAIP3, PRKCH, CD6, ITK, CD2, SCIMP, AIM2, POU2AF1, TNFRSF13B, LDHB, LEPROTL1, FYB, GIMAP7,
## GIMAP4, CD3D, CD3E, IL7R, SMIM14, CD19, LINC00926, CD37, FCRLA, CD79B, BANK1, CD79A, MS4A1</code></pre>
<p><img src="2020_scWorkshop_files/figure-html/do_heatmap-1.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb262"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb262-1" data-line-number="1"><span class="co">## pbmc markers</span></a>
<a class="sourceLine" id="cb262-2" data-line-number="2"><span class="co"># new.cluster.ids &lt;- c(&quot;Memory CD4 T&quot;, &quot;Naive CD4 T&quot;, &quot;CD14+ Mono&quot;, &quot;B&quot;, &quot;CD8 T&quot;, </span></a>
<a class="sourceLine" id="cb262-3" data-line-number="3"><span class="co">#                      &quot;FCGR3A+ Mono&quot;, &quot;NK&quot;, &quot;DC&quot;, &quot;Mk&quot;)</span></a>
<a class="sourceLine" id="cb262-4" data-line-number="4"></a>
<a class="sourceLine" id="cb262-5" data-line-number="5"><span class="co">## making generic cell type names</span></a>
<a class="sourceLine" id="cb262-6" data-line-number="6">new.cluster.ids &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;CellType&quot;</span>, <span class="kw">levels</span>(seurat<span class="op">@</span>active.ident))</a>
<a class="sourceLine" id="cb262-7" data-line-number="7"><span class="kw">names</span>(<span class="dt">x =</span> new.cluster.ids) &lt;-<span class="st"> </span><span class="kw">levels</span>(<span class="dt">x =</span> seurat)</a>
<a class="sourceLine" id="cb262-8" data-line-number="8">seurat &lt;-<span class="st"> </span><span class="kw">RenameIdents</span>(<span class="dt">object =</span> seurat, new.cluster.ids)</a>
<a class="sourceLine" id="cb262-9" data-line-number="9"></a>
<a class="sourceLine" id="cb262-10" data-line-number="10"><span class="kw">DimPlot</span>(<span class="dt">object =</span> seurat, <span class="dt">reduction =</span> <span class="st">&#39;tsne&#39;</span>, <span class="dt">label =</span> <span class="ot">TRUE</span>, <span class="dt">pt.size =</span> <span class="fl">0.5</span>) <span class="op">+</span><span class="st"> </span><span class="kw">NoLegend</span>()</a></code></pre></div>
<pre><code>## Warning: Using `as.character()` on a quosure is deprecated as of rlang 0.3.0.
## Please use `as_label()` or `as_name()` instead.
## This warning is displayed once per session.</code></pre>
<p><img src="2020_scWorkshop_files/figure-html/new_names-1.png" width="90%" style="display: block; margin: auto;" /></p>
</div>
<div id="further-subdivisions-within-cell-types" class="section level4">
<h4><span class="header-section-number">10.2.3.2</span> Further subdivisions within cell types</h4>
<p>If you perturb some of our parameter choices above (for example, setting resolution=0.8 or changing the number of PCs), you might see cells subdivide into two groups. You can explore this subdivision to find markers separating the two cell subsets. However, before reclustering (which will overwrite <a href="mailto:object@ident" class="email">object@ident</a>), we can stash our renamed identities to be easily recovered later.</p>
<div class="sourceCode" id="cb264"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb264-1" data-line-number="1"><span class="co"># First lets stash our identities for later</span></a>
<a class="sourceLine" id="cb264-2" data-line-number="2">seurat[[<span class="st">&quot;ClusterNames_0.6&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">Idents</span>(<span class="dt">object =</span> seurat)</a>
<a class="sourceLine" id="cb264-3" data-line-number="3"></a>
<a class="sourceLine" id="cb264-4" data-line-number="4"><span class="co"># Note that if you set save.snn=T above, you don&#39;t need to recalculate the</span></a>
<a class="sourceLine" id="cb264-5" data-line-number="5"><span class="co"># SNN, and can simply put: seurat &lt;- FindClusters(seurat, resolution = 0.8)</span></a>
<a class="sourceLine" id="cb264-6" data-line-number="6">seurat &lt;-<span class="st"> </span><span class="kw">FindClusters</span>(<span class="dt">object =</span> seurat, </a>
<a class="sourceLine" id="cb264-7" data-line-number="7">                       <span class="dt">reduction =</span> <span class="st">&quot;pca&quot;</span>, </a>
<a class="sourceLine" id="cb264-8" data-line-number="8">                       <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, </a>
<a class="sourceLine" id="cb264-9" data-line-number="9">                       <span class="dt">resolution =</span> <span class="fl">0.8</span>)</a></code></pre></div>
<pre><code>## Warning: The following arguments are not used: reduction, dims

## Warning: The following arguments are not used: reduction, dims</code></pre>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 6187
## Number of edges: 202352
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8664
## Number of communities: 15
## Elapsed time: 0 seconds</code></pre>
<div class="sourceCode" id="cb267"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb267-1" data-line-number="1"><span class="co">## Warning in BuildSNN(object = object, genes.use = genes.use, reduction.type</span></a>
<a class="sourceLine" id="cb267-2" data-line-number="2"><span class="co">## = reduction.type, : Build parameters exactly match those of already</span></a>
<a class="sourceLine" id="cb267-3" data-line-number="3"><span class="co">## computed and stored SNN. To force recalculation, set force.recalc to TRUE.</span></a>
<a class="sourceLine" id="cb267-4" data-line-number="4"><span class="co"># Demonstration of how to plot two tSNE plots side by side, and how to color</span></a>
<a class="sourceLine" id="cb267-5" data-line-number="5"><span class="co"># points based on different criteria</span></a>
<a class="sourceLine" id="cb267-6" data-line-number="6">plot1 &lt;-<span class="st"> </span><span class="kw">DimPlot</span>(<span class="dt">object =</span> seurat, </a>
<a class="sourceLine" id="cb267-7" data-line-number="7">                 <span class="dt">reduction=</span> <span class="st">&quot;tsne&quot;</span>,</a>
<a class="sourceLine" id="cb267-8" data-line-number="8">                 <span class="dt">label =</span> <span class="ot">TRUE</span>) <span class="op">+</span><span class="st"> </span><span class="kw">NoLegend</span>()</a>
<a class="sourceLine" id="cb267-9" data-line-number="9"></a>
<a class="sourceLine" id="cb267-10" data-line-number="10">plot2 &lt;-<span class="st"> </span><span class="kw">DimPlot</span>(<span class="dt">object =</span> seurat, </a>
<a class="sourceLine" id="cb267-11" data-line-number="11">                  <span class="dt">reduction =</span> <span class="st">&quot;tsne&quot;</span>, </a>
<a class="sourceLine" id="cb267-12" data-line-number="12">                  <span class="dt">group.by =</span> <span class="st">&quot;ClusterNames_0.6&quot;</span>,</a>
<a class="sourceLine" id="cb267-13" data-line-number="13">                  <span class="dt">label =</span> <span class="ot">TRUE</span>) <span class="op">+</span><span class="st"> </span><span class="kw">NoLegend</span>()</a>
<a class="sourceLine" id="cb267-14" data-line-number="14"></a>
<a class="sourceLine" id="cb267-15" data-line-number="15"><span class="co"># patchwork system</span></a>
<a class="sourceLine" id="cb267-16" data-line-number="16">plot1 <span class="op">+</span><span class="st"> </span>plot2</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/compare_params-1.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb268"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb268-1" data-line-number="1"><span class="co"># Find discriminating markers</span></a>
<a class="sourceLine" id="cb268-2" data-line-number="2">cell.markers &lt;-<span class="st"> </span><span class="kw">FindMarkers</span>(<span class="dt">object =</span> seurat, <span class="dt">ident.1 =</span> <span class="dv">0</span>, <span class="dt">ident.2 =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb268-3" data-line-number="3"></a>
<a class="sourceLine" id="cb268-4" data-line-number="4"><span class="co"># Checking some markers found</span></a>
<a class="sourceLine" id="cb268-5" data-line-number="5"><span class="kw">FeaturePlot</span>(<span class="dt">object =</span> seurat, <span class="dt">features =</span> <span class="kw">c</span>(<span class="st">&quot;S100A4&quot;</span>, <span class="st">&quot;CCR7&quot;</span>), <span class="dt">cols =</span> <span class="kw">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;blue&quot;</span>))</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/cell_markers-1.png" width="90%" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="feature-selection" class="section level2">
<h2><span class="header-section-number">10.3</span> Feature Selection</h2>
<div id="differential-expression-analysis" class="section level3">
<h3><span class="header-section-number">10.3.1</span> Differential Expression Analysis</h3>
<div id="differential-expression-tests" class="section level4">
<h4><span class="header-section-number">10.3.1.1</span> Differential Expression Tests</h4>
<p>One of the most commonly performed tasks for RNA-seq data is differential gene expression (DE) analysis. Although well-established tools exist for such analysis in bulk RNA-seq data, methods for scRNA-seq data are just emerging. Given the special characteristics of scRNA-seq data, including generally low library sizes, high noise levels and a large fraction of so-called ‘dropout’ events, it is unclear whether DE methods that have been developed for bulk RNA-seq are suitable also for scRNA-seq. Check the help page out for the FindMarkers function by using <code> ?FindMarkers </code></p>
<div class="sourceCode" id="cb269"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb269-1" data-line-number="1"><span class="co">## Differential expression using t-test</span></a>
<a class="sourceLine" id="cb269-2" data-line-number="2"><span class="kw">FindMarkers</span>(<span class="dt">object =</span> seurat, <span class="dt">ident.1 =</span> <span class="dv">0</span>, <span class="dt">ident.2 =</span> <span class="dv">1</span>, <span class="dt">test.use =</span> <span class="st">&quot;t&quot;</span>)</a></code></pre></div>
<p><strong>Task:</strong> Try to use different test for diffential expression analysis (hint: <code>?FindMarkers</code>)</p>
<div class="sourceCode" id="cb270"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb270-1" data-line-number="1"><span class="co">## Use the help function and run other tests.  Do they find similar markers?</span></a>
<a class="sourceLine" id="cb270-2" data-line-number="2"><span class="kw">FindMarkers</span>(<span class="dt">object =</span> seurat, <span class="dt">ident.1 =</span> <span class="dv">0</span>, <span class="dt">ident.2 =</span> <span class="dv">1</span>, <span class="dt">test.use =</span> ??)</a></code></pre></div>
</div>
</div>
<div id="check-clusters" class="section level3">
<h3><span class="header-section-number">10.3.2</span> Check Clusters</h3>
<p>How do we test the cell types identified? How do we know how reliable they are?</p>
<p>Use Classifier to predict cell cluster. See how it predicts using hold out data.
<a href="https://satijalab.org/seurat/v3.0/integration.html">reference</a></p>
<div class="sourceCode" id="cb271"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb271-1" data-line-number="1"><span class="co"># Assign the test object a three level attribute</span></a>
<a class="sourceLine" id="cb271-2" data-line-number="2">groups &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="kw">c</span>(<span class="st">&quot;train&quot;</span>, <span class="st">&quot;test&quot;</span>), <span class="dt">size =</span> <span class="kw">NROW</span>(seurat<span class="op">@</span>meta.data), <span class="dt">replace =</span> <span class="ot">TRUE</span>, <span class="dt">prob =</span> <span class="kw">c</span>(<span class="fl">0.8</span>, <span class="fl">0.2</span>))</a>
<a class="sourceLine" id="cb271-3" data-line-number="3"><span class="kw">names</span>(groups) &lt;-<span class="st"> </span><span class="kw">colnames</span>(seurat)</a>
<a class="sourceLine" id="cb271-4" data-line-number="4">seurat &lt;-<span class="st"> </span><span class="kw">AddMetaData</span>(<span class="dt">object =</span> seurat, <span class="dt">metadata =</span> groups, <span class="dt">col.name =</span> <span class="st">&quot;group&quot;</span>)</a>
<a class="sourceLine" id="cb271-5" data-line-number="5"></a>
<a class="sourceLine" id="cb271-6" data-line-number="6"><span class="co"># Find Anchors</span></a>
<a class="sourceLine" id="cb271-7" data-line-number="7">seurat.list &lt;-<span class="st"> </span><span class="kw">SplitObject</span>(seurat, <span class="dt">split.by =</span> <span class="st">&quot;group&quot;</span>)</a>
<a class="sourceLine" id="cb271-8" data-line-number="8">seurat.anchors &lt;-<span class="st"> </span><span class="kw">FindIntegrationAnchors</span>(<span class="dt">object.list =</span> seurat.list, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">30</span>)</a></code></pre></div>
<pre><code>## Computing 2000 integration features</code></pre>
<pre><code>## Scaling features for provided objects</code></pre>
<pre><code>## Finding all pairwise anchors</code></pre>
<pre><code>## Running CCA</code></pre>
<pre><code>## Merging objects</code></pre>
<pre><code>## Finding neighborhoods</code></pre>
<pre><code>## Finding anchors</code></pre>
<pre><code>##  Found 5057 anchors</code></pre>
<pre><code>## Filtering anchors</code></pre>
<pre><code>##  Retained 3633 anchors</code></pre>
<pre><code>## Extracting within-dataset neighbors</code></pre>
<div class="sourceCode" id="cb283"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb283-1" data-line-number="1">seurat.integrated &lt;-<span class="st"> </span><span class="kw">IntegrateData</span>(<span class="dt">anchorset =</span> seurat.anchors, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">30</span>)</a></code></pre></div>
<pre><code>## Merging dataset 1 into 2</code></pre>
<pre><code>## Extracting anchors for merged samples</code></pre>
<pre><code>## Finding integration vectors</code></pre>
<pre><code>## Finding integration vector weights</code></pre>
<pre><code>## Integrating data</code></pre>
<pre><code>## Warning: Adding a command log without an assay associated with it</code></pre>
<div class="sourceCode" id="cb290"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb290-1" data-line-number="1">seurat.query &lt;-<span class="st"> </span>seurat.list[[<span class="st">&quot;train&quot;</span>]]</a>
<a class="sourceLine" id="cb290-2" data-line-number="2">seurat.anchors &lt;-<span class="st"> </span><span class="kw">FindTransferAnchors</span>(<span class="dt">reference =</span> seurat.integrated, </a>
<a class="sourceLine" id="cb290-3" data-line-number="3">                                      <span class="dt">query =</span> seurat.query, </a>
<a class="sourceLine" id="cb290-4" data-line-number="4">                                      <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">30</span>)</a></code></pre></div>
<pre><code>## Performing PCA on the provided reference using 2000 features as input.</code></pre>
<pre><code>## Projecting PCA</code></pre>
<pre><code>## Finding neighborhoods</code></pre>
<pre><code>## Finding anchors</code></pre>
<pre><code>##  Found 13441 anchors</code></pre>
<pre><code>## Filtering anchors</code></pre>
<pre><code>##  Retained 10905 anchors</code></pre>
<pre><code>## Extracting within-dataset neighbors</code></pre>
<div class="sourceCode" id="cb299"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb299-1" data-line-number="1">predictions &lt;-<span class="st"> </span><span class="kw">TransferData</span>(<span class="dt">anchorset =</span> seurat.anchors, </a>
<a class="sourceLine" id="cb299-2" data-line-number="2">                            <span class="dt">refdata =</span> seurat.integrated<span class="op">$</span>ClusterNames_<span class="fl">0.6</span>, </a>
<a class="sourceLine" id="cb299-3" data-line-number="3">                            <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">30</span>)</a></code></pre></div>
<pre><code>## Finding integration vectors</code></pre>
<pre><code>## Finding integration vector weights</code></pre>
<pre><code>## Predicting cell labels</code></pre>
<div class="sourceCode" id="cb303"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb303-1" data-line-number="1">seurat.query &lt;-<span class="st"> </span><span class="kw">AddMetaData</span>(seurat.query, <span class="dt">metadata =</span> predictions)</a>
<a class="sourceLine" id="cb303-2" data-line-number="2"><span class="kw">table</span>(seurat.query<span class="op">@</span>meta.data<span class="op">$</span>ClusterNames_<span class="fl">0.6</span>, seurat.query<span class="op">@</span>meta.data<span class="op">$</span>predicted.id)</a></code></pre></div>
<pre><code>##             
##              CellType0 CellType1 CellType10 CellType11 CellType12 CellType13 CellType14 CellType2 CellType3
##   CellType0       1249         0          0          0          0          0          0        21         0
##   CellType1          0       695          0          0          3          0          0         0        24
##   CellType2        439         0          0          0          0          2          0       126         0
##   CellType3          0        40          0          0          0          1          0         0       479
##   CellType4          0         0          4          0          0          0          0         0         0
##   CellType5          0         0          0          0          0          0          0         0         0
##   CellType6          0         0          0          0          0          0          0         0         0
##   CellType7          0         0          0          0          0          0          0         0        18
##   CellType8          0         0          0          0          0          0          0         3         1
##   CellType9          0         0          0          0          0          0          0         0         0
##   CellType10         1         1        116          0          0          0          0         0         0
##   CellType11         0         0          0        112          0          0          0         0         0
##   CellType12         0        50          0          0         15          2          0         0        12
##   CellType13         4         2          0          0          0         53          0         0         1
##   CellType14         0         0          0          0          0          0         30         0         0
##             
##              CellType4 CellType5 CellType6 CellType7 CellType8 CellType9
##   CellType0          0         0         0         0         0         0
##   CellType1          0         0         0         0         0         0
##   CellType2          0         0         0         0         0         0
##   CellType3          0         0         0        14         0         0
##   CellType4        400         0         0         0         0         0
##   CellType5          0       253         7         0         1         0
##   CellType6          0         3       232         0         4         0
##   CellType7          0         0         0       171         0         0
##   CellType8          0         2         4         0       164         1
##   CellType9          0         0         0         0         0       133
##   CellType10         5         0         0         0         0         0
##   CellType11         0         0         0         0         0         0
##   CellType12         0         0         0         1         0         0
##   CellType13         0         0         0         0         0         0
##   CellType14         0         0         0         0         0         0</code></pre>
</div>
<div id="view-entire-object-structure" class="section level3">
<h3><span class="header-section-number">10.3.3</span> View Entire Object Structure</h3>
<p>Notice all the slots and elements added to the object.</p>
<div class="sourceCode" id="cb305"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb305-1" data-line-number="1"><span class="kw">str</span>(seurat)</a></code></pre></div>
</div>
</div>
<div id="probabilistic-lda-clustering" class="section level2">
<h2><span class="header-section-number">10.4</span> Probabilistic (LDA) Clustering</h2>
<p>Another type of clustering we can do is a fuzzy or probablistic clustering. This is where cells are not assigned to specifically only one cluster. They get assigned a score for how much the cells belong to each of the clusters (sometimes called topics). This can be helpful for when your dataset continuous processes and/or cellular states as opposed to distinct cell types.</p>
<div id="example-lda-in-bulk" class="section level3">
<h3><span class="header-section-number">10.4.1</span> Example LDA in Bulk</h3>
<div class="sourceCode" id="cb306"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb306-1" data-line-number="1"><span class="kw">data</span>(<span class="st">&quot;MouseDeng2014.FitGoM&quot;</span>)</a>
<a class="sourceLine" id="cb306-2" data-line-number="2"><span class="kw">names</span>(MouseDeng2014.FitGoM)</a></code></pre></div>
<pre><code>## [1] &quot;clust_3&quot; &quot;clust_6&quot;</code></pre>
<div class="sourceCode" id="cb308"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb308-1" data-line-number="1">omega &lt;-<span class="st"> </span>MouseDeng2014.FitGoM<span class="op">$</span>clust_<span class="dv">6</span><span class="op">$</span>omega</a>
<a class="sourceLine" id="cb308-2" data-line-number="2"></a>
<a class="sourceLine" id="cb308-3" data-line-number="3"></a>
<a class="sourceLine" id="cb308-4" data-line-number="4">annotation &lt;-<span class="st"> </span><span class="kw">data.frame</span>(</a>
<a class="sourceLine" id="cb308-5" data-line-number="5">  <span class="dt">sample_id =</span> <span class="kw">paste0</span>(<span class="st">&quot;X&quot;</span>, <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">NROW</span>(omega))),</a>
<a class="sourceLine" id="cb308-6" data-line-number="6">  <span class="dt">tissue_label =</span> <span class="kw">factor</span>(<span class="kw">rownames</span>(omega),</a>
<a class="sourceLine" id="cb308-7" data-line-number="7">                        <span class="dt">levels =</span> <span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;zy&quot;</span>, <span class="st">&quot;early2cell&quot;</span>, <span class="st">&quot;mid2cell&quot;</span>, <span class="st">&quot;late2cell&quot;</span>,</a>
<a class="sourceLine" id="cb308-8" data-line-number="8">                                        <span class="st">&quot;4cell&quot;</span>, <span class="st">&quot;8cell&quot;</span>, <span class="st">&quot;16cell&quot;</span>, <span class="st">&quot;earlyblast&quot;</span>, <span class="st">&quot;midblast&quot;</span>, <span class="st">&quot;lateblast&quot;</span>)))</a>
<a class="sourceLine" id="cb308-9" data-line-number="9">  )</a>
<a class="sourceLine" id="cb308-10" data-line-number="10"></a>
<a class="sourceLine" id="cb308-11" data-line-number="11"><span class="kw">rownames</span>(omega) &lt;-<span class="st"> </span>annotation<span class="op">$</span>sample_id;</a>
<a class="sourceLine" id="cb308-12" data-line-number="12"></a>
<a class="sourceLine" id="cb308-13" data-line-number="13"><span class="kw">StructureGGplot</span>(<span class="dt">omega =</span> omega,</a>
<a class="sourceLine" id="cb308-14" data-line-number="14">                <span class="dt">annotation =</span> annotation,</a>
<a class="sourceLine" id="cb308-15" data-line-number="15">                <span class="dt">palette =</span> RColorBrewer<span class="op">::</span><span class="kw">brewer.pal</span>(<span class="dv">8</span>, <span class="st">&quot;Accent&quot;</span>),</a>
<a class="sourceLine" id="cb308-16" data-line-number="16">                <span class="dt">yaxis_label =</span> <span class="st">&quot;Amplification batch&quot;</span>,</a>
<a class="sourceLine" id="cb308-17" data-line-number="17">                <span class="dt">order_sample =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb308-18" data-line-number="18">                <span class="dt">axis_tick =</span> <span class="kw">list</span>(<span class="dt">axis_ticks_length =</span> <span class="fl">.1</span>,</a>
<a class="sourceLine" id="cb308-19" data-line-number="19">                                 <span class="dt">axis_ticks_lwd_y =</span> <span class="fl">.1</span>,</a>
<a class="sourceLine" id="cb308-20" data-line-number="20">                                 <span class="dt">axis_ticks_lwd_x =</span> <span class="fl">.1</span>,</a>
<a class="sourceLine" id="cb308-21" data-line-number="21">                                 <span class="dt">axis_label_size =</span> <span class="dv">7</span>,</a>
<a class="sourceLine" id="cb308-22" data-line-number="22">                                 <span class="dt">axis_label_face =</span> <span class="st">&quot;bold&quot;</span>))</a></code></pre></div>
<pre><code>## Warning: Use of `df_mlt$document` is discouraged. Use `document` instead.</code></pre>
<pre><code>## Warning: Use of `df_mlt$value` is discouraged. Use `value` instead.</code></pre>
<pre><code>## Warning: Use of `df_mlt$topic` is discouraged. Use `topic` instead.</code></pre>
<p><img src="2020_scWorkshop_files/figure-html/example_lda-1.png" width="90%" style="display: block; margin: auto;" /></p>
</div>
<div id="pbmc-lda" class="section level3">
<h3><span class="header-section-number">10.4.2</span> PBMC LDA</h3>
<div id="fitting-the-model" class="section level4">
<h4><span class="header-section-number">10.4.2.1</span> Fitting the Model</h4>
<div class="sourceCode" id="cb312"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb312-1" data-line-number="1"><span class="kw">set.seed</span>(<span class="dv">2020</span>)</a>
<a class="sourceLine" id="cb312-2" data-line-number="2"></a>
<a class="sourceLine" id="cb312-3" data-line-number="3"><span class="co">## Preprocessing Steps</span></a>
<a class="sourceLine" id="cb312-4" data-line-number="4">pbmc_small &lt;-<span class="st"> </span><span class="kw">NormalizeData</span>(<span class="dt">object =</span> pbmc_small, <span class="dt">normalization.method =</span> <span class="st">&quot;LogNormalize&quot;</span>, <span class="dt">scale.factor =</span> <span class="dv">10000</span>)</a>
<a class="sourceLine" id="cb312-5" data-line-number="5">pbmc_small &lt;-<span class="st"> </span><span class="kw">RunPCA</span>(<span class="dt">object =</span> pbmc_small)</a></code></pre></div>
<pre><code>## Warning in irlba(A = t(x = object), nv = npcs, ...): You&#39;re computing too large a percentage of total singular
## values, use a standard svd instead.</code></pre>
<pre><code>## PC_ 1 
## Positive:  SDPR, PF4, PPBP, TUBB1, CA2, TREML1, MYL9, PGRMC1, RUFY1, PARVB 
##     VDAC3, IGLL5, AKR1C3, CD1C, RP11-290F20.3, GNLY, S100A8, S100A9, HLA-DQA1, HLA-DPB1 
## Negative:  HLA-DPB1, HLA-DQA1, S100A9, S100A8, GNLY, RP11-290F20.3, CD1C, AKR1C3, IGLL5, VDAC3 
##     PARVB, RUFY1, PGRMC1, MYL9, TREML1, CA2, TUBB1, PPBP, PF4, SDPR 
## PC_ 2 
## Positive:  HLA-DPB1, HLA-DQA1, S100A8, S100A9, CD1C, RP11-290F20.3, PARVB, IGLL5, MYL9, SDPR 
##     PPBP, CA2, RUFY1, TREML1, PF4, TUBB1, PGRMC1, VDAC3, AKR1C3, GNLY 
## Negative:  GNLY, AKR1C3, VDAC3, PGRMC1, TUBB1, PF4, TREML1, RUFY1, CA2, PPBP 
##     SDPR, MYL9, IGLL5, PARVB, RP11-290F20.3, CD1C, S100A9, S100A8, HLA-DQA1, HLA-DPB1 
## PC_ 3 
## Positive:  S100A9, S100A8, RP11-290F20.3, AKR1C3, PARVB, GNLY, PPBP, PGRMC1, MYL9, TUBB1 
##     CA2, TREML1, SDPR, VDAC3, PF4, RUFY1, HLA-DPB1, IGLL5, CD1C, HLA-DQA1 
## Negative:  HLA-DQA1, CD1C, IGLL5, HLA-DPB1, RUFY1, PF4, VDAC3, SDPR, TREML1, CA2 
##     TUBB1, MYL9, PGRMC1, PPBP, GNLY, PARVB, AKR1C3, RP11-290F20.3, S100A8, S100A9 
## PC_ 4 
## Positive:  IGLL5, RP11-290F20.3, VDAC3, PPBP, TUBB1, TREML1, PF4, CA2, PARVB, MYL9 
##     SDPR, RUFY1, PGRMC1, S100A9, HLA-DQA1, HLA-DPB1, GNLY, S100A8, AKR1C3, CD1C 
## Negative:  CD1C, AKR1C3, S100A8, GNLY, HLA-DPB1, HLA-DQA1, S100A9, PGRMC1, RUFY1, SDPR 
##     MYL9, PARVB, CA2, PF4, TREML1, TUBB1, PPBP, VDAC3, RP11-290F20.3, IGLL5 
## PC_ 5 
## Positive:  MYL9, PARVB, IGLL5, TREML1, AKR1C3, PGRMC1, HLA-DPB1, S100A9, TUBB1, PF4 
##     SDPR, GNLY, PPBP, S100A8, CA2, HLA-DQA1, CD1C, RUFY1, RP11-290F20.3, VDAC3 
## Negative:  VDAC3, RP11-290F20.3, RUFY1, CD1C, HLA-DQA1, CA2, S100A8, PPBP, GNLY, SDPR 
##     PF4, TUBB1, S100A9, HLA-DPB1, PGRMC1, AKR1C3, TREML1, IGLL5, PARVB, MYL9</code></pre>
<div class="sourceCode" id="cb315"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb315-1" data-line-number="1">pbmc_small &lt;-<span class="st"> </span><span class="kw">FindClusters</span>(<span class="dt">object =</span> pbmc_small, </a>
<a class="sourceLine" id="cb315-2" data-line-number="2">                           <span class="dt">reduction =</span> <span class="st">&quot;pca&quot;</span>, </a>
<a class="sourceLine" id="cb315-3" data-line-number="3">                           <span class="dt">dims.use =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, </a>
<a class="sourceLine" id="cb315-4" data-line-number="4">                           <span class="dt">resolution =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb315-5" data-line-number="5">                           <span class="dt">print.output =</span> <span class="dv">0</span>)</a></code></pre></div>
<pre><code>## Warning: The following arguments are not used: reduction, dims.use, print.output</code></pre>
<pre><code>## Suggested parameter: dims instead of dims.use; verbose instead of print.output</code></pre>
<pre><code>## Warning: The following arguments are not used: reduction, dims.use, print.output</code></pre>
<pre><code>## Suggested parameter: dims instead of dims.use; verbose instead of print.output</code></pre>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 80
## Number of edges: 3124
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.1433
## Number of communities: 3
## Elapsed time: 0 seconds</code></pre>
<pre><code>## Warning: Adding a command log without an assay associated with it</code></pre>
<div class="sourceCode" id="cb322"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb322-1" data-line-number="1"><span class="co">## Grab the Raw Count Matrix from Seurat Object</span></a>
<a class="sourceLine" id="cb322-2" data-line-number="2"><span class="co">## Needs count matrix</span></a>
<a class="sourceLine" id="cb322-3" data-line-number="3">pbmc_counts &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(pbmc_small<span class="op">@</span>assays<span class="op">$</span>RNA<span class="op">@</span>counts)</a>
<a class="sourceLine" id="cb322-4" data-line-number="4">pbmc_meta &lt;-<span class="st"> </span>pbmc_small<span class="op">@</span>meta.data</a>
<a class="sourceLine" id="cb322-5" data-line-number="5">gene_names &lt;-<span class="st"> </span><span class="kw">rownames</span>(pbmc_counts)</a>
<a class="sourceLine" id="cb322-6" data-line-number="6"></a>
<a class="sourceLine" id="cb322-7" data-line-number="7"></a>
<a class="sourceLine" id="cb322-8" data-line-number="8"><span class="co">## Fit LDA  Model, called GoM for this R pacakge</span></a>
<a class="sourceLine" id="cb322-9" data-line-number="9"><span class="co">## Number of topics to fit is K=4</span></a>
<a class="sourceLine" id="cb322-10" data-line-number="10">pbmc_FitGoM &lt;-<span class="st"> </span><span class="kw">FitGoM</span>(<span class="kw">t</span>(pbmc_counts), <span class="dt">K=</span><span class="dv">4</span>)</a></code></pre></div>
<pre><code>## options not specified: switching to default BIC, other choice is BF for Bayes factor</code></pre>
<pre><code>## Fitting a Grade of Membership model
##             (Taddy M., AISTATS 2012, JMLR 22,
##             http://proceedings.mlr.press/v22/taddy12/taddy12.pdf)</code></pre>
<pre><code>## 
## Estimating on a 80 samples collection.
## Fit and Bayes Factor Estimation for K = 4
## log posterior increase: 6685.6, 59.7, done.
## log BF( 4 ) = 10162.23</code></pre>
<div class="sourceCode" id="cb326"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb326-1" data-line-number="1"><span class="co">## Grab topic scores</span></a>
<a class="sourceLine" id="cb326-2" data-line-number="2">omega &lt;-<span class="st"> </span><span class="kw">data.frame</span>(pbmc_FitGoM<span class="op">$</span>fit<span class="op">$</span>omega)</a>
<a class="sourceLine" id="cb326-3" data-line-number="3"></a>
<a class="sourceLine" id="cb326-4" data-line-number="4"><span class="co">## Annotate topics, if identies known</span></a>
<a class="sourceLine" id="cb326-5" data-line-number="5">annotation &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">sample_id =</span> <span class="kw">rownames</span>(omega),</a>
<a class="sourceLine" id="cb326-6" data-line-number="6">                         <span class="dt">tissue_label =</span> <span class="kw">paste0</span>(<span class="st">&quot;cluster&quot;</span>, pbmc_small<span class="op">@</span>active.ident))</a>
<a class="sourceLine" id="cb326-7" data-line-number="7"></a>
<a class="sourceLine" id="cb326-8" data-line-number="8"><span class="kw">colnames</span>(omega) &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;topic&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">4</span>)</a>
<a class="sourceLine" id="cb326-9" data-line-number="9"><span class="kw">rownames</span>(omega) &lt;-<span class="st"> </span>annotation<span class="op">$</span>sample_id;</a>
<a class="sourceLine" id="cb326-10" data-line-number="10"></a>
<a class="sourceLine" id="cb326-11" data-line-number="11"><span class="co">## Make plot to see topic contribution</span></a>
<a class="sourceLine" id="cb326-12" data-line-number="12"><span class="kw">StructureGGplot</span>(<span class="dt">omega =</span> omega,</a>
<a class="sourceLine" id="cb326-13" data-line-number="13">                <span class="dt">annotation =</span> annotation,</a>
<a class="sourceLine" id="cb326-14" data-line-number="14">                <span class="dt">palette =</span> RColorBrewer<span class="op">::</span><span class="kw">brewer.pal</span>(<span class="dv">4</span>, <span class="st">&quot;Dark2&quot;</span>),</a>
<a class="sourceLine" id="cb326-15" data-line-number="15">                <span class="dt">yaxis_label =</span> <span class="st">&quot;Cells&quot;</span>,</a>
<a class="sourceLine" id="cb326-16" data-line-number="16">                <span class="dt">order_sample =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb326-17" data-line-number="17">                <span class="dt">axis_tick =</span> <span class="kw">list</span>(<span class="dt">axis_ticks_length =</span> <span class="fl">.1</span>,</a>
<a class="sourceLine" id="cb326-18" data-line-number="18">                                 <span class="dt">axis_ticks_lwd_y =</span> <span class="fl">.1</span>,</a>
<a class="sourceLine" id="cb326-19" data-line-number="19">                                 <span class="dt">axis_ticks_lwd_x =</span> <span class="fl">.1</span>,</a>
<a class="sourceLine" id="cb326-20" data-line-number="20">                                 <span class="dt">axis_label_size =</span> <span class="dv">7</span>,</a>
<a class="sourceLine" id="cb326-21" data-line-number="21">                                 <span class="dt">axis_label_face =</span> <span class="st">&quot;bold&quot;</span>))</a></code></pre></div>
<pre><code>## Warning: Use of `df_mlt$document` is discouraged. Use `document` instead.</code></pre>
<pre><code>## Warning: Use of `df_mlt$value` is discouraged. Use `value` instead.</code></pre>
<pre><code>## Warning: Use of `df_mlt$topic` is discouraged. Use `topic` instead.</code></pre>
<p><img src="2020_scWorkshop_files/figure-html/pbmc_lda-1.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb330"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb330-1" data-line-number="1"><span class="co"># ## Add Topic Scores to Meta Data Part of the Seurat Object</span></a>
<a class="sourceLine" id="cb330-2" data-line-number="2">pbmc_small &lt;-<span class="st"> </span><span class="kw">AddMetaData</span>(pbmc_small, omega)</a></code></pre></div>
</div>
<div id="summarizing-topic-by-cluster" class="section level4">
<h4><span class="header-section-number">10.4.2.2</span> Summarizing Topic By Cluster</h4>
<div class="sourceCode" id="cb331"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb331-1" data-line-number="1">pbmc_small<span class="op">@</span>meta.data <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb331-2" data-line-number="2"><span class="st">  </span><span class="kw">group_by</span>(RNA_snn_res<span class="fl">.1</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb331-3" data-line-number="3"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">topic1 =</span> <span class="kw">mean</span>(topic1),</a>
<a class="sourceLine" id="cb331-4" data-line-number="4">            <span class="dt">topic2 =</span> <span class="kw">mean</span>(topic2),</a>
<a class="sourceLine" id="cb331-5" data-line-number="5">            <span class="dt">topic3 =</span> <span class="kw">mean</span>(topic3),</a>
<a class="sourceLine" id="cb331-6" data-line-number="6">            <span class="dt">topic4 =</span> <span class="kw">mean</span>(topic4))</a></code></pre></div>
<pre><code>## # A tibble: 3 x 5
##   RNA_snn_res.1 topic1 topic2  topic3  topic4
##   &lt;fct&gt;          &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;
## 1 0             0.0771 0.326  0.342   0.255  
## 2 1             0.973  0.0128 0.00660 0.00778
## 3 2             0.543  0.352  0.103   0.00247</code></pre>
</div>
<div id="visualizing-topic-scores" class="section level4">
<h4><span class="header-section-number">10.4.2.3</span> Visualizing Topic Scores</h4>
<div class="sourceCode" id="cb333"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb333-1" data-line-number="1"><span class="co">## ggplot object, you can add layers</span></a>
<a class="sourceLine" id="cb333-2" data-line-number="2">p1 &lt;-<span class="st"> </span><span class="kw">DimPlot</span>(pbmc_small, <span class="dt">reduction =</span> <span class="st">&quot;tsne&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">labs</span>(<span class="dt">title =</span> <span class="st">&quot;Resolution 1&quot;</span>) <span class="co">## return ggplot object</span></a>
<a class="sourceLine" id="cb333-3" data-line-number="3">p1</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/tsne_plot_pbmc-1.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb334"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb334-1" data-line-number="1">p2 &lt;-<span class="st"> </span><span class="kw">FeaturePlot</span>(<span class="dt">object =</span> pbmc_small, </a>
<a class="sourceLine" id="cb334-2" data-line-number="2">                  <span class="dt">features =</span> <span class="kw">c</span>(<span class="st">&quot;topic1&quot;</span>, <span class="st">&quot;topic2&quot;</span>, <span class="st">&quot;topic3&quot;</span>, <span class="st">&quot;topic4&quot;</span>), </a>
<a class="sourceLine" id="cb334-3" data-line-number="3">                  <span class="dt">cols =</span> <span class="kw">c</span>(<span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>), </a>
<a class="sourceLine" id="cb334-4" data-line-number="4">                  <span class="dt">reduction =</span> <span class="st">&quot;tsne&quot;</span>) <span class="co">## return ggplot object</span></a>
<a class="sourceLine" id="cb334-5" data-line-number="5"></a>
<a class="sourceLine" id="cb334-6" data-line-number="6">p2</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/topic_plot-1.png" width="90%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb335"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb335-1" data-line-number="1">p1 <span class="op">+</span><span class="st"> </span>p2</a></code></pre></div>
<p><img src="2020_scWorkshop_files/figure-html/combine_topic_plot-1.png" width="90%" style="display: block; margin: auto;" /></p>
</div>
<div id="extract-top-feature" class="section level4">
<h4><span class="header-section-number">10.4.2.4</span> Extract Top Feature</h4>
<div class="sourceCode" id="cb336"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb336-1" data-line-number="1"><span class="co">## Gene scores for each topics</span></a>
<a class="sourceLine" id="cb336-2" data-line-number="2">theta_mat &lt;-<span class="st"> </span>pbmc_FitGoM<span class="op">$</span>fit<span class="op">$</span>theta</a>
<a class="sourceLine" id="cb336-3" data-line-number="3"></a>
<a class="sourceLine" id="cb336-4" data-line-number="4">top_features &lt;-<span class="st"> </span><span class="kw">ExtractTopFeatures</span>(theta_mat, </a>
<a class="sourceLine" id="cb336-5" data-line-number="5">                                   <span class="dt">top_features=</span><span class="dv">100</span>,</a>
<a class="sourceLine" id="cb336-6" data-line-number="6">                                   <span class="dt">method=</span><span class="st">&quot;poisson&quot;</span>, </a>
<a class="sourceLine" id="cb336-7" data-line-number="7">                                   <span class="dt">options=</span><span class="st">&quot;min&quot;</span>)</a>
<a class="sourceLine" id="cb336-8" data-line-number="8"></a>
<a class="sourceLine" id="cb336-9" data-line-number="9">gene_list &lt;-<span class="st"> </span><span class="kw">do.call</span>(rbind, </a>
<a class="sourceLine" id="cb336-10" data-line-number="10">                     <span class="kw">lapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">dim</span>(top_features<span class="op">$</span>indices)[<span class="dv">1</span>],</a>
<a class="sourceLine" id="cb336-11" data-line-number="11">                            <span class="cf">function</span>(x) gene_names[top_features<span class="op">$</span>indices[x,]]))</a></code></pre></div>
<p>We tabulate the top <code>5</code> genes for these <code>4</code> topics</p>
<div class="sourceCode" id="cb337"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb337-1" data-line-number="1">out_table &lt;-<span class="st"> </span><span class="kw">do.call</span>(rbind, <span class="kw">lapply</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>, <span class="cf">function</span>(i) <span class="kw">toString</span>(gene_list[i,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>])))</a>
<a class="sourceLine" id="cb337-2" data-line-number="2"><span class="kw">rownames</span>(out_table) &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;Topic&quot;</span>, <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>))</a>
<a class="sourceLine" id="cb337-3" data-line-number="3">out_table</a></code></pre></div>
<pre><code>##         [,1]                                      
## Topic 1 &quot;S100A9, LYZ, HLA-DPB1, HLA-DRA, HLA-DPA1&quot;
## Topic 2 &quot;LTB, LYAR, IGLL5, CD79A, SRSF7&quot;          
## Topic 3 &quot;GNLY, NKG7, GZMB, GZMA, FGFBP2&quot;          
## Topic 4 &quot;PPBP, PF4, GNG11, CLU, TUBB1&quot;</code></pre>
</div>
</div>
<div id="practice-visualizingembedding" class="section level3">
<h3><span class="header-section-number">10.4.3</span> Practice Visualizing/Embedding</h3>
<div id="tsne" class="section level4">
<h4><span class="header-section-number">10.4.3.1</span> tSNE</h4>
<p>Change the parameter settings for tSNE</p>
<div class="sourceCode" id="cb339"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb339-1" data-line-number="1"><span class="kw">RunTSNE</span>()</a></code></pre></div>
</div>
<div id="umap" class="section level4">
<h4><span class="header-section-number">10.4.3.2</span> UMAP</h4>
<p>Change the parameter settings for UMAP</p>
<div class="sourceCode" id="cb340"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb340-1" data-line-number="1"><span class="kw">RunUMAP</span>()</a></code></pre></div>
</div>
</div>
</div>
<div id="other-options-for-analysis" class="section level2">
<h2><span class="header-section-number">10.5</span> Other Options For Analysis</h2>
<ul>
<li><a href="https://satijalab.org/seurat/vignettes.html">More Seurat Vignettes</a></li>
<li><a href="https://broadinstitute.github.io/2019_scWorkshop/">Single Cell Analysis Workshop</a></li>
<li><a href="https://scrnaseq-course.cog.sanger.ac.uk/website/index.html">Hemberg Lab Course</a></li>
<li><a href="https://bioconductor.org/packages/release/bioc/vignettes/SingleCellExperiment/inst/doc/intro.html">SingleCellExperiment</a></li>
<li><a href="https://scanpy-tutorials.readthedocs.io/en/latest/pbmc3k.html">Scanpy</a></li>
<li><a href="https://github.com/klarman-cell-observatory/pegasus/blob/master/notebooks/Pegasus%20Tutorial%20in%20Workshop.ipynb">Pegasus</a></li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="identifying-cell-populations.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="batch-correction-lecture.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["2020_scWorkshop.pdf", "2020_scWorkshop.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
