<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Processing scRNAseq Data | ANALYSIS OF SINGLE CELL RNA-SEQ DATA</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Processing scRNAseq Data | ANALYSIS OF SINGLE CELL RNA-SEQ DATA" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://broadinstitute.github.io/2020_scWorkshop/" />
  <meta property="og:image" content="https://broadinstitute.github.io/2020_scWorkshop/images/cover.png" />
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="broadinstitute/2019_scWorkship" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Processing scRNAseq Data | ANALYSIS OF SINGLE CELL RNA-SEQ DATA" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="twitter:image" content="https://broadinstitute.github.io/2020_scWorkshop/images/cover.png" />

<meta name="author" content="Orr Ashenberg" />
<meta name="author" content="Dana Silverbush" />
<meta name="author" content="Kirk Gosik" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="transcriptome-quantification.html"/>
<link rel="next" href="introduction-rbioconductor.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">ANALYSIS OF SINGLE CELL RNA-SEQ DATA</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#course-overview"><i class="fa fa-check"></i><b>1.1</b> COURSE OVERVIEW</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#targeted-audience-assumed-background"><i class="fa fa-check"></i><b>1.2</b> TARGETED AUDIENCE &amp; ASSUMED BACKGROUND</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#course-format"><i class="fa fa-check"></i><b>1.3</b> COURSE FORMAT</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#getting-started"><i class="fa fa-check"></i><b>1.4</b> Getting Started</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#session-content"><i class="fa fa-check"></i><b>1.5</b> SESSION CONTENT</a><ul>
<li class="chapter" data-level="1.5.1" data-path="index.html"><a href="index.html#monday-classes-from-0800-to-1600-lunch-break-1-hr-40-min-of-total-coffee-breaks"><i class="fa fa-check"></i><b>1.5.1</b> Monday – Classes from 08:00 to 16:00 (lunch break-1 hr, 40 min of total coffee breaks)</a></li>
<li class="chapter" data-level="1.5.2" data-path="index.html"><a href="index.html#tuesday-classes-from-0800-to-1600"><i class="fa fa-check"></i><b>1.5.2</b> Tuesday – Classes from 08:00 to 16:00</a></li>
<li class="chapter" data-level="1.5.3" data-path="index.html"><a href="index.html#wednesday-classes-from-0800-to-1600"><i class="fa fa-check"></i><b>1.5.3</b> Wednesday – Classes from 08:00 to 16:00</a></li>
<li class="chapter" data-level="1.5.4" data-path="index.html"><a href="index.html#thursday-classes-from-0800-to-1600"><i class="fa fa-check"></i><b>1.5.4</b> Thursday – Classes from 08:00 to 16:00</a></li>
<li class="chapter" data-level="1.5.5" data-path="index.html"><a href="index.html#friday-classes-from-0800-to-1600"><i class="fa fa-check"></i><b>1.5.5</b> Friday – Classes from 08:00 to 16:00</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction-1.html"><a href="introduction-1.html"><i class="fa fa-check"></i><b>2</b> Introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="introduction-1.html"><a href="introduction-1.html#slides"><i class="fa fa-check"></i><b>2.1</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html"><i class="fa fa-check"></i><b>3</b> Understanding Sequencing Raw Data</a><ul>
<li class="chapter" data-level="3.1" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#class-environment"><i class="fa fa-check"></i><b>3.1</b> Class Environment</a><ul>
<li class="chapter" data-level="3.1.1" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#getting-into-aws-instance"><i class="fa fa-check"></i><b>3.1.1</b> Getting into AWS Instance</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#shell-and-unix-commands"><i class="fa fa-check"></i><b>3.2</b> Shell and Unix commands</a><ul>
<li class="chapter" data-level="3.2.1" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#common-linux-commands"><i class="fa fa-check"></i><b>3.2.1</b> Common Linux Commands</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#file-formats"><i class="fa fa-check"></i><b>3.3</b> File formats</a><ul>
<li class="chapter" data-level="3.3.1" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#view-fastq-files"><i class="fa fa-check"></i><b>3.3.1</b> View FASTQ Files</a></li>
<li class="chapter" data-level="3.3.2" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#view-bam-files"><i class="fa fa-check"></i><b>3.3.2</b> View BAM Files</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#public-data-repositories"><i class="fa fa-check"></i><b>3.4</b> Public data repositories</a><ul>
<li class="chapter" data-level="3.4.1" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#cellranger10x"><i class="fa fa-check"></i><b>3.4.1</b> Cellranger/10x</a></li>
<li class="chapter" data-level="3.4.2" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#geo"><i class="fa fa-check"></i><b>3.4.2</b> GEO</a></li>
<li class="chapter" data-level="3.4.3" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#single-cell-portal"><i class="fa fa-check"></i><b>3.4.3</b> Single Cell Portal</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="understanding-sequencing-raw-data.html"><a href="understanding-sequencing-raw-data.html#docker-commands"><i class="fa fa-check"></i><b>3.5</b> Docker Commands</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="transcriptome-quantification.html"><a href="transcriptome-quantification.html"><i class="fa fa-check"></i><b>4</b> Transcriptome Quantification</a><ul>
<li class="chapter" data-level="4.1" data-path="transcriptome-quantification.html"><a href="transcriptome-quantification.html#google-slides"><i class="fa fa-check"></i><b>4.1</b> Google Slides</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html"><i class="fa fa-check"></i><b>5</b> Processing scRNAseq Data</a><ul>
<li class="chapter" data-level="5.1" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#goal"><i class="fa fa-check"></i><b>5.1</b> Goal</a></li>
<li class="chapter" data-level="5.2" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#further-reading"><i class="fa fa-check"></i><b>5.2</b> Further reading</a></li>
<li class="chapter" data-level="5.3" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#download-data"><i class="fa fa-check"></i><b>5.3</b> Download data</a><ul>
<li class="chapter" data-level="5.3.1" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#fastq-file-format"><i class="fa fa-check"></i><b>5.3.1</b> Fastq file format</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#align-the-reads"><i class="fa fa-check"></i><b>5.4</b> Align the reads</a><ul>
<li class="chapter" data-level="5.4.1" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#star-align"><i class="fa fa-check"></i><b>5.4.1</b> STAR align</a></li>
<li class="chapter" data-level="5.4.2" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#bam-file-format"><i class="fa fa-check"></i><b>5.4.2</b> Bam file format</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="processing-scrnaseq-data.html"><a href="processing-scrnaseq-data.html#visualization"><i class="fa fa-check"></i><b>5.5</b> Visualization</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html"><i class="fa fa-check"></i><b>6</b> Introduction R/Bioconductor</a><ul>
<li class="chapter" data-level="6.1" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#start-environment"><i class="fa fa-check"></i><b>6.1</b> Start Environment</a><ul>
<li class="chapter" data-level="6.1.1" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#local-command"><i class="fa fa-check"></i><b>6.1.1</b> Local Command</a></li>
<li class="chapter" data-level="6.1.2" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#aws-command"><i class="fa fa-check"></i><b>6.1.2</b> AWS Command</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#installing-packages"><i class="fa fa-check"></i><b>6.2</b> Installing packages</a><ul>
<li class="chapter" data-level="6.2.1" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#cran"><i class="fa fa-check"></i><b>6.2.1</b> CRAN</a></li>
<li class="chapter" data-level="6.2.2" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#github"><i class="fa fa-check"></i><b>6.2.2</b> Github</a></li>
<li class="chapter" data-level="6.2.3" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#bioconductor"><i class="fa fa-check"></i><b>6.2.3</b> Bioconductor</a></li>
<li class="chapter" data-level="6.2.4" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#source"><i class="fa fa-check"></i><b>6.2.4</b> Source</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#installation-instructions"><i class="fa fa-check"></i><b>6.3</b> Installation instructions:</a><ul>
<li class="chapter" data-level="6.3.1" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#classestypes"><i class="fa fa-check"></i><b>6.3.1</b> Classes/Types</a></li>
<li class="chapter" data-level="6.3.2" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#data-structures"><i class="fa fa-check"></i><b>6.3.2</b> Data structures</a></li>
<li class="chapter" data-level="6.3.3" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#detour-to-s3s4"><i class="fa fa-check"></i><b>6.3.3</b> Detour to S3/S4</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#more-information"><i class="fa fa-check"></i><b>6.4</b> More information</a><ul>
<li class="chapter" data-level="6.4.1" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#checking-for-help-for-any-function"><i class="fa fa-check"></i><b>6.4.1</b> Checking for help for any function!</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#grammer-of-graphics-ggplot2"><i class="fa fa-check"></i><b>6.5</b> Grammer of Graphics (ggplot2)</a><ul>
<li class="chapter" data-level="6.5.1" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#what-is-ggplot2"><i class="fa fa-check"></i><b>6.5.1</b> What is ggplot2?</a></li>
<li class="chapter" data-level="6.5.2" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#principles-of-ggplot2"><i class="fa fa-check"></i><b>6.5.2</b> Principles of ggplot2</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="introduction-rbioconductor.html"><a href="introduction-rbioconductor.html#reference"><i class="fa fa-check"></i><b>6.6</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>7</b> Quality Control</a><ul>
<li class="chapter" data-level="7.1" data-path="quality-control.html"><a href="quality-control.html#slides-1"><i class="fa fa-check"></i><b>7.1</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html"><i class="fa fa-check"></i><b>8</b> Data Wrangling scRNAseq</a><ul>
<li class="chapter" data-level="8.1" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#goal-1"><i class="fa fa-check"></i><b>8.1</b> Goal</a></li>
<li class="chapter" data-level="8.2" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#introduction-2"><i class="fa fa-check"></i><b>8.2</b> Introduction</a><ul>
<li class="chapter" data-level="8.2.1" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#load-necessary-packages"><i class="fa fa-check"></i><b>8.2.1</b> Load necessary packages</a></li>
<li class="chapter" data-level="8.2.2" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#read-in-nsclc-counts-matrix."><i class="fa fa-check"></i><b>8.2.2</b> Read in NSCLC counts matrix.</a></li>
<li class="chapter" data-level="8.2.3" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#lets-examine-the-sparse-counts-matrix"><i class="fa fa-check"></i><b>8.2.3</b> Let’s examine the sparse counts matrix</a></li>
<li class="chapter" data-level="8.2.4" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#how-big-is-the-matrix"><i class="fa fa-check"></i><b>8.2.4</b> How big is the matrix?</a></li>
<li class="chapter" data-level="8.2.5" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#how-much-memory-does-a-sparse-matrix-take-up-relative-to-a-dense-matrix"><i class="fa fa-check"></i><b>8.2.5</b> How much memory does a sparse matrix take up relative to a dense matrix?</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#filtering-low-quality-cells"><i class="fa fa-check"></i><b>8.3</b> Filtering low-quality cells</a><ul>
<li class="chapter" data-level="8.3.1" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#look-at-the-summary-counts-for-genes-and-cells"><i class="fa fa-check"></i><b>8.3.1</b> Look at the summary counts for genes and cells</a></li>
<li class="chapter" data-level="8.3.2" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#plot-cells-ranked-by-their-number-of-detected-genes."><i class="fa fa-check"></i><b>8.3.2</b> Plot cells ranked by their number of detected genes.</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#beginning-with-seurat-httpsatijalab.orgseurat"><i class="fa fa-check"></i><b>8.4</b> Beginning with Seurat: <span>http://satijalab.org/seurat/</span></a><ul>
<li class="chapter" data-level="8.4.1" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#creating-a-seurat-object"><i class="fa fa-check"></i><b>8.4.1</b> Creating a seurat object</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#preprocessing-step-1-filter-out-low-quality-cells"><i class="fa fa-check"></i><b>8.5</b> Preprocessing step 1 : Filter out low-quality cells</a></li>
<li class="chapter" data-level="8.6" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#examine-contents-of-seurat-object"><i class="fa fa-check"></i><b>8.6</b> Examine contents of Seurat object</a><ul>
<li class="chapter" data-level="8.6.1" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#preprocessing-step-2-expression-normalization"><i class="fa fa-check"></i><b>8.6.1</b> Preprocessing step 2 : Expression normalization</a></li>
</ul></li>
<li class="chapter" data-level="8.7" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#detection-of-variable-genes-across-the-single-cells"><i class="fa fa-check"></i><b>8.7</b> Detection of variable genes across the single cells</a></li>
<li class="chapter" data-level="8.8" data-path="data-wrangling-scrnaseq.html"><a href="data-wrangling-scrnaseq.html#gene-set-expression-across-cells"><i class="fa fa-check"></i><b>8.8</b> Gene set expression across cells</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="identifying-cell-populations.html"><a href="identifying-cell-populations.html"><i class="fa fa-check"></i><b>9</b> Identifying Cell Populations</a><ul>
<li class="chapter" data-level="9.1" data-path="identifying-cell-populations.html"><a href="identifying-cell-populations.html#slides-2"><i class="fa fa-check"></i><b>9.1</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html"><i class="fa fa-check"></i><b>10</b> Feature Selection and Cluster Analysis</a><ul>
<li class="chapter" data-level="10.1" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#abstract"><i class="fa fa-check"></i><b>10.1</b> Abstract</a></li>
<li class="chapter" data-level="10.2" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#seurat-tutorial-redo"><i class="fa fa-check"></i><b>10.2</b> Seurat Tutorial Redo</a><ul>
<li class="chapter" data-level="10.2.1" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#preprocessing-steps"><i class="fa fa-check"></i><b>10.2.1</b> Preprocessing Steps</a></li>
<li class="chapter" data-level="10.2.2" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#start-of-identifying-cell-types"><i class="fa fa-check"></i><b>10.2.2</b> Start of Identifying Cell Types</a></li>
<li class="chapter" data-level="10.2.3" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#run-non-linear-dimensional-reduction-umaptsne"><i class="fa fa-check"></i><b>10.2.3</b> Run non-linear dimensional reduction (UMAP/tSNE)</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#feature-selection"><i class="fa fa-check"></i><b>10.3</b> Feature Selection</a><ul>
<li class="chapter" data-level="10.3.1" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#differential-expression-analysis"><i class="fa fa-check"></i><b>10.3.1</b> Differential Expression Analysis</a></li>
<li class="chapter" data-level="10.3.2" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#check-clusters"><i class="fa fa-check"></i><b>10.3.2</b> Check Clusters</a></li>
<li class="chapter" data-level="10.3.3" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#view-entire-object-structure"><i class="fa fa-check"></i><b>10.3.3</b> View Entire Object Structure</a></li>
<li class="chapter" data-level="10.3.4" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#practice-visualizingembedding"><i class="fa fa-check"></i><b>10.3.4</b> Practice Visualizing/Embedding</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="feature-selection-and-cluster-analysis.html"><a href="feature-selection-and-cluster-analysis.html#other-options-for-analysis"><i class="fa fa-check"></i><b>10.4</b> Other Options For Analysis</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="batch-correction-lecture.html"><a href="batch-correction-lecture.html"><i class="fa fa-check"></i><b>11</b> Batch Correction Lecture</a><ul>
<li class="chapter" data-level="11.1" data-path="batch-correction-lecture.html"><a href="batch-correction-lecture.html#slides-3"><i class="fa fa-check"></i><b>11.1</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html"><i class="fa fa-check"></i><b>12</b> Batch Correction Lab</a><ul>
<li class="chapter" data-level="12.1" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#load-settings-and-packages"><i class="fa fa-check"></i><b>12.1</b> Load settings and packages</a></li>
<li class="chapter" data-level="12.2" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#read-in-pancreas-expression-matrices"><i class="fa fa-check"></i><b>12.2</b> Read in pancreas expression matrices</a></li>
<li class="chapter" data-level="12.3" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#preparing-the-individual-seurat-objects-for-each-pancreas-dataset-without-batch-correction"><i class="fa fa-check"></i><b>12.3</b> Preparing the individual Seurat objects for each pancreas dataset without batch correction</a></li>
<li class="chapter" data-level="12.4" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#cluster-pancreatic-datasets-without-batch-correction"><i class="fa fa-check"></i><b>12.4</b> Cluster pancreatic datasets without batch correction</a><ul>
<li class="chapter" data-level="12.4.1" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#batch-correction-canonical-correlation-analysis-cca-mutual-nearest-neighbors-mnn-using-seurat-v3"><i class="fa fa-check"></i><b>12.4.1</b> Batch correction: canonical correlation analysis (CCA) + mutual nearest neighbors (MNN) using Seurat v3</a></li>
<li class="chapter" data-level="12.4.2" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#batch-correction-integrative-non-negative-matrix-factorization-nmf-using-liger"><i class="fa fa-check"></i><b>12.4.2</b> Batch correction: integrative non-negative matrix factorization (NMF) using LIGER</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#additional-exploration-regressing-out-unwanted-covariates"><i class="fa fa-check"></i><b>12.5</b> Additional exploration: Regressing out unwanted covariates</a></li>
<li class="chapter" data-level="12.6" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#additional-exploration-kbet"><i class="fa fa-check"></i><b>12.6</b> Additional exploration: kBET</a></li>
<li class="chapter" data-level="12.7" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#additional-exploration-seurat-3"><i class="fa fa-check"></i><b>12.7</b> Additional exploration: Seurat 3</a></li>
<li class="chapter" data-level="12.8" data-path="batch-correction-lab.html"><a href="batch-correction-lab.html#acknowledgements"><i class="fa fa-check"></i><b>12.8</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="advanced-topics-pseudotime-cell-trajectories.html"><a href="advanced-topics-pseudotime-cell-trajectories.html"><i class="fa fa-check"></i><b>13</b> Advanced Topics: Pseudotime Cell Trajectories</a><ul>
<li class="chapter" data-level="13.1" data-path="advanced-topics-pseudotime-cell-trajectories.html"><a href="advanced-topics-pseudotime-cell-trajectories.html#slides-4"><i class="fa fa-check"></i><b>13.1</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html"><i class="fa fa-check"></i><b>14</b> Trajectory Analysis</a><ul>
<li class="chapter" data-level="14.1" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#load-settings-and-packages-1"><i class="fa fa-check"></i><b>14.1</b> Load settings and packages</a></li>
<li><a href="trajectory-analysis.html#first-look-at-the-differentiation-data-from-deng-et-al.we-will-use-a-nice-smart-seq2-single-cell-rna-seq-data-from-single-cell-rna-seq-reveals-dynamic-random-monoallelic-gene-expression-in-mammalian-cells.-here-is-one-relevant-detail-from-their-paper-to-investigate-allele-specific-gene-expression-at-single-cell-resolution-we-isolated-269-individual-cells-dissociated-from-in-vivo-f1-embryos-casteij-c57bl6j-hereafter-abbreviated-as-cast-and-c57-respectively-from-oocyte-to-blastocyst-stages-of-mouse-preimplantation-development-pd"><span class="toc-section-number">14.2</span> First look at the differentiation data from Deng et
al. We will use a nice SMART-Seq2 single cell RNA-seq data from <span>Single-Cell RNA-Seq Reveals Dynamic, Random Monoallelic Gene Expression in Mammalian Cells</span>. Here is one relevant detail from their paper: “To investigate allele-specific gene expression at single-cell resolution, we isolated 269 individual cells dissociated from in vivo F1 embryos (CAST/EiJ × C57BL/6J, hereafter abbreviated as CAST and C57, respectively) from oocyte to blastocyst stages of mouse preimplantation development (PD)”</a></li>
<li class="chapter" data-level="14.3" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#principle-components-analysis"><i class="fa fa-check"></i><b>14.3</b> Principle Components Analysis</a></li>
<li class="chapter" data-level="14.4" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#diffusion-map-pseudotime"><i class="fa fa-check"></i><b>14.4</b> Diffusion map pseudotime</a></li>
<li class="chapter" data-level="14.5" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#slingshot-map-pseudotime"><i class="fa fa-check"></i><b>14.5</b> Slingshot map pseudotime</a></li>
<li class="chapter" data-level="14.6" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#find-temporally-expressed-genes"><i class="fa fa-check"></i><b>14.6</b> Find temporally expressed genes</a></li>
<li class="chapter" data-level="14.7" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#comparison-of-the-different-trajectory-inference-methods"><i class="fa fa-check"></i><b>14.7</b> Comparison of the different trajectory inference methods</a></li>
<li class="chapter" data-level="14.8" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#plots-of-gene-expression-over-time."><i class="fa fa-check"></i><b>14.8</b> Plots of gene expression over time.</a></li>
<li class="chapter" data-level="14.9" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#acknowledgements-1"><i class="fa fa-check"></i><b>14.9</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="transcriptome-quantification-1.html"><a href="transcriptome-quantification-1.html"><i class="fa fa-check"></i><b>15</b> Transcriptome Quantification</a><ul>
<li class="chapter" data-level="15.1" data-path="transcriptome-quantification-1.html"><a href="transcriptome-quantification-1.html#slides-5"><i class="fa fa-check"></i><b>15.1</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="cite-seq.html"><a href="cite-seq.html"><i class="fa fa-check"></i><b>16</b> Cite-Seq</a><ul>
<li class="chapter" data-level="16.1" data-path="cite-seq.html"><a href="cite-seq.html#load-settings-and-packages-2"><i class="fa fa-check"></i><b>16.1</b> Load settings and packages</a></li>
<li class="chapter" data-level="16.2" data-path="cite-seq.html"><a href="cite-seq.html#load-in-the-data"><i class="fa fa-check"></i><b>16.2</b> Load in the data</a></li>
<li class="chapter" data-level="16.3" data-path="cite-seq.html"><a href="cite-seq.html#setup-a-seurat-object-and-cluster-cells-based-on-rna-expression"><i class="fa fa-check"></i><b>16.3</b> Setup a Seurat object, and cluster cells based on RNA expression</a></li>
<li class="chapter" data-level="16.4" data-path="cite-seq.html"><a href="cite-seq.html#add-the-protein-expression-levels-to-the-seurat-object"><i class="fa fa-check"></i><b>16.4</b> Add the protein expression levels to the Seurat object</a></li>
<li class="chapter" data-level="16.5" data-path="cite-seq.html"><a href="cite-seq.html#visualize-protein-levels-on-rna-clusters"><i class="fa fa-check"></i><b>16.5</b> Visualize protein levels on RNA clusters</a></li>
<li class="chapter" data-level="16.6" data-path="cite-seq.html"><a href="cite-seq.html#identify-differentially-expressed-proteins-between-clusters"><i class="fa fa-check"></i><b>16.6</b> Identify differentially expressed proteins between clusters</a></li>
<li class="chapter" data-level="16.7" data-path="cite-seq.html"><a href="cite-seq.html#cluster-directly-on-protein-levels"><i class="fa fa-check"></i><b>16.7</b> Cluster directly on protein levels</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="additional-exploration-another-example-of-multi-modal-analysis.html"><a href="additional-exploration-another-example-of-multi-modal-analysis.html"><i class="fa fa-check"></i><b>17</b> Additional exploration: another example of multi-modal analysis</a><ul>
<li class="chapter" data-level="17.1" data-path="additional-exploration-another-example-of-multi-modal-analysis.html"><a href="additional-exploration-another-example-of-multi-modal-analysis.html#acknowledgements-2"><i class="fa fa-check"></i><b>17.1</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="single-cell-resources.html"><a href="single-cell-resources.html"><i class="fa fa-check"></i><b>18</b> Single Cell Resources</a><ul>
<li class="chapter" data-level="18.1" data-path="single-cell-resources.html"><a href="single-cell-resources.html#comprehensive-list-of-single-cell-resources"><i class="fa fa-check"></i><b>18.1</b> Comprehensive list of single-cell resources</a></li>
<li class="chapter" data-level="18.2" data-path="single-cell-resources.html"><a href="single-cell-resources.html#computational-packages-for-single-cell-analysis"><i class="fa fa-check"></i><b>18.2</b> Computational packages for single-cell analysis</a></li>
<li class="chapter" data-level="18.3" data-path="single-cell-resources.html"><a href="single-cell-resources.html#elife-commentary-on-the-human-cell-atlas"><i class="fa fa-check"></i><b>18.3</b> eLife Commentary on the Human Cell Atlas</a></li>
<li class="chapter" data-level="18.4" data-path="single-cell-resources.html"><a href="single-cell-resources.html#online-courses"><i class="fa fa-check"></i><b>18.4</b> Online courses</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ANALYSIS OF SINGLE CELL RNA-SEQ DATA</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="processing-scrnaseq-data" class="section level1">
<h1><span class="header-section-number">5</span> Processing scRNAseq Data</h1>
<div id="goal" class="section level2">
<h2><span class="header-section-number">5.1</span> Goal</h2>
<ul>
<li>To give you experience with examining and aligning fastq files</li>
</ul>
</div>
<div id="further-reading" class="section level2">
<h2><span class="header-section-number">5.2</span> Further reading</h2>
<p>This lab is based on a lab given in:
<a href="http://hemberg-lab.github.io/scRNA.seq.course/processing-raw-scrna-seq-data.html" class="uri">http://hemberg-lab.github.io/scRNA.seq.course/processing-raw-scrna-seq-data.html</a>
For more exercises and ideas please visit their web-site!</p>
</div>
<div id="download-data" class="section level2">
<h2><span class="header-section-number">5.3</span> Download data</h2>
<p>Please downlaod the 6 files from the dropbox folder: <a href="https://www.dropbox.com/sh/98573jes82w0fi7/AAB7Yhwe05MCZTnTZmppxZuta?dl=0" class="uri">https://www.dropbox.com/sh/98573jes82w0fi7/AAB7Yhwe05MCZTnTZmppxZuta?dl=0</a> into the data folder of your copy of github:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1">cd <span class="dv">2020</span>_scWorkshop<span class="op">/</span>data</a>
<a class="sourceLine" id="cb28-2" data-line-number="2">mkdir lab2data</a></code></pre></div>
<p>and copy the files into it
## FastQC</p>
<p>Once you’ve obtained your single-cell RNA-seq data, the first thing you need to do with it is check the quality of the reads you have sequenced. For this task, today we will be using a tool called FastQC. FastQC is a quality control tool for sequencing data, which can be used for both bulk and single-cell RNA-seq data. FastQC takes sequencing data as input and returns a report on read quality. Copy and paste this link into your browser to visit the FastQC website:</p>
<p><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" class="uri">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a></p>
<p>This website contains links to download and install FastQC and documentation on the reports produced. Fortunately we have already installed FastQC for you today, so instead we will take a look at the documentation. Scroll down the webpage to ‘Example Reports’ and click ‘Good Illumina Data’. This gives an example of what an ideal report should look like for high quality Illumina reads data.</p>
<p>Now let’s make a FastQC report ourselves.</p>
<p>Today we will be performing our analysis using a single cell from an mESC dataset produced by (Kolodziejczyk et al. 2015), which you downloaded froom the github link. The cells were sequenced using the SMART-seq2 library preparation protocol and the reads are paired end.</p>
<p>First, let’s open the docker in a bash mode. open a terminal, cd to the docer folder (the folder you downloaded from github) and run this command:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1">docker run <span class="op">--</span>rm <span class="op">-</span>ti <span class="op">-</span>v <span class="op">$</span>PWD<span class="op">:</span><span class="er">/</span>home<span class="op">/</span>rstudio <span class="op">-</span>e DISABLE_AUTH=true kdgosik<span class="op">/</span>2020scworkshop bash</a></code></pre></div>
<p>navigate to your data folder:</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" data-line-number="1">cd home<span class="op">/</span>rstudio<span class="op">/</span>lab2data</a>
<a class="sourceLine" id="cb30-2" data-line-number="2">ls</a></code></pre></div>
<p>You should see the files that you downloaded from the dropbox link.</p>
<p>Now let’s look at the files:</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1">less Teichmann_2i_<span class="dv">2</span>_<span class="dv">2</span>_<span class="fl">1.</span>fastq</a>
<a class="sourceLine" id="cb31-2" data-line-number="2">less Teichmann_2i_<span class="dv">2</span>_<span class="dv">2</span>_<span class="fl">2.</span>fastq</a></code></pre></div>
<p>We run fastqc from /usr/local/src/FastQC. You may need to give yourself permissions to run the file (hint: chmod)</p>
<p>Task 1: run fastqc to view the quality of the reads</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" data-line-number="1">chmod <span class="dv">755</span> <span class="op">/</span>usr<span class="op">/</span>local<span class="op">/</span>src<span class="op">/</span>FastQC<span class="op">/</span>fastqc</a>
<a class="sourceLine" id="cb32-2" data-line-number="2"><span class="op">/</span>usr<span class="op">/</span>local<span class="op">/</span>src<span class="op">/</span>FastQC<span class="op">/</span>fastqc <span class="op">-</span>h</a></code></pre></div>
<p>This command will tell you what options are available to pass to FastQC. Let us direct our output to our personal directories (under the folder results). Feel free to ask for help if you get stuck! If you are successful, you should generate a .zip and a .html file for both the forwards and the reverse reads files. Once you have been successful, feel free to have a go at the next section.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1"><span class="op">/</span>usr<span class="op">/</span>local<span class="op">/</span>src<span class="op">/</span>FastQC<span class="op">/</span>fastqc <span class="op">-</span>o <span class="op">&lt;</span>output_folder<span class="op">&gt;</span><span class="st"> </span>Teichmann_2i_<span class="dv">2</span>_<span class="dv">2</span>_<span class="fl">1.</span>fastq Teichmann_2i_<span class="dv">2</span>_<span class="dv">2</span>_<span class="fl">2.</span>fastq</a></code></pre></div>
<p>Once the command has finished executing, you should have a total of four files - one zip file for each of the paired end reads, and one html file for each of the paired end reads. The report is in the html file.</p>
<p>for those working in AWS, if you want to view the file you will need to download it to your computer. The scp command is:</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1">scp <span class="op">-</span>r <span class="op">-</span>i <span class="op">&lt;</span>your pem file<span class="op">&gt;</span><span class="st"> </span><span class="er">&lt;</span>username<span class="op">&gt;</span><span class="er">@</span>ec2<span class="dv">-34-213-180</span><span class="fl">-241.</span>us<span class="op">-</span>west<span class="fl">-2.</span>compute.amazonaws.com<span class="op">:</span><span class="er">~/&lt;</span>file to copy<span class="op">&gt;</span><span class="st"> </span><span class="er">&lt;</span>destination <span class="cf">in</span> your computer<span class="op">&gt;</span></a></code></pre></div>
<p>Once the file is on you computer, click on it. Your FastQC report should open. Have a look through the file. Remember to look at both the forwards and the reverse end read reports! How good quality are the reads? Is there anything we should be concerned about?</p>
<div id="fastq-file-format" class="section level3">
<h3><span class="header-section-number">5.3.1</span> Fastq file format</h3>
<p>FastQ is the most raw form of scRNASeq data you will encounter. All scRNASeq protocols are sequenced with paired-end sequencing. Barcode sequences may occur in one or both reads depending on the protocol employed. However, protocols using unique molecular identifiers (UMIs) will generally contain one read with the cell and UMI barcodes plus adapters but without any transcript sequence. Thus reads will be mapped as if they are single-end sequenced despite actually being paired end.</p>
<p>FastQ files have the format:</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1"><span class="op">&gt;</span>ReadID</a>
<a class="sourceLine" id="cb35-2" data-line-number="2">READ SEQUENCE</a>
<a class="sourceLine" id="cb35-3" data-line-number="3"><span class="op">+</span></a>
<a class="sourceLine" id="cb35-4" data-line-number="4">SEQUENCING QUALITY SCORES</a></code></pre></div>
</div>
</div>
<div id="align-the-reads" class="section level2">
<h2><span class="header-section-number">5.4</span> Align the reads</h2>
<div id="star-align" class="section level3">
<h3><span class="header-section-number">5.4.1</span> STAR align</h3>
<p>Now we have established that our reads are of good quality, we would like to map them to a reference genome. This process is known as alignment. Some form of alignment is generally required if we want to quantify gene expression or find genes which are differentially expressed between samples.</p>
<p>Many tools have been developed for read alignment, but today we will focus on STAR. For each read in our reads data, STAR tries to find the longest possible sequence which matches one or more sequences in the reference genome. Because STAR is able to recognize splicing events in this way, it is described as a ‘splice aware’ aligner.</p>
<p>Usually STAR aligns reads to a reference genome, potentially allowing it to detect novel splicing events or chromosomal rearrangements. However, one issue with STAR is that it needs a lot of RAM, especially if your reference genome is large (eg. mouse and human). To speed up our analysis today, we will use STAR to align reads from to a reference transcriptome of 2000 transcripts. Note that this is NOT normal or recommended practice, we only do it here for reasons of time. We recommend that normally you should align to a reference genome.</p>
<p>Two steps are required to perform STAR alignment. In the first step, the user provides STAR with reference genome sequences (FASTA) and annotations (GTF), which STAR uses to create a genome index. In the second step, STAR maps the user’s reads data to the genome index.</p>
<p>Let’s create the index now. Remember, for reasons of time we are aligning to a transcriptome rather than a genome today, meaning we only need to provide STAR with the sequences of the transcripts we will be aligning reads to. You can obtain transcriptomes for many model organisms from Ensembl (<a href="https://www.ensembl.org/info/data/ftp/index.html" class="uri">https://www.ensembl.org/info/data/ftp/index.html</a>).</p>
<p>Task 2: Create a genome index</p>
<p>First create the output folder for the index in your personal folder under results (recommended /home/rstudio/lab2data/STAR/indices).</p>
<p>We run STAR from:</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" data-line-number="1"><span class="op">/</span>usr<span class="op">/</span>local<span class="op">/</span>src<span class="op">/</span>STAR<span class="op">/</span>bin<span class="op">/</span>Linux_x86_<span class="dv">64</span></a></code></pre></div>
<p>using the command:</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb37-1" data-line-number="1"><span class="op">/</span>usr<span class="op">/</span>local<span class="op">/</span>src<span class="op">/</span>STAR<span class="op">/</span>bin<span class="op">/</span>Linux_x86_<span class="dv">64</span><span class="op">/</span>STAR <span class="op">--</span>runThreadN <span class="dv">4</span> <span class="op">--</span>runMode genomeGenerate <span class="op">--</span>genomeDir <span class="op">&lt;</span>output STAR indices folder<span class="op">&gt;</span><span class="st"> </span><span class="op">--</span>genomeFastaFiles <span class="op">/</span>home<span class="op">/</span>rstudio<span class="op">/</span>lab2data<span class="op">/</span><span class="dv">2000</span>_reference.transcripts.fa</a></code></pre></div>
<p>Now that we have created the index, we can perform the mapping step.</p>
<p>Task 4: Try to work out what command you should use to map our fastq files to the index you created. Use the STAR manual to help you. Once you think you know the answer use ./STAR command to align the fastq files to a BAM file.
You can either create a SAM file and convert it to BAM using samtools, or use STAR to directly output a BAM file (–outSAMtype BAM Unsorted)</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb38-1" data-line-number="1"><span class="op">/</span>usr<span class="op">/</span>local<span class="op">/</span>src<span class="op">/</span>STAR<span class="op">/</span>bin<span class="op">/</span>Linux_x86_<span class="dv">64</span><span class="op">/</span>STAR <span class="op">--</span>runThreadN <span class="dv">4</span> <span class="op">--</span>genomeDir <span class="op">&lt;</span>genome_reference<span class="op">&gt;</span><span class="st"> </span><span class="op">--</span>readFilesIn <span class="op">/</span>home<span class="op">/</span>rstudio<span class="op">/</span>lab2data<span class="op">/</span>Teichmann_2i_<span class="dv">2</span>_<span class="dv">2</span>_<span class="fl">1.</span>fastq <span class="op">/</span>home<span class="op">/</span>rstudio<span class="op">/</span>lab2data<span class="op">/</span>Teichmann_2i_<span class="dv">2</span>_<span class="dv">2</span>_<span class="fl">2.</span>fastq <span class="op">--</span>outFileNamePrefix <span class="op">&lt;</span>output_folder<span class="op">&gt;</span><span class="st"> </span><span class="op">--</span>outSAMtype BAM Unsorted</a></code></pre></div>
<p>The alignment may take awhile, if you wish to you can complete tasks 7-10 in the meanwhile.</p>
</div>
<div id="bam-file-format" class="section level3">
<h3><span class="header-section-number">5.4.2</span> Bam file format</h3>
<p>BAM file format stores mapped reads in a standard and efficient manner. The human-readable version is called a SAM file, while the BAM file is the highly compressed version. BAM/SAM files contain a header which typically includes
information on the sample preparation, sequencing and mapping; and a tab-separated row for each individual alignment of each read.</p>
<p>Alignment rows employ a standard format with the following columns:</p>
<p>QNAME : read name (generally will include UMI barcode if applicable)</p>
<p>FLAG : number tag indicating the “type” of alignment, link to explanation of all possible “types”</p>
<p>RNAME : reference sequence name (i.e. chromosome read is mapped to).</p>
<p>POS : leftmost mapping position</p>
<p>MAPQ : Mapping quality</p>
<p>CIGAR : string indicating the matching/mismatching parts of the read (may include soft-clipping).</p>
<p>RNEXT : reference name of the mate/next read</p>
<p>PNEXT : POS for mate/next read</p>
<p>TLEN : Template length (length of reference region the read is mapped to)</p>
<p>SEQ : read sequence</p>
<p>QUAL : read quality</p>
<p>BAM/SAM files can be converted to the other format using ‘samtools’:</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb39-1" data-line-number="1">samtools view <span class="op">-</span>S <span class="op">-</span>b file.sam <span class="op">&gt;</span><span class="st"> </span>file.bam</a>
<a class="sourceLine" id="cb39-2" data-line-number="2">samtools view <span class="op">-</span>h file.bam <span class="op">&gt;</span><span class="st"> </span>file.sam</a></code></pre></div>
<p>Some sequencing facilities will automatically map your reads to the a standard genome and deliver either BAM or CRAM formatted files. Generally they will not have included ERCC sequences in the genome thus no ERCC reads will be mapped in the BAM/CRAM file. To quantify ERCCs (or any other genetic alterations) or if you just want to use a different alignment algorithm than whatever is in the generic pipeline (often outdated), then you will need to convert the BAM/CRAM files back to FastQs:</p>
<p>BAM files can be converted to FastQ using bedtools. To ensure a single copy for multi-mapping reads first sort by read name and remove secondary alignments using samtools. Picard also contains a method for converting BAM to FastQ files.</p>
<p>Bonus:</p>
<p>To make our aligned BAM file easy to navigate (needed for IGViewer) we will sort and index it using samtools.
Sam tools can be run from everywhere (no need to go to a special directory!) using the command:</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb40-1" data-line-number="1">samtools</a></code></pre></div>
<p>Let us start by sorting the BAM file:</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb41-1" data-line-number="1">samtools sort Aligned.out.bam <span class="op">-</span>o Aligned.out.sorted.bam</a></code></pre></div>
<p>Task 6: can you index the file? hint: try looking at samtools -h</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1">samtools index Aligned.out.sorted.bam</a></code></pre></div>
<p>Once you sorted and indexed the files you should have a BAM and a bai files. The BAM file is the aligned reads, and the bai is an index file. To view them in IGViewer (IGV) first copy them into your computer. Go ahead and copy the fa file as well, we will need a reference genome file.</p>
</div>
</div>
<div id="visualization" class="section level2">
<h2><span class="header-section-number">5.5</span> Visualization</h2>
<p>To view the file we will use the IGV you installed on your personal computer.
Open IGV: the default genomes are human HG19 and HG38. Through the class we will be using the PBMC dataset. You have the BAM file in your data folder. Go ahead and transfer it to your computer and upload it to IGV with hg38 as reference genome.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb43-1" data-line-number="1">Task <span class="dv">7</span><span class="op">:</span><span class="st"> </span>Browse to MS4A1, this is a blood cell marker. Can you see the exons and the introns? Where are most of the aligned reads?</a>
<a class="sourceLine" id="cb43-2" data-line-number="2"></a>
<a class="sourceLine" id="cb43-3" data-line-number="3">Task <span class="dv">8</span><span class="op">:</span><span class="st"> </span>Search <span class="cf">in</span> IGV or online <span class="op">-</span><span class="st"> </span>can you present splice junctions? (right click -&gt;<span class="st"> </span>“Show splice junction track”)</a>
<a class="sourceLine" id="cb43-4" data-line-number="4"></a>
<a class="sourceLine" id="cb43-5" data-line-number="5">Task <span class="dv">9</span><span class="op">:</span><span class="st"> </span>Try further tasks that interest you <span class="cf">in</span> IGV. For example, can you detect reads that are within one exon and reads that start <span class="cf">in</span> one exon and continue <span class="cf">in</span> the <span class="cf">next</span>? Can you copy the sequence of exon2 <span class="cf">in</span> MS4A1?</a>
<a class="sourceLine" id="cb43-6" data-line-number="6"></a>
<a class="sourceLine" id="cb43-7" data-line-number="7">Task <span class="dv">10</span><span class="op">:</span><span class="st"> </span>What would have happened <span class="cf">if</span> you chose the wrong reference genome, such as hg19?</a></code></pre></div>
<p>Bonus 2 (IGV sometimes has difficulties loading small fa files. So if this becomes difficult - don’t worry! It’s not your alignment):</p>
<p>The default genomes are human HG19 and HG38. However you can also upload your reference genome of choice. As we created our own fasta file we can now upload it as a reference genome.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" data-line-number="1">Task <span class="dv">11</span><span class="op">:</span><span class="st"> </span>Load new genome<span class="op">:</span><span class="st"> </span>go to “Genomes”-&gt;”Load from file” and load the file <span class="dv">2000</span>_reference.transcripts.fa</a>
<a class="sourceLine" id="cb44-2" data-line-number="2"></a>
<a class="sourceLine" id="cb44-3" data-line-number="3">Task <span class="dv">12</span><span class="op">:</span><span class="st"> </span>Now load your reads<span class="op">:</span><span class="st"> </span>go to “File”-&gt;”Load from file” and load your BAM file. Notice that IGV needs a BAM and a bai saved <span class="cf">in</span> the same location. IGV uses the bai to navigate through the BAM file. </a>
<a class="sourceLine" id="cb44-4" data-line-number="4"></a>
<a class="sourceLine" id="cb44-5" data-line-number="5">Task <span class="dv">13</span><span class="op">:</span><span class="st"> </span>Some of the reads have a nucleotide substitution <span class="cf">in</span> position <span class="dv">993</span> <span class="op">-</span><span class="st"> </span>what is the reference nucleotide? What is the substitution?</a></code></pre></div>

<pre><code>## No non-system installation of Python could be found.
## Would you like to download and install Miniconda?
## Miniconda is an open source environment management system for Python.
## See https://docs.conda.io/en/latest/miniconda.html for more details.</code></pre>
<pre><code>## Installation aborted.</code></pre>
<pre><code>## Use suppressPackageStartupMessages() to eliminate package startup messages</code></pre>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="transcriptome-quantification.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="introduction-rbioconductor.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["2020_scWorkshop.pdf", "2020_scWorkshop.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
